<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.6.40">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">

<meta name="author" content="Nick Sunderland">
<meta name="dcterms.date" content="2025-01-10">

<title>Tissue-partitioned BMI instrument procedure – Nick Sunderland’s lab book</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
/* CSS for syntax highlighting */
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { display: inline-block; text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
  }
pre.numberSource { margin-left: 3em;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
</style>


<script src="../../site_libs/quarto-nav/quarto-nav.js"></script>
<script src="../../site_libs/quarto-nav/headroom.min.js"></script>
<script src="../../site_libs/clipboard/clipboard.min.js"></script>
<script src="../../site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="../../site_libs/quarto-search/fuse.min.js"></script>
<script src="../../site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="../../">
<script src="../../site_libs/quarto-html/quarto.js"></script>
<script src="../../site_libs/quarto-html/popper.min.js"></script>
<script src="../../site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="../../site_libs/quarto-html/anchor.min.js"></script>
<link href="../../site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="../../site_libs/quarto-html/quarto-syntax-highlighting-549806ee2085284f45b00abea8c6df48.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="../../site_libs/bootstrap/bootstrap.min.js"></script>
<link href="../../site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="../../site_libs/bootstrap/bootstrap-d045e70ebd9269c25e164ecdb4898f56.min.css" rel="stylesheet" append-hash="true" id="quarto-bootstrap" data-mode="light">
<script id="quarto-search-options" type="application/json">{
  "location": "navbar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "end",
  "type": "overlay",
  "limit": 50,
  "keyboard-shortcut": [
    "f",
    "/",
    "s"
  ],
  "show-item-context": false,
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-text-placeholder": "",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit",
    "search-label": "Search"
  }
}</script>


<link rel="stylesheet" href="../../styles.css">
</head>

<body class="nav-fixed fullcontent">

<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top">
    <nav class="navbar navbar-expand-lg " data-bs-theme="dark">
      <div class="navbar-container container-fluid">
      <div class="navbar-brand-container mx-auto">
    <a class="navbar-brand" href="../../index.html">
    <span class="navbar-title">Nick Sunderland’s lab book</span>
    </a>
  </div>
            <div id="quarto-search" class="" title="Search"></div>
          <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbarCollapse" aria-controls="navbarCollapse" role="menu" aria-expanded="false" aria-label="Toggle navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
  <span class="navbar-toggler-icon"></span>
</button>
          <div class="collapse navbar-collapse" id="navbarCollapse">
            <ul class="navbar-nav navbar-nav-scroll ms-auto">
  <li class="nav-item">
    <a class="nav-link" href="../../about.html"> 
<span class="menu-text">About</span></a>
  </li>  
  <li class="nav-item compact">
    <a class="nav-link" href="https://github.com/"> <i class="bi bi-github" role="img">
</i> 
<span class="menu-text"></span></a>
  </li>  
  <li class="nav-item compact">
    <a class="nav-link" href="https://twitter.com"> <i class="bi bi-twitter" role="img">
</i> 
<span class="menu-text"></span></a>
  </li>  
</ul>
          </div> <!-- /navcollapse -->
            <div class="quarto-navbar-tools">
</div>
      </div> <!-- /container-fluid -->
    </nav>
</header>
<!-- content -->
<div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-article page-navbar">
<!-- sidebar -->
<!-- margin-sidebar -->
    
<!-- main -->
<main class="content" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default">
<div class="quarto-title">
<h1 class="title">Tissue-partitioned BMI instrument procedure</h1>
</div>



<div class="quarto-title-meta">

    <div>
    <div class="quarto-title-meta-heading">Author</div>
    <div class="quarto-title-meta-contents">
             <p>Nick Sunderland </p>
          </div>
  </div>
    
    <div>
    <div class="quarto-title-meta-heading">Published</div>
    <div class="quarto-title-meta-contents">
      <p class="date">January 10, 2025</p>
    </div>
  </div>
  
    
  </div>
  


</header>


<section id="motivation" class="level2">
<h2 class="anchored" data-anchor-id="motivation">Motivation</h2>
<p>What does the weighting by PPH4 do to tissue partitioned BMI instruments? Let’s simulate some tissue partitioned BMI MR results.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb1"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(data.table)</span>
<span id="cb1-2"><a href="#cb1-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ggplot2)</span>
<span id="cb1-3"><a href="#cb1-3" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ggpubr)</span>
<span id="cb1-4"><a href="#cb1-4" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(genepi.utils)</span>
<span id="cb1-5"><a href="#cb1-5" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(tibble)</span>
<span id="cb1-6"><a href="#cb1-6" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(knitr)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="bmi-data" class="level2">
<h2 class="anchored" data-anchor-id="bmi-data">BMI data</h2>
<p>First load the BMI GWAS.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb2"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb2-1"><a href="#cb2-1" aria-hidden="true" tabindex="-1"></a>bmi    <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(<span class="fu">file.path</span>(TPMR_DIR, <span class="st">"output"</span>, <span class="st">"tmp_objects"</span>, <span class="st">"obj_bmi.RDS"</span>))</span>
<span id="cb2-2"><a href="#cb2-2" aria-hidden="true" tabindex="-1"></a>clumps <span class="ot">&lt;-</span> <span class="fu">clump</span>(<span class="fu">as.data.table</span>(bmi), <span class="at">p1=</span><span class="fl">5e-8</span>, <span class="at">r2=</span><span class="fl">0.001</span>, <span class="at">kb=</span><span class="dv">10000</span>, <span class="at">plink2=</span>PLINK2, <span class="at">plink_ref=</span>UKBB_REFERENCE)[index<span class="sc">==</span><span class="cn">TRUE</span>]</span>
<span id="cb2-3"><a href="#cb2-3" aria-hidden="true" tabindex="-1"></a>bmi    <span class="ot">&lt;-</span> <span class="fu">subset_gwas</span>(bmi, clumps<span class="sc">$</span>rsid)</span>
<span id="cb2-4"><a href="#cb2-4" aria-hidden="true" tabindex="-1"></a><span class="fu">kable</span>(<span class="fu">head</span>(clumps, <span class="dv">5</span>), <span class="at">caption =</span> <span class="st">"BMI lead variants"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<table class="caption-top table table-sm table-striped small">
<caption>BMI lead variants</caption>
<colgroup>
<col style="width: 4%">
<col style="width: 1%">
<col style="width: 4%">
<col style="width: 1%">
<col style="width: 1%">
<col style="width: 3%">
<col style="width: 4%">
<col style="width: 4%">
<col style="width: 1%">
<col style="width: 2%">
<col style="width: 2%">
<col style="width: 2%">
<col style="width: 13%">
<col style="width: 2%">
<col style="width: 3%">
<col style="width: 2%">
<col style="width: 1%">
<col style="width: 1%">
<col style="width: 1%">
<col style="width: 4%">
<col style="width: 4%">
<col style="width: 3%">
<col style="width: 3%">
<col style="width: 3%">
<col style="width: 4%">
<col style="width: 3%">
<col style="width: 4%">
<col style="width: 2%">
<col style="width: 2%">
</colgroup>
<thead>
<tr class="header">
<th style="text-align: left;">rsid</th>
<th style="text-align: left;">chr</th>
<th style="text-align: right;">bp</th>
<th style="text-align: left;">ea</th>
<th style="text-align: left;">oa</th>
<th style="text-align: right;">eaf</th>
<th style="text-align: right;">beta</th>
<th style="text-align: right;">se</th>
<th style="text-align: right;">p</th>
<th style="text-align: right;">n</th>
<th style="text-align: right;">ncase</th>
<th style="text-align: left;">trait</th>
<th style="text-align: left;">id</th>
<th style="text-align: left;">strand</th>
<th style="text-align: left;">imputed</th>
<th style="text-align: right;">info</th>
<th style="text-align: right;">q</th>
<th style="text-align: right;">q_p</th>
<th style="text-align: right;">i2</th>
<th style="text-align: left;">proxy_rsid</th>
<th style="text-align: left;">proxy_chr</th>
<th style="text-align: right;">proxy_bp</th>
<th style="text-align: left;">proxy_ea</th>
<th style="text-align: left;">proxy_oa</th>
<th style="text-align: right;">proxy_eaf</th>
<th style="text-align: right;">proxy_r2</th>
<th style="text-align: left;">snp</th>
<th style="text-align: left;">index</th>
<th style="text-align: left;">clump</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">rs10007906</td>
<td style="text-align: left;">4</td>
<td style="text-align: right;">31023610</td>
<td style="text-align: left;">A</td>
<td style="text-align: left;">C</td>
<td style="text-align: right;">0.360300</td>
<td style="text-align: right;">0.0133000</td>
<td style="text-align: right;">0.0018000</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">686479</td>
<td style="text-align: right;">NA</td>
<td style="text-align: left;">bmi</td>
<td style="text-align: left;">combined_bmi_giant_ukb_annot.txt</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">NA</td>
<td style="text-align: right;">NA</td>
<td style="text-align: right;">NA</td>
<td style="text-align: right;">NA</td>
<td style="text-align: right;">NA</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">NA</td>
<td style="text-align: right;">NA</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">NA</td>
<td style="text-align: right;">NA</td>
<td style="text-align: right;">NA</td>
<td style="text-align: left;">rs10007906</td>
<td style="text-align: left;">TRUE</td>
<td style="text-align: left;">291</td>
</tr>
<tr class="even">
<td style="text-align: left;">rs1000940</td>
<td style="text-align: left;">17</td>
<td style="text-align: right;">5283252</td>
<td style="text-align: left;">A</td>
<td style="text-align: left;">G</td>
<td style="text-align: right;">0.701300</td>
<td style="text-align: right;">-0.0154000</td>
<td style="text-align: right;">0.0018000</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">794558</td>
<td style="text-align: right;">NA</td>
<td style="text-align: left;">bmi</td>
<td style="text-align: left;">combined_bmi_giant_ukb_annot.txt</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">NA</td>
<td style="text-align: right;">NA</td>
<td style="text-align: right;">NA</td>
<td style="text-align: right;">NA</td>
<td style="text-align: right;">NA</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">NA</td>
<td style="text-align: right;">NA</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">NA</td>
<td style="text-align: right;">NA</td>
<td style="text-align: right;">NA</td>
<td style="text-align: left;">rs1000940</td>
<td style="text-align: left;">TRUE</td>
<td style="text-align: left;">142</td>
</tr>
<tr class="odd">
<td style="text-align: left;">rs1007934</td>
<td style="text-align: left;">14</td>
<td style="text-align: right;">73463479</td>
<td style="text-align: left;">A</td>
<td style="text-align: left;">G</td>
<td style="text-align: right;">0.415000</td>
<td style="text-align: right;">-0.0120000</td>
<td style="text-align: right;">0.0017000</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">692411</td>
<td style="text-align: right;">NA</td>
<td style="text-align: left;">bmi</td>
<td style="text-align: left;">combined_bmi_giant_ukb_annot.txt</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">NA</td>
<td style="text-align: right;">NA</td>
<td style="text-align: right;">NA</td>
<td style="text-align: right;">NA</td>
<td style="text-align: right;">NA</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">NA</td>
<td style="text-align: right;">NA</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">NA</td>
<td style="text-align: right;">NA</td>
<td style="text-align: right;">NA</td>
<td style="text-align: left;">rs1007934</td>
<td style="text-align: left;">TRUE</td>
<td style="text-align: left;">407</td>
</tr>
<tr class="even">
<td style="text-align: left;">rs10125137</td>
<td style="text-align: left;">9</td>
<td style="text-align: right;">118644190</td>
<td style="text-align: left;">T</td>
<td style="text-align: left;">C</td>
<td style="text-align: right;">0.641786</td>
<td style="text-align: right;">-0.0117273</td>
<td style="text-align: right;">0.0020523</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">463005</td>
<td style="text-align: right;">NA</td>
<td style="text-align: left;">bmi</td>
<td style="text-align: left;">combined_bmi_giant_ukb_annot.txt</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">NA</td>
<td style="text-align: right;">NA</td>
<td style="text-align: right;">NA</td>
<td style="text-align: right;">NA</td>
<td style="text-align: right;">NA</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">NA</td>
<td style="text-align: right;">NA</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">NA</td>
<td style="text-align: right;">NA</td>
<td style="text-align: right;">NA</td>
<td style="text-align: left;">rs10125137</td>
<td style="text-align: left;">TRUE</td>
<td style="text-align: left;">811</td>
</tr>
<tr class="odd">
<td style="text-align: left;">rs10131890</td>
<td style="text-align: left;">14</td>
<td style="text-align: right;">97258752</td>
<td style="text-align: left;">A</td>
<td style="text-align: left;">C</td>
<td style="text-align: right;">0.950300</td>
<td style="text-align: right;">-0.0222000</td>
<td style="text-align: right;">0.0039000</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">691777</td>
<td style="text-align: right;">NA</td>
<td style="text-align: left;">bmi</td>
<td style="text-align: left;">combined_bmi_giant_ukb_annot.txt</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">NA</td>
<td style="text-align: right;">NA</td>
<td style="text-align: right;">NA</td>
<td style="text-align: right;">NA</td>
<td style="text-align: right;">NA</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">NA</td>
<td style="text-align: right;">NA</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">NA</td>
<td style="text-align: right;">NA</td>
<td style="text-align: right;">NA</td>
<td style="text-align: left;">rs10131890</td>
<td style="text-align: left;">TRUE</td>
<td style="text-align: left;">846</td>
</tr>
</tbody>
</table>
</div>
</div>
</section>
<section id="heart-failure-data" class="level2">
<h2 class="anchored" data-anchor-id="heart-failure-data">Heart failure data</h2>
<p>Load the heart failure GWAS.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb3"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1" aria-hidden="true" tabindex="-1"></a>hf <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(<span class="fu">file.path</span>(TPMR_DIR, <span class="st">"output"</span>, <span class="st">"tmp_objects"</span>, <span class="st">"obj_hf_allcause.RDS"</span>))</span>
<span id="cb3-2"><a href="#cb3-2" aria-hidden="true" tabindex="-1"></a>hf <span class="ot">&lt;-</span> <span class="fu">subset_gwas</span>(hf, clumps<span class="sc">$</span>rsid)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="harmonise-the-datasets" class="level2">
<h2 class="anchored" data-anchor-id="harmonise-the-datasets">Harmonise the datasets</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb4"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb4-1"><a href="#cb4-1" aria-hidden="true" tabindex="-1"></a>h <span class="ot">&lt;-</span> <span class="fu">MR</span>(bmi, hf)</span>
<span id="cb4-2"><a href="#cb4-2" aria-hidden="true" tabindex="-1"></a><span class="fu">kable</span>(<span class="fu">head</span>(<span class="fu">as.data.table</span>(h),<span class="dv">5</span>), <span class="at">caption =</span> <span class="st">"Harmonised BMI:HF dataset"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<table class="caption-top table table-sm table-striped small">
<caption>Harmonised BMI:HF dataset</caption>
<colgroup>
<col style="width: 9%">
<col style="width: 2%">
<col style="width: 6%">
<col style="width: 1%">
<col style="width: 1%">
<col style="width: 7%">
<col style="width: 6%">
<col style="width: 1%">
<col style="width: 5%">
<col style="width: 4%">
<col style="width: 5%">
<col style="width: 7%">
<col style="width: 7%">
<col style="width: 6%">
<col style="width: 3%">
<col style="width: 5%">
<col style="width: 5%">
<col style="width: 7%">
</colgroup>
<thead>
<tr class="header">
<th style="text-align: left;">rsid</th>
<th style="text-align: left;">chr</th>
<th style="text-align: right;">bp</th>
<th style="text-align: left;">ea</th>
<th style="text-align: left;">oa</th>
<th style="text-align: right;">bx</th>
<th style="text-align: right;">bxse</th>
<th style="text-align: right;">px</th>
<th style="text-align: right;">by</th>
<th style="text-align: right;">byse</th>
<th style="text-align: right;">py</th>
<th style="text-align: left;">proxy_e_snp</th>
<th style="text-align: left;">proxy_o_snp</th>
<th style="text-align: left;">index_snp</th>
<th style="text-align: right;">group</th>
<th style="text-align: left;">ld_info</th>
<th style="text-align: left;">exposure</th>
<th style="text-align: left;">outcome</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">rs10007906_C_A</td>
<td style="text-align: left;">4</td>
<td style="text-align: right;">31023610</td>
<td style="text-align: left;">A</td>
<td style="text-align: left;">C</td>
<td style="text-align: right;">0.0133000</td>
<td style="text-align: right;">0.0018000</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">0.0039</td>
<td style="text-align: right;">0.0051</td>
<td style="text-align: right;">0.44980</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">TRUE</td>
<td style="text-align: right;">NA</td>
<td style="text-align: left;">FALSE</td>
<td style="text-align: left;">bmi</td>
<td style="text-align: left;">hf_allcause</td>
</tr>
<tr class="even">
<td style="text-align: left;">rs1000940_G_A</td>
<td style="text-align: left;">17</td>
<td style="text-align: right;">5283252</td>
<td style="text-align: left;">A</td>
<td style="text-align: left;">G</td>
<td style="text-align: right;">-0.0154000</td>
<td style="text-align: right;">0.0018000</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">0.0011</td>
<td style="text-align: right;">0.0050</td>
<td style="text-align: right;">0.82970</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">TRUE</td>
<td style="text-align: right;">NA</td>
<td style="text-align: left;">FALSE</td>
<td style="text-align: left;">bmi</td>
<td style="text-align: left;">hf_allcause</td>
</tr>
<tr class="odd">
<td style="text-align: left;">rs1007934_G_A</td>
<td style="text-align: left;">14</td>
<td style="text-align: right;">73463479</td>
<td style="text-align: left;">A</td>
<td style="text-align: left;">G</td>
<td style="text-align: right;">-0.0120000</td>
<td style="text-align: right;">0.0017000</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">-0.0092</td>
<td style="text-align: right;">0.0049</td>
<td style="text-align: right;">0.05931</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">TRUE</td>
<td style="text-align: right;">NA</td>
<td style="text-align: left;">FALSE</td>
<td style="text-align: left;">bmi</td>
<td style="text-align: left;">hf_allcause</td>
</tr>
<tr class="even">
<td style="text-align: left;">rs10125137_C_T</td>
<td style="text-align: left;">9</td>
<td style="text-align: right;">118644190</td>
<td style="text-align: left;">T</td>
<td style="text-align: left;">C</td>
<td style="text-align: right;">-0.0117273</td>
<td style="text-align: right;">0.0020523</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">0.0033</td>
<td style="text-align: right;">0.0049</td>
<td style="text-align: right;">0.50820</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">TRUE</td>
<td style="text-align: right;">NA</td>
<td style="text-align: left;">FALSE</td>
<td style="text-align: left;">bmi</td>
<td style="text-align: left;">hf_allcause</td>
</tr>
<tr class="odd">
<td style="text-align: left;">rs10131890_C_A</td>
<td style="text-align: left;">14</td>
<td style="text-align: right;">97258752</td>
<td style="text-align: left;">A</td>
<td style="text-align: left;">C</td>
<td style="text-align: right;">-0.0222000</td>
<td style="text-align: right;">0.0039000</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">-0.0211</td>
<td style="text-align: right;">0.0099</td>
<td style="text-align: right;">0.03310</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">TRUE</td>
<td style="text-align: right;">NA</td>
<td style="text-align: left;">FALSE</td>
<td style="text-align: left;">bmi</td>
<td style="text-align: left;">hf_allcause</td>
</tr>
</tbody>
</table>
</div>
</div>
</section>
<section id="run-the-mr" class="level2">
<h2 class="anchored" data-anchor-id="run-the-mr">Run the MR</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb5"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb5-1"><a href="#cb5-1" aria-hidden="true" tabindex="-1"></a>res <span class="ot">&lt;-</span> <span class="fu">run_mr</span>(h)</span>
<span id="cb5-2"><a href="#cb5-2" aria-hidden="true" tabindex="-1"></a><span class="fu">kable</span>(res, <span class="at">caption =</span> <span class="st">"BMI:HF Mendelian randomisation"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<table class="caption-top table table-sm table-striped small">
<caption>BMI:HF Mendelian randomisation</caption>
<colgroup>
<col style="width: 9%">
<col style="width: 5%">
<col style="width: 4%">
<col style="width: 5%">
<col style="width: 2%">
<col style="width: 4%">
<col style="width: 4%">
<col style="width: 1%">
<col style="width: 5%">
<col style="width: 4%">
<col style="width: 4%">
<col style="width: 4%">
<col style="width: 3%">
<col style="width: 4%">
<col style="width: 4%">
<col style="width: 7%">
<col style="width: 2%">
<col style="width: 4%">
<col style="width: 7%">
<col style="width: 7%">
</colgroup>
<thead>
<tr class="header">
<th style="text-align: left;">method</th>
<th style="text-align: left;">correlation</th>
<th style="text-align: left;">exposure</th>
<th style="text-align: left;">outcome</th>
<th style="text-align: right;">n_snp</th>
<th style="text-align: right;">b</th>
<th style="text-align: right;">b_se</th>
<th style="text-align: right;">p</th>
<th style="text-align: right;">intercept</th>
<th style="text-align: right;">int_se</th>
<th style="text-align: right;">int_p</th>
<th style="text-align: right;">qstat</th>
<th style="text-align: right;">qstat_p</th>
<th style="text-align: right;">fstat</th>
<th style="text-align: right;">condfstat</th>
<th style="text-align: right;">overdispersion</th>
<th style="text-align: right;">n_pc</th>
<th style="text-align: right;">n_hunted</th>
<th style="text-align: right;">slopehunter_pi</th>
<th style="text-align: right;">slopehunter_ent</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">mr_ivw</td>
<td style="text-align: left;">FALSE</td>
<td style="text-align: left;">bmi</td>
<td style="text-align: left;">hf_allcause</td>
<td style="text-align: right;">874</td>
<td style="text-align: right;">0.4872820</td>
<td style="text-align: right;">0.0165599</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">0.0000000</td>
<td style="text-align: right;">NA</td>
<td style="text-align: right;">NA</td>
<td style="text-align: right;">1592.460</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">60.70091</td>
<td style="text-align: right;">NA</td>
<td style="text-align: right;">NA</td>
<td style="text-align: right;">NA</td>
<td style="text-align: right;">NA</td>
<td style="text-align: right;">NA</td>
<td style="text-align: right;">NA</td>
</tr>
<tr class="even">
<td style="text-align: left;">mr_egger</td>
<td style="text-align: left;">FALSE</td>
<td style="text-align: left;">bmi</td>
<td style="text-align: left;">hf_allcause</td>
<td style="text-align: right;">874</td>
<td style="text-align: right;">0.5367727</td>
<td style="text-align: right;">0.0464790</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">-0.0008232</td>
<td style="text-align: right;">0.0007224</td>
<td style="text-align: right;">0.2544728</td>
<td style="text-align: right;">1590.092</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">NA</td>
<td style="text-align: right;">NA</td>
<td style="text-align: right;">NA</td>
<td style="text-align: right;">NA</td>
<td style="text-align: right;">NA</td>
<td style="text-align: right;">NA</td>
<td style="text-align: right;">NA</td>
</tr>
<tr class="odd">
<td style="text-align: left;">mr_weighted_median</td>
<td style="text-align: left;">FALSE</td>
<td style="text-align: left;">bmi</td>
<td style="text-align: left;">hf_allcause</td>
<td style="text-align: right;">874</td>
<td style="text-align: right;">0.5078217</td>
<td style="text-align: right;">0.0236696</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">0.0000000</td>
<td style="text-align: right;">NA</td>
<td style="text-align: right;">NA</td>
<td style="text-align: right;">NA</td>
<td style="text-align: right;">NA</td>
<td style="text-align: right;">NA</td>
<td style="text-align: right;">NA</td>
<td style="text-align: right;">NA</td>
<td style="text-align: right;">NA</td>
<td style="text-align: right;">NA</td>
<td style="text-align: right;">NA</td>
<td style="text-align: right;">NA</td>
</tr>
<tr class="even">
<td style="text-align: left;">mr_weighted_mode</td>
<td style="text-align: left;">FALSE</td>
<td style="text-align: left;">bmi</td>
<td style="text-align: left;">hf_allcause</td>
<td style="text-align: right;">874</td>
<td style="text-align: right;">0.5456300</td>
<td style="text-align: right;">0.0523114</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">0.0000000</td>
<td style="text-align: right;">NA</td>
<td style="text-align: right;">NA</td>
<td style="text-align: right;">NA</td>
<td style="text-align: right;">NA</td>
<td style="text-align: right;">NA</td>
<td style="text-align: right;">NA</td>
<td style="text-align: right;">NA</td>
<td style="text-align: right;">NA</td>
<td style="text-align: right;">NA</td>
<td style="text-align: right;">NA</td>
<td style="text-align: right;">NA</td>
</tr>
</tbody>
</table>
</div>
</div>
</section>
<section id="plot-mr" class="level2">
<h2 class="anchored" data-anchor-id="plot-mr">Plot MR</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb6"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb6-1"><a href="#cb6-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plot_mr</span>(h, res)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="index_files/figure-html/unnamed-chunk-5-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
</section>
<section id="select-variants---mr-with-a-subset-of-bmi-variants" class="level2">
<h2 class="anchored" data-anchor-id="select-variants---mr-with-a-subset-of-bmi-variants">Select variants - MR with a subset of BMI variants</h2>
<p>In the Leyden et al.&nbsp;2022 paper 86 variant colocalise with adipose tissue eQTLs and 140 colocalise with brain tissue eQTLs. There is overlap between the two tissues of 43 variants.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb7"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb7-1"><a href="#cb7-1" aria-hidden="true" tabindex="-1"></a><span class="fu">set.seed</span>(<span class="dv">123</span>)</span>
<span id="cb7-2"><a href="#cb7-2" aria-hidden="true" tabindex="-1"></a>tissue1_varid <span class="ot">&lt;-</span> <span class="fu">sample</span>(h<span class="sc">@</span>snps, <span class="dv">140</span>)</span>
<span id="cb7-3"><a href="#cb7-3" aria-hidden="true" tabindex="-1"></a>overlap_varid <span class="ot">&lt;-</span> <span class="fu">sample</span>(tissue1_varid, <span class="dv">43</span>)</span>
<span id="cb7-4"><a href="#cb7-4" aria-hidden="true" tabindex="-1"></a>tissue2_varid <span class="ot">&lt;-</span> <span class="fu">c</span>(overlap_varid, <span class="fu">sample</span>(h<span class="sc">@</span>snps[<span class="sc">!</span>h<span class="sc">@</span>snps <span class="sc">%in%</span> overlap_varid], <span class="dv">86-43</span>))</span>
<span id="cb7-5"><a href="#cb7-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-6"><a href="#cb7-6" aria-hidden="true" tabindex="-1"></a><span class="co"># turn of variants not in either tissue</span></span>
<span id="cb7-7"><a href="#cb7-7" aria-hidden="true" tabindex="-1"></a>h<span class="sc">@</span>index_snp[<span class="sc">!</span>h<span class="sc">@</span>snps <span class="sc">%in%</span> <span class="fu">c</span>(tissue1_varid, tissue2_varid)] <span class="ot">&lt;-</span> <span class="cn">FALSE</span></span>
<span id="cb7-8"><a href="#cb7-8" aria-hidden="true" tabindex="-1"></a>h<span class="sc">@</span>exposure <span class="ot">&lt;-</span> <span class="st">"BMI tissue 1/2 SNPs"</span></span>
<span id="cb7-9"><a href="#cb7-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-10"><a href="#cb7-10" aria-hidden="true" tabindex="-1"></a>res1 <span class="ot">&lt;-</span> <span class="fu">run_mr</span>(h)</span>
<span id="cb7-11"><a href="#cb7-11" aria-hidden="true" tabindex="-1"></a><span class="fu">kable</span>(res1, <span class="at">caption =</span> <span class="st">"BMI:HF Mendelian randomisation (Tissue 1 &amp; 2 SNPs only)"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<table class="caption-top table table-sm table-striped small">
<caption>BMI:HF Mendelian randomisation (Tissue 1 &amp; 2 SNPs only)</caption>
<colgroup>
<col style="width: 8%">
<col style="width: 5%">
<col style="width: 9%">
<col style="width: 5%">
<col style="width: 2%">
<col style="width: 4%">
<col style="width: 4%">
<col style="width: 2%">
<col style="width: 4%">
<col style="width: 4%">
<col style="width: 4%">
<col style="width: 4%">
<col style="width: 3%">
<col style="width: 4%">
<col style="width: 4%">
<col style="width: 6%">
<col style="width: 2%">
<col style="width: 4%">
<col style="width: 6%">
<col style="width: 7%">
</colgroup>
<thead>
<tr class="header">
<th style="text-align: left;">method</th>
<th style="text-align: left;">correlation</th>
<th style="text-align: left;">exposure</th>
<th style="text-align: left;">outcome</th>
<th style="text-align: right;">n_snp</th>
<th style="text-align: right;">b</th>
<th style="text-align: right;">b_se</th>
<th style="text-align: right;">p</th>
<th style="text-align: right;">intercept</th>
<th style="text-align: right;">int_se</th>
<th style="text-align: right;">int_p</th>
<th style="text-align: right;">qstat</th>
<th style="text-align: right;">qstat_p</th>
<th style="text-align: right;">fstat</th>
<th style="text-align: right;">condfstat</th>
<th style="text-align: right;">overdispersion</th>
<th style="text-align: right;">n_pc</th>
<th style="text-align: right;">n_hunted</th>
<th style="text-align: right;">slopehunter_pi</th>
<th style="text-align: right;">slopehunter_ent</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">mr_ivw</td>
<td style="text-align: left;">FALSE</td>
<td style="text-align: left;">BMI tissue 1/2 SNPs</td>
<td style="text-align: left;">hf_allcause</td>
<td style="text-align: right;">177</td>
<td style="text-align: right;">0.5108471</td>
<td style="text-align: right;">0.0363407</td>
<td style="text-align: right;">0e+00</td>
<td style="text-align: right;">0.0000000</td>
<td style="text-align: right;">NA</td>
<td style="text-align: right;">NA</td>
<td style="text-align: right;">304.2402</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">61.28134</td>
<td style="text-align: right;">NA</td>
<td style="text-align: right;">NA</td>
<td style="text-align: right;">NA</td>
<td style="text-align: right;">NA</td>
<td style="text-align: right;">NA</td>
<td style="text-align: right;">NA</td>
</tr>
<tr class="even">
<td style="text-align: left;">mr_egger</td>
<td style="text-align: left;">FALSE</td>
<td style="text-align: left;">BMI tissue 1/2 SNPs</td>
<td style="text-align: left;">hf_allcause</td>
<td style="text-align: right;">177</td>
<td style="text-align: right;">0.5475313</td>
<td style="text-align: right;">0.1037260</td>
<td style="text-align: right;">1e-07</td>
<td style="text-align: right;">-0.0006034</td>
<td style="text-align: right;">0.0015974</td>
<td style="text-align: right;">0.7056336</td>
<td style="text-align: right;">303.9924</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">NA</td>
<td style="text-align: right;">NA</td>
<td style="text-align: right;">NA</td>
<td style="text-align: right;">NA</td>
<td style="text-align: right;">NA</td>
<td style="text-align: right;">NA</td>
<td style="text-align: right;">NA</td>
</tr>
<tr class="odd">
<td style="text-align: left;">mr_weighted_median</td>
<td style="text-align: left;">FALSE</td>
<td style="text-align: left;">BMI tissue 1/2 SNPs</td>
<td style="text-align: left;">hf_allcause</td>
<td style="text-align: right;">177</td>
<td style="text-align: right;">0.5730810</td>
<td style="text-align: right;">0.0494983</td>
<td style="text-align: right;">0e+00</td>
<td style="text-align: right;">0.0000000</td>
<td style="text-align: right;">NA</td>
<td style="text-align: right;">NA</td>
<td style="text-align: right;">NA</td>
<td style="text-align: right;">NA</td>
<td style="text-align: right;">NA</td>
<td style="text-align: right;">NA</td>
<td style="text-align: right;">NA</td>
<td style="text-align: right;">NA</td>
<td style="text-align: right;">NA</td>
<td style="text-align: right;">NA</td>
<td style="text-align: right;">NA</td>
</tr>
<tr class="even">
<td style="text-align: left;">mr_weighted_mode</td>
<td style="text-align: left;">FALSE</td>
<td style="text-align: left;">BMI tissue 1/2 SNPs</td>
<td style="text-align: left;">hf_allcause</td>
<td style="text-align: right;">177</td>
<td style="text-align: right;">0.5887868</td>
<td style="text-align: right;">0.1022882</td>
<td style="text-align: right;">0e+00</td>
<td style="text-align: right;">0.0000000</td>
<td style="text-align: right;">NA</td>
<td style="text-align: right;">NA</td>
<td style="text-align: right;">NA</td>
<td style="text-align: right;">NA</td>
<td style="text-align: right;">NA</td>
<td style="text-align: right;">NA</td>
<td style="text-align: right;">NA</td>
<td style="text-align: right;">NA</td>
<td style="text-align: right;">NA</td>
<td style="text-align: right;">NA</td>
<td style="text-align: right;">NA</td>
</tr>
</tbody>
</table>
</div>
<div class="sourceCode cell-code" id="cb8"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb8-1"><a href="#cb8-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plot_mr</span>(h, res1)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="index_files/figure-html/unnamed-chunk-6-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
</section>
<section id="run-multivariable-mr-unweighted" class="level2">
<h2 class="anchored" data-anchor-id="run-multivariable-mr-unweighted">Run multivariable MR (unweighted)</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb9"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb9-1"><a href="#cb9-1" aria-hidden="true" tabindex="-1"></a><span class="co"># copy of BMI for tissue 1</span></span>
<span id="cb9-2"><a href="#cb9-2" aria-hidden="true" tabindex="-1"></a>bmi_t1 <span class="ot">&lt;-</span> bmi</span>
<span id="cb9-3"><a href="#cb9-3" aria-hidden="true" tabindex="-1"></a>bmi_t1<span class="sc">@</span>id <span class="ot">&lt;-</span> <span class="st">"bmi tissue 1"</span></span>
<span id="cb9-4"><a href="#cb9-4" aria-hidden="true" tabindex="-1"></a>bmi_t1<span class="sc">@</span>trait <span class="ot">&lt;-</span> <span class="st">"bmi tissue 1"</span></span>
<span id="cb9-5"><a href="#cb9-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb9-6"><a href="#cb9-6" aria-hidden="true" tabindex="-1"></a><span class="co"># copy of BMI for tissue 2</span></span>
<span id="cb9-7"><a href="#cb9-7" aria-hidden="true" tabindex="-1"></a>bmi_t2 <span class="ot">&lt;-</span> bmi</span>
<span id="cb9-8"><a href="#cb9-8" aria-hidden="true" tabindex="-1"></a>bmi_t2<span class="sc">@</span>id <span class="ot">&lt;-</span> <span class="st">"bmi tissue 2"</span></span>
<span id="cb9-9"><a href="#cb9-9" aria-hidden="true" tabindex="-1"></a>bmi_t2<span class="sc">@</span>trait <span class="ot">&lt;-</span> <span class="st">"bmi tissue 2"</span></span>
<span id="cb9-10"><a href="#cb9-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb9-11"><a href="#cb9-11" aria-hidden="true" tabindex="-1"></a><span class="co"># harmonise</span></span>
<span id="cb9-12"><a href="#cb9-12" aria-hidden="true" tabindex="-1"></a>h_mvmr <span class="ot">&lt;-</span> <span class="fu">MR</span>(<span class="fu">list</span>(bmi_t1, bmi_t2), hf)</span>
<span id="cb9-13"><a href="#cb9-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb9-14"><a href="#cb9-14" aria-hidden="true" tabindex="-1"></a><span class="co"># turn of variants not in either tissue</span></span>
<span id="cb9-15"><a href="#cb9-15" aria-hidden="true" tabindex="-1"></a>h_mvmr<span class="sc">@</span>index_snp[<span class="sc">!</span>h_mvmr<span class="sc">@</span>snps <span class="sc">%in%</span> <span class="fu">c</span>(tissue1_varid, tissue2_varid)] <span class="ot">&lt;-</span> <span class="cn">FALSE</span></span>
<span id="cb9-16"><a href="#cb9-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb9-17"><a href="#cb9-17" aria-hidden="true" tabindex="-1"></a>res2 <span class="ot">&lt;-</span> <span class="fu">run_mr</span>(h_mvmr, <span class="at">methods=</span><span class="st">"mr_ivw"</span>)</span>
<span id="cb9-18"><a href="#cb9-18" aria-hidden="true" tabindex="-1"></a><span class="fu">kable</span>(res2, <span class="at">caption =</span> <span class="st">"BMI:HF MV Mendelian randomisation"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<table class="caption-top table table-sm table-striped small">
<caption>BMI:HF MV Mendelian randomisation</caption>
<colgroup>
<col style="width: 4%">
<col style="width: 6%">
<col style="width: 6%">
<col style="width: 6%">
<col style="width: 3%">
<col style="width: 5%">
<col style="width: 5%">
<col style="width: 1%">
<col style="width: 5%">
<col style="width: 3%">
<col style="width: 3%">
<col style="width: 4%">
<col style="width: 4%">
<col style="width: 3%">
<col style="width: 5%">
<col style="width: 7%">
<col style="width: 2%">
<col style="width: 4%">
<col style="width: 7%">
<col style="width: 8%">
</colgroup>
<thead>
<tr class="header">
<th style="text-align: left;">method</th>
<th style="text-align: left;">correlation</th>
<th style="text-align: left;">exposure</th>
<th style="text-align: left;">outcome</th>
<th style="text-align: right;">n_snp</th>
<th style="text-align: right;">b</th>
<th style="text-align: right;">b_se</th>
<th style="text-align: right;">p</th>
<th style="text-align: right;">intercept</th>
<th style="text-align: right;">int_se</th>
<th style="text-align: right;">int_p</th>
<th style="text-align: right;">qstat</th>
<th style="text-align: right;">qstat_p</th>
<th style="text-align: right;">fstat</th>
<th style="text-align: right;">condfstat</th>
<th style="text-align: right;">overdispersion</th>
<th style="text-align: right;">n_pc</th>
<th style="text-align: right;">n_hunted</th>
<th style="text-align: right;">slopehunter_pi</th>
<th style="text-align: right;">slopehunter_ent</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">mr_mvivw</td>
<td style="text-align: left;">FALSE</td>
<td style="text-align: left;">bmi tissue 1</td>
<td style="text-align: left;">hf_allcause</td>
<td style="text-align: right;">177</td>
<td style="text-align: right;">0.5108471</td>
<td style="text-align: right;">0.0363407</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">NA</td>
<td style="text-align: right;">NA</td>
<td style="text-align: right;">302.5116</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">NA</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">NA</td>
<td style="text-align: right;">NA</td>
<td style="text-align: right;">NA</td>
<td style="text-align: right;">NA</td>
<td style="text-align: right;">NA</td>
</tr>
<tr class="even">
<td style="text-align: left;">mr_mvivw</td>
<td style="text-align: left;">FALSE</td>
<td style="text-align: left;">bmi tissue 2</td>
<td style="text-align: left;">hf_allcause</td>
<td style="text-align: right;">177</td>
<td style="text-align: right;">0.5108471</td>
<td style="text-align: right;">0.0363407</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">NA</td>
<td style="text-align: right;">NA</td>
<td style="text-align: right;">302.5116</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">NA</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">NA</td>
<td style="text-align: right;">NA</td>
<td style="text-align: right;">NA</td>
<td style="text-align: right;">NA</td>
<td style="text-align: right;">NA</td>
</tr>
</tbody>
</table>
</div>
<div class="sourceCode cell-code" id="cb10"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb10-1"><a href="#cb10-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plot_mr</span>(h_mvmr, res2)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="index_files/figure-html/unnamed-chunk-7-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
</section>
<section id="bmieqtl-pph4-distributions-by-tissue" class="level2">
<h2 class="anchored" data-anchor-id="bmieqtl-pph4-distributions-by-tissue">BMI:eQTL PPH4 distributions by tissue</h2>
<section id="number-of-colocalising-variants-per-tissue" class="level3">
<h3 class="anchored" data-anchor-id="number-of-colocalising-variants-per-tissue">Number of colocalising variants per tissue</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb11"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb11-1"><a href="#cb11-1" aria-hidden="true" tabindex="-1"></a>coloc_thresh <span class="ot">&lt;-</span> <span class="fl">0.8</span></span>
<span id="cb11-2"><a href="#cb11-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-3"><a href="#cb11-3" aria-hidden="true" tabindex="-1"></a>coloc_results <span class="ot">&lt;-</span> <span class="fu">fread</span>(<span class="fu">file.path</span>(TPMR_DIR, <span class="st">"output"</span>, <span class="st">"tables"</span>, <span class="st">"coloc_results.tsv.gz"</span>))</span>
<span id="cb11-4"><a href="#cb11-4" aria-hidden="true" tabindex="-1"></a>coloc_counts <span class="ot">&lt;-</span> coloc_results[locus_trait<span class="sc">==</span><span class="st">"bmi"</span> <span class="sc">&amp;</span> <span class="fu">grepl</span>(<span class="st">"^bmi"</span>, trait.x) <span class="sc">&amp;</span> <span class="sc">!</span><span class="fu">grepl</span>(<span class="st">"^bmi|^hf_"</span>, trait.y), .SD[<span class="fu">which.max</span>(pph4)], by<span class="ot">=</span>.(locus_index_rsid,trait.y)][, .(<span class="st">`</span><span class="at">num_coloc_pph4</span><span class="st">`</span> <span class="ot">=</span> <span class="fu">sum</span>(pph4<span class="sc">&gt;</span>coloc_thresh)), by<span class="ot">=</span><span class="fu">c</span>(<span class="at">tissue=</span><span class="st">"trait.y"</span>)][, tissue <span class="sc">:</span><span class="er">=</span> <span class="fu">factor</span>(tissue, <span class="at">levels =</span> tissue[<span class="fu">order</span>(<span class="sc">-</span><span class="st">`</span><span class="at">num_coloc_pph4</span><span class="st">`</span>)])]</span>
<span id="cb11-5"><a href="#cb11-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-6"><a href="#cb11-6" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(coloc_counts, <span class="fu">aes</span>(<span class="at">y =</span> tissue, <span class="at">x =</span> <span class="st">`</span><span class="at">num_coloc_pph4</span><span class="st">`</span>)) <span class="sc">+</span></span>
<span id="cb11-7"><a href="#cb11-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_classic</span>() <span class="sc">+</span></span>
<span id="cb11-8"><a href="#cb11-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_col</span>(<span class="at">fill=</span> <span class="st">"blue"</span>, <span class="at">alpha =</span> <span class="fl">0.3</span>) <span class="sc">+</span></span>
<span id="cb11-9"><a href="#cb11-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">subtitle =</span> <span class="fu">paste</span>(<span class="st">"coloc thresh:"</span>, coloc_thresh), </span>
<span id="cb11-10"><a href="#cb11-10" aria-hidden="true" tabindex="-1"></a>       <span class="at">x =</span> <span class="st">"Number of colocalising BMI variants"</span>) <span class="sc">+</span></span>
<span id="cb11-11"><a href="#cb11-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">axis.title.y =</span> <span class="fu">element_blank</span>())</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="index_files/figure-html/unnamed-chunk-8-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
</section>
<section id="max.-pph4-distributions-by-tissue" class="level3">
<h3 class="anchored" data-anchor-id="max.-pph4-distributions-by-tissue">Max. PPH4 distributions by tissue</h3>
<p>Take the maximum PPH4 from genes in close proximity to each variant, in each tissue.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb12"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb12-1"><a href="#cb12-1" aria-hidden="true" tabindex="-1"></a>num_tissues <span class="ot">&lt;-</span> <span class="dv">20</span></span>
<span id="cb12-2"><a href="#cb12-2" aria-hidden="true" tabindex="-1"></a>tissues_subset <span class="ot">&lt;-</span> <span class="fu">levels</span>(coloc_counts<span class="sc">$</span>tissue)[<span class="fu">as.integer</span>(<span class="fu">seq</span>(<span class="dv">1</span>, <span class="fu">length</span>(<span class="fu">levels</span>(coloc_counts<span class="sc">$</span>tissue)), <span class="at">length.out=</span>num_tissues))] </span>
<span id="cb12-3"><a href="#cb12-3" aria-hidden="true" tabindex="-1"></a>coloc_subset <span class="ot">&lt;-</span> coloc_results[locus_trait<span class="sc">==</span><span class="st">"bmi"</span> <span class="sc">&amp;</span> trait.x<span class="sc">==</span><span class="st">"bmi"</span> <span class="sc">&amp;</span> (trait.y <span class="sc">%in%</span> tissues_subset), .SD[<span class="fu">which.max</span>(pph4)], by<span class="ot">=</span>.(locus_index_rsid,trait.y)]</span>
<span id="cb12-4"><a href="#cb12-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb12-5"><a href="#cb12-5" aria-hidden="true" tabindex="-1"></a>density <span class="ot">&lt;-</span> coloc_subset[, {</span>
<span id="cb12-6"><a href="#cb12-6" aria-hidden="true" tabindex="-1"></a>  den <span class="ot">&lt;-</span> <span class="fu">density</span>(pph4, <span class="at">n=</span><span class="dv">1000</span>)</span>
<span id="cb12-7"><a href="#cb12-7" aria-hidden="true" tabindex="-1"></a>  .(<span class="at">pph4 =</span> scales<span class="sc">::</span><span class="fu">rescale</span>(den<span class="sc">$</span>x, <span class="dv">0</span><span class="sc">:</span><span class="dv">1</span>), <span class="at">prob =</span> den<span class="sc">$</span>y)</span>
<span id="cb12-8"><a href="#cb12-8" aria-hidden="true" tabindex="-1"></a>}, by <span class="ot">=</span> <span class="fu">c</span>(<span class="at">tissue=</span><span class="st">"trait.y"</span>)][, tissue <span class="sc">:</span><span class="er">=</span> <span class="fu">factor</span>(tissue, <span class="at">levels =</span> <span class="fu">levels</span>(coloc_counts<span class="sc">$</span>tissue))]</span>
<span id="cb12-9"><a href="#cb12-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb12-10"><a href="#cb12-10" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(density, <span class="fu">aes</span>(<span class="at">x=</span>pph4, <span class="at">y=</span>prob, <span class="at">color=</span>tissue)) <span class="sc">+</span></span>
<span id="cb12-11"><a href="#cb12-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_classic</span>() <span class="sc">+</span></span>
<span id="cb12-12"><a href="#cb12-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_line</span>() <span class="sc">+</span></span>
<span id="cb12-13"><a href="#cb12-13" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">y =</span> <span class="st">"Density"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="index_files/figure-html/unnamed-chunk-9-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
</section>
</section>
<section id="can-we-achieve-these-densities-randomly" class="level2">
<h2 class="anchored" data-anchor-id="can-we-achieve-these-densities-randomly">Can we achieve these densities randomly</h2>
<p>Yes, just sample the PPH4 from the actual density distributions above.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb13"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb13-1"><a href="#cb13-1" aria-hidden="true" tabindex="-1"></a>tissues <span class="ot">&lt;-</span> <span class="fu">expand.grid</span>(<span class="at">tissue =</span> <span class="fu">unique</span>(density<span class="sc">$</span>tissue), <span class="at">stringsAsFactors =</span> F) <span class="sc">|&gt;</span> <span class="fu">as.data.table</span>()</span>
<span id="cb13-2"><a href="#cb13-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-3"><a href="#cb13-3" aria-hidden="true" tabindex="-1"></a><span class="fu">set.seed</span>(<span class="dv">123</span>)</span>
<span id="cb13-4"><a href="#cb13-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-5"><a href="#cb13-5" aria-hidden="true" tabindex="-1"></a>res_sim_density <span class="ot">&lt;-</span> tissues[, {</span>
<span id="cb13-6"><a href="#cb13-6" aria-hidden="true" tabindex="-1"></a>  .(<span class="at">sim_pph4 =</span> <span class="fu">sample</span>(density[tissue<span class="sc">==</span>.BY<span class="sc">$</span>tissue, pph4], <span class="fu">length</span>(h_mvmr<span class="sc">@</span>bx[,<span class="dv">1</span>]), <span class="at">prob =</span> density[tissue<span class="sc">==</span>.BY<span class="sc">$</span>tissue, prob], <span class="at">replace =</span> <span class="cn">TRUE</span>))</span>
<span id="cb13-7"><a href="#cb13-7" aria-hidden="true" tabindex="-1"></a>}, by <span class="ot">=</span> .(tissue)]</span>
<span id="cb13-8"><a href="#cb13-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-9"><a href="#cb13-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-10"><a href="#cb13-10" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(res_sim_density, <span class="fu">aes</span>(<span class="at">x=</span>sim_pph4, <span class="at">color=</span>tissue)) <span class="sc">+</span></span>
<span id="cb13-11"><a href="#cb13-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_classic</span>() <span class="sc">+</span></span>
<span id="cb13-12"><a href="#cb13-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_density</span>() <span class="sc">+</span></span>
<span id="cb13-13"><a href="#cb13-13" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">y =</span> <span class="st">"Simulated density"</span>, </span>
<span id="cb13-14"><a href="#cb13-14" aria-hidden="true" tabindex="-1"></a>       <span class="at">x =</span> <span class="st">"Simulated PPH4"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="index_files/figure-html/unnamed-chunk-10-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
</section>
<section id="effect-different-pph4-weights" class="level2">
<h2 class="anchored" data-anchor-id="effect-different-pph4-weights">Effect different PPH4 weights</h2>
<p>Weight the BMI betas by their respective maximum PPH4 value, see how this changes the relationship between the beta for BMI and beta for HF, dependent on which tissue is providing the betas. Tissues with greater or fewer colocalising variants affect push the betas up or down more, respectively. I hold one tissue constant (reference) to more easily appreciate how the other tissue’s colocalisation characteristics affect things.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb14"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb14-1"><a href="#cb14-1" aria-hidden="true" tabindex="-1"></a>ref_tissue <span class="ot">&lt;-</span> <span class="fu">levels</span>(density<span class="sc">$</span>tissue)[<span class="dv">1</span>]</span>
<span id="cb14-2"><a href="#cb14-2" aria-hidden="true" tabindex="-1"></a>tissue_pairs <span class="ot">&lt;-</span> <span class="fu">expand.grid</span>(<span class="at">t1 =</span> ref_tissue, <span class="at">t2 =</span> <span class="fu">unique</span>(density<span class="sc">$</span>tissue), <span class="at">stringsAsFactors =</span> F) <span class="sc">|&gt;</span> <span class="fu">as.data.table</span>()</span>
<span id="cb14-3"><a href="#cb14-3" aria-hidden="true" tabindex="-1"></a>tissue_pairs[density[, .(<span class="at">mean_pph4 =</span> <span class="fu">mean</span>(pph4<span class="sc">*</span>prob)), by<span class="ot">=</span>tissue], t1_mean_pph4 <span class="sc">:</span><span class="er">=</span> i.mean_pph4, on<span class="ot">=</span><span class="fu">c</span>(<span class="st">"t1"</span><span class="ot">=</span><span class="st">"tissue"</span>)]</span>
<span id="cb14-4"><a href="#cb14-4" aria-hidden="true" tabindex="-1"></a>tissue_pairs[density[, .(<span class="at">mean_pph4 =</span> <span class="fu">mean</span>(pph4<span class="sc">*</span>prob)), by<span class="ot">=</span>tissue], t2_mean_pph4 <span class="sc">:</span><span class="er">=</span> i.mean_pph4, on<span class="ot">=</span><span class="fu">c</span>(<span class="st">"t2"</span><span class="ot">=</span><span class="st">"tissue"</span>)]</span>
<span id="cb14-5"><a href="#cb14-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb14-6"><a href="#cb14-6" aria-hidden="true" tabindex="-1"></a><span class="fu">set.seed</span>(<span class="dv">123</span>)</span>
<span id="cb14-7"><a href="#cb14-7" aria-hidden="true" tabindex="-1"></a>t1_pph4 <span class="ot">&lt;-</span> <span class="fu">sample</span>(density[tissue<span class="sc">==</span>ref_tissue, pph4], <span class="fu">length</span>(h_mvmr<span class="sc">@</span>bx[,<span class="dv">1</span>]), <span class="at">prob =</span> density[tissue<span class="sc">==</span>ref_tissue, prob], <span class="at">replace =</span> <span class="cn">TRUE</span>)</span>
<span id="cb14-8"><a href="#cb14-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb14-9"><a href="#cb14-9" aria-hidden="true" tabindex="-1"></a>res_weighted <span class="ot">&lt;-</span> tissue_pairs[, {</span>
<span id="cb14-10"><a href="#cb14-10" aria-hidden="true" tabindex="-1"></a>  h_tmp <span class="ot">&lt;-</span> h_mvmr</span>
<span id="cb14-11"><a href="#cb14-11" aria-hidden="true" tabindex="-1"></a>  t2_pph4 <span class="ot">&lt;-</span> <span class="fu">sample</span>(density[tissue<span class="sc">==</span>.BY<span class="sc">$</span>t2, pph4], <span class="fu">length</span>(h_tmp<span class="sc">@</span>bx[,<span class="dv">1</span>]), <span class="at">prob =</span> density[tissue<span class="sc">==</span>.BY<span class="sc">$</span>t2, prob], <span class="at">replace =</span> <span class="cn">TRUE</span>)</span>
<span id="cb14-12"><a href="#cb14-12" aria-hidden="true" tabindex="-1"></a>  h_tmp<span class="sc">@</span>bx[,<span class="dv">1</span>] <span class="ot">&lt;-</span> h_tmp<span class="sc">@</span>bx[,<span class="dv">1</span>] <span class="sc">*</span> t1_pph4</span>
<span id="cb14-13"><a href="#cb14-13" aria-hidden="true" tabindex="-1"></a>  h_tmp<span class="sc">@</span>bx[,<span class="dv">2</span>] <span class="ot">&lt;-</span> h_tmp<span class="sc">@</span>bx[,<span class="dv">2</span>] <span class="sc">*</span> t2_pph4</span>
<span id="cb14-14"><a href="#cb14-14" aria-hidden="true" tabindex="-1"></a>  h_tmp<span class="sc">@</span>index_snp <span class="ot">&lt;-</span> <span class="fu">ifelse</span>(t1_pph4<span class="sc">&gt;</span>coloc_thresh <span class="sc">|</span> t2_pph4<span class="sc">&gt;</span>coloc_thresh, <span class="cn">TRUE</span>, <span class="cn">FALSE</span>)</span>
<span id="cb14-15"><a href="#cb14-15" aria-hidden="true" tabindex="-1"></a>  .(<span class="at">tissue   =</span> <span class="fu">rep</span>(<span class="fu">c</span>(<span class="fu">paste0</span>(<span class="st">"Ref: "</span>, ref_tissue),<span class="st">"Comparator"</span>), <span class="at">each=</span><span class="fu">sum</span>(h_tmp<span class="sc">@</span>index_snp)), </span>
<span id="cb14-16"><a href="#cb14-16" aria-hidden="true" tabindex="-1"></a>    <span class="at">beta_bmi =</span> <span class="fu">c</span>(h_tmp<span class="sc">@</span>bx[h_tmp<span class="sc">@</span>index_snp,<span class="dv">1</span>], h_tmp<span class="sc">@</span>bx[h_tmp<span class="sc">@</span>index_snp,<span class="dv">2</span>]), </span>
<span id="cb14-17"><a href="#cb14-17" aria-hidden="true" tabindex="-1"></a>    <span class="at">beta_hf  =</span> <span class="fu">rep</span>(h_tmp<span class="sc">@</span>by[h_tmp<span class="sc">@</span>index_snp],<span class="dv">2</span>),</span>
<span id="cb14-18"><a href="#cb14-18" aria-hidden="true" tabindex="-1"></a>    <span class="at">mean_pph4_t1 =</span> <span class="fu">mean</span>(t1_pph4[t1_pph4<span class="sc">&gt;</span>coloc_thresh <span class="sc">|</span> t2_pph4<span class="sc">&gt;</span>coloc_thresh]),</span>
<span id="cb14-19"><a href="#cb14-19" aria-hidden="true" tabindex="-1"></a>    <span class="at">mean_pph4_t2 =</span> <span class="fu">mean</span>(t2_pph4[t1_pph4<span class="sc">&gt;</span>coloc_thresh <span class="sc">|</span> t2_pph4<span class="sc">&gt;</span>coloc_thresh]))</span>
<span id="cb14-20"><a href="#cb14-20" aria-hidden="true" tabindex="-1"></a>}, by <span class="ot">=</span> .(t1,t2)]</span>
<span id="cb14-21"><a href="#cb14-21" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb14-22"><a href="#cb14-22" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb14-23"><a href="#cb14-23" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(res_weighted, <span class="fu">aes</span>(<span class="at">x=</span>beta_bmi, <span class="at">y=</span>beta_hf, <span class="at">color=</span>tissue)) <span class="sc">+</span></span>
<span id="cb14-24"><a href="#cb14-24" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>() <span class="sc">+</span></span>
<span id="cb14-25"><a href="#cb14-25" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_text</span>(<span class="at">data =</span> <span class="fu">unique</span>(res_weighted[<span class="fu">grepl</span>(<span class="st">"Ref"</span>, tissue)], <span class="at">by=</span><span class="st">"t2"</span>), </span>
<span id="cb14-26"><a href="#cb14-26" aria-hidden="true" tabindex="-1"></a>            <span class="fu">aes</span>(<span class="at">x =</span> <span class="fl">0.0</span>, </span>
<span id="cb14-27"><a href="#cb14-27" aria-hidden="true" tabindex="-1"></a>                <span class="at">y =</span> <span class="sc">-</span><span class="fl">0.04</span>,</span>
<span id="cb14-28"><a href="#cb14-28" aria-hidden="true" tabindex="-1"></a>                <span class="at">label =</span> <span class="fu">sprintf</span>(<span class="st">"Tis mPPH4=%.3f"</span>, mean_pph4_t2)),</span>
<span id="cb14-29"><a href="#cb14-29" aria-hidden="true" tabindex="-1"></a>            <span class="at">color=</span><span class="st">"black"</span>, <span class="at">size=</span><span class="dv">2</span>) <span class="sc">+</span></span>
<span id="cb14-30"><a href="#cb14-30" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_text</span>(<span class="at">data =</span> <span class="fu">unique</span>(res_weighted[<span class="sc">!</span><span class="fu">grepl</span>(<span class="st">"Ref"</span>, tissue)], <span class="at">by=</span><span class="st">"t2"</span>), </span>
<span id="cb14-31"><a href="#cb14-31" aria-hidden="true" tabindex="-1"></a>            <span class="fu">aes</span>(<span class="at">x =</span> <span class="fl">0.0</span>, </span>
<span id="cb14-32"><a href="#cb14-32" aria-hidden="true" tabindex="-1"></a>                <span class="at">y =</span> <span class="sc">-</span><span class="fl">0.07</span>,</span>
<span id="cb14-33"><a href="#cb14-33" aria-hidden="true" tabindex="-1"></a>                <span class="at">label =</span> <span class="fu">sprintf</span>(<span class="st">"Ref mPPH4=%.3f"</span>, mean_pph4_t1)),</span>
<span id="cb14-34"><a href="#cb14-34" aria-hidden="true" tabindex="-1"></a>            <span class="at">color=</span><span class="st">"black"</span>, <span class="at">size=</span><span class="dv">2</span>) <span class="sc">+</span></span>
<span id="cb14-35"><a href="#cb14-35" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_smooth</span>(<span class="at">method=</span><span class="st">"lm"</span>, <span class="at">se=</span>F,<span class="at">fullrange =</span> <span class="cn">TRUE</span>) <span class="sc">+</span></span>
<span id="cb14-36"><a href="#cb14-36" aria-hidden="true" tabindex="-1"></a>  <span class="fu">facet_wrap</span>(<span class="sc">~</span>t2, <span class="at">labeller =</span> <span class="fu">labeller</span>(<span class="at">t2 =</span> <span class="cf">function</span>(x) <span class="fu">gsub</span>(<span class="st">"_"</span>, <span class="st">" "</span>, x))) <span class="sc">+</span></span>
<span id="cb14-37"><a href="#cb14-37" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">y =</span> <span class="st">"Heart failure beta"</span>, </span>
<span id="cb14-38"><a href="#cb14-38" aria-hidden="true" tabindex="-1"></a>       <span class="at">x =</span> <span class="st">"BMI beta (PPH4-weighted)"</span>, </span>
<span id="cb14-39"><a href="#cb14-39" aria-hidden="true" tabindex="-1"></a>       <span class="at">color =</span> <span class="st">"Tissue"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="index_files/figure-html/unnamed-chunk-11-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
</section>
<section id="ivw-regression-by-different-pph4-weighting-distributions" class="level2">
<h2 class="anchored" data-anchor-id="ivw-regression-by-different-pph4-weighting-distributions">IVW regression by different PPH4 weighting distributions</h2>
<p>Perform MVMR with the weighted BMI instruments. Lets look at holding a poorly colocalising tissue constant compared to the best colocalising tissues.</p>
<section id="reference-top-colocalising-tissue" class="level3">
<h3 class="anchored" data-anchor-id="reference-top-colocalising-tissue">Reference: top colocalising tissue</h3>
<p>Different distributions against the top colocalising tissue.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb15"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb15-1"><a href="#cb15-1" aria-hidden="true" tabindex="-1"></a>ref_tissue <span class="ot">&lt;-</span> <span class="fu">levels</span>(density<span class="sc">$</span>tissue)[<span class="dv">1</span>] <span class="co"># top colocalising tissue</span></span>
<span id="cb15-2"><a href="#cb15-2" aria-hidden="true" tabindex="-1"></a>tissue_pairs <span class="ot">&lt;-</span> <span class="fu">expand.grid</span>(<span class="at">t1 =</span> ref_tissue, <span class="at">t2 =</span> <span class="fu">unique</span>(density<span class="sc">$</span>tissue), <span class="at">stringsAsFactors =</span> F) <span class="sc">|&gt;</span> <span class="fu">as.data.table</span>()</span>
<span id="cb15-3"><a href="#cb15-3" aria-hidden="true" tabindex="-1"></a>tissue_pairs[density[, .(<span class="at">mean_pph4 =</span> <span class="fu">mean</span>(pph4<span class="sc">*</span>prob)), by<span class="ot">=</span>tissue], t1_mean_pph4 <span class="sc">:</span><span class="er">=</span> i.mean_pph4, on<span class="ot">=</span><span class="fu">c</span>(<span class="st">"t1"</span><span class="ot">=</span><span class="st">"tissue"</span>)]</span>
<span id="cb15-4"><a href="#cb15-4" aria-hidden="true" tabindex="-1"></a>tissue_pairs[density[, .(<span class="at">mean_pph4 =</span> <span class="fu">mean</span>(pph4<span class="sc">*</span>prob)), by<span class="ot">=</span>tissue], t2_mean_pph4 <span class="sc">:</span><span class="er">=</span> i.mean_pph4, on<span class="ot">=</span><span class="fu">c</span>(<span class="st">"t2"</span><span class="ot">=</span><span class="st">"tissue"</span>)]</span>
<span id="cb15-5"><a href="#cb15-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-6"><a href="#cb15-6" aria-hidden="true" tabindex="-1"></a><span class="fu">set.seed</span>(<span class="dv">123</span>)</span>
<span id="cb15-7"><a href="#cb15-7" aria-hidden="true" tabindex="-1"></a>t1_pph4 <span class="ot">&lt;-</span> <span class="fu">sample</span>(density[tissue<span class="sc">==</span>ref_tissue, pph4], <span class="fu">length</span>(h_mvmr<span class="sc">@</span>bx[,<span class="dv">1</span>]), <span class="at">prob =</span> density[tissue<span class="sc">==</span>ref_tissue, prob], <span class="at">replace =</span> <span class="cn">TRUE</span>)</span>
<span id="cb15-8"><a href="#cb15-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-9"><a href="#cb15-9" aria-hidden="true" tabindex="-1"></a>mr_weighted <span class="ot">&lt;-</span> tissue_pairs[, {</span>
<span id="cb15-10"><a href="#cb15-10" aria-hidden="true" tabindex="-1"></a>  h_tmp <span class="ot">&lt;-</span> h_mvmr</span>
<span id="cb15-11"><a href="#cb15-11" aria-hidden="true" tabindex="-1"></a>  t2_pph4 <span class="ot">&lt;-</span> <span class="fu">sample</span>(density[tissue<span class="sc">==</span>.BY<span class="sc">$</span>t2, pph4], <span class="fu">length</span>(h_tmp<span class="sc">@</span>bx[,<span class="dv">1</span>]), <span class="at">prob =</span> density[tissue<span class="sc">==</span>.BY<span class="sc">$</span>t2, prob], <span class="at">replace =</span> <span class="cn">TRUE</span>)</span>
<span id="cb15-12"><a href="#cb15-12" aria-hidden="true" tabindex="-1"></a>  h_tmp<span class="sc">@</span>bx[,<span class="dv">1</span>] <span class="ot">&lt;-</span> h_tmp<span class="sc">@</span>bx[,<span class="dv">1</span>] <span class="sc">*</span> t1_pph4</span>
<span id="cb15-13"><a href="#cb15-13" aria-hidden="true" tabindex="-1"></a>  h_tmp<span class="sc">@</span>bx[,<span class="dv">2</span>] <span class="ot">&lt;-</span> h_tmp<span class="sc">@</span>bx[,<span class="dv">2</span>] <span class="sc">*</span> t2_pph4</span>
<span id="cb15-14"><a href="#cb15-14" aria-hidden="true" tabindex="-1"></a>  h_tmp<span class="sc">@</span>index_snp <span class="ot">&lt;-</span> <span class="fu">ifelse</span>(t1_pph4<span class="sc">&gt;</span>coloc_thresh <span class="sc">|</span> t2_pph4<span class="sc">&gt;</span>coloc_thresh, <span class="cn">TRUE</span>, <span class="cn">FALSE</span>)</span>
<span id="cb15-15"><a href="#cb15-15" aria-hidden="true" tabindex="-1"></a>  h_tmp<span class="sc">@</span>exposure[<span class="dv">1</span>] <span class="ot">&lt;-</span> <span class="fu">paste0</span>(<span class="st">"Ref: "</span>, ref_tissue)</span>
<span id="cb15-16"><a href="#cb15-16" aria-hidden="true" tabindex="-1"></a>  h_tmp<span class="sc">@</span>exposure[<span class="dv">2</span>] <span class="ot">&lt;-</span> <span class="st">"Comparator"</span></span>
<span id="cb15-17"><a href="#cb15-17" aria-hidden="true" tabindex="-1"></a>  r <span class="ot">&lt;-</span> <span class="fu">run_mr</span>(h_tmp, <span class="at">methods=</span><span class="st">"mr_ivw"</span>)</span>
<span id="cb15-18"><a href="#cb15-18" aria-hidden="true" tabindex="-1"></a>  r[, <span class="st">`</span><span class="at">:=</span><span class="st">`</span>(<span class="at">n_sig_snps =</span> <span class="fu">c</span>(<span class="fu">sum</span>(t1_pph4<span class="sc">&gt;</span>coloc_thresh), <span class="fu">sum</span>(t2_pph4<span class="sc">&gt;</span>coloc_thresh)), </span>
<span id="cb15-19"><a href="#cb15-19" aria-hidden="true" tabindex="-1"></a>           <span class="at">n_overlap_snps =</span> <span class="fu">sum</span>(t1_pph4<span class="sc">&gt;</span>coloc_thresh <span class="sc">&amp;</span> t2_pph4<span class="sc">&gt;</span>coloc_thresh))]</span>
<span id="cb15-20"><a href="#cb15-20" aria-hidden="true" tabindex="-1"></a>}, by <span class="ot">=</span> .(t1,t2,t1_mean_pph4,t2_mean_pph4)]</span>
<span id="cb15-21"><a href="#cb15-21" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-22"><a href="#cb15-22" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(mr_weighted, <span class="fu">aes</span>(<span class="at">x=</span>t2, <span class="at">y=</span><span class="fu">exp</span>(b), <span class="at">ymin=</span><span class="fu">exp</span>(b<span class="sc">-</span>b_se<span class="sc">*</span><span class="fl">1.96</span>), <span class="at">ymax=</span><span class="fu">exp</span>(b<span class="sc">+</span>b_se<span class="sc">*</span><span class="fl">1.96</span>), <span class="at">color=</span>exposure)) <span class="sc">+</span></span>
<span id="cb15-23"><a href="#cb15-23" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_errorbar</span>(<span class="at">width=</span><span class="fl">0.2</span>, <span class="at">position =</span> <span class="fu">position_dodge</span>(<span class="fl">0.2</span>)) <span class="sc">+</span></span>
<span id="cb15-24"><a href="#cb15-24" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="fu">aes</span>(<span class="at">size=</span>n_sig_snps), <span class="at">position =</span> <span class="fu">position_dodge</span>(<span class="fl">0.2</span>)) <span class="sc">+</span></span>
<span id="cb15-25"><a href="#cb15-25" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">axis.text.x =</span> <span class="fu">element_text</span>(<span class="at">angle=</span><span class="dv">45</span>, <span class="at">hjust=</span><span class="dv">1</span>), </span>
<span id="cb15-26"><a href="#cb15-26" aria-hidden="true" tabindex="-1"></a>        <span class="at">axis.title.y =</span> <span class="fu">element_blank</span>()) <span class="sc">+</span></span>
<span id="cb15-27"><a href="#cb15-27" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">y =</span> <span class="st">"MR estimate (tpBMI vs HF)"</span>, <span class="at">color =</span> <span class="st">"Tissue"</span>, <span class="at">size =</span> <span class="st">"Num. SNPs contributed to model"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="index_files/figure-html/unnamed-chunk-12-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
</section>
<section id="reference-worst-colocalising-tissue" class="level3">
<h3 class="anchored" data-anchor-id="reference-worst-colocalising-tissue">Reference: worst colocalising tissue</h3>
<p>Then, different distributions against the worst colocalising tissue.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb16"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb16-1"><a href="#cb16-1" aria-hidden="true" tabindex="-1"></a>ref_tissue <span class="ot">&lt;-</span> <span class="fu">levels</span>(density<span class="sc">$</span>tissue)[<span class="fu">length</span>(<span class="fu">levels</span>(density<span class="sc">$</span>tissue))] <span class="co"># bottom/worst colocalising tissue</span></span>
<span id="cb16-2"><a href="#cb16-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-3"><a href="#cb16-3" aria-hidden="true" tabindex="-1"></a>tissue_pairs <span class="ot">&lt;-</span> <span class="fu">expand.grid</span>(<span class="at">t1 =</span> ref_tissue, <span class="at">t2 =</span> <span class="fu">unique</span>(density<span class="sc">$</span>tissue), <span class="at">stringsAsFactors =</span> F) <span class="sc">|&gt;</span> <span class="fu">as.data.table</span>()</span>
<span id="cb16-4"><a href="#cb16-4" aria-hidden="true" tabindex="-1"></a>tissue_pairs[density[, .(<span class="at">mean_pph4 =</span> <span class="fu">mean</span>(pph4<span class="sc">*</span>prob)), by<span class="ot">=</span>tissue], t1_mean_pph4 <span class="sc">:</span><span class="er">=</span> i.mean_pph4, on<span class="ot">=</span><span class="fu">c</span>(<span class="st">"t1"</span><span class="ot">=</span><span class="st">"tissue"</span>)]</span>
<span id="cb16-5"><a href="#cb16-5" aria-hidden="true" tabindex="-1"></a>tissue_pairs[density[, .(<span class="at">mean_pph4 =</span> <span class="fu">mean</span>(pph4<span class="sc">*</span>prob)), by<span class="ot">=</span>tissue], t2_mean_pph4 <span class="sc">:</span><span class="er">=</span> i.mean_pph4, on<span class="ot">=</span><span class="fu">c</span>(<span class="st">"t2"</span><span class="ot">=</span><span class="st">"tissue"</span>)]</span>
<span id="cb16-6"><a href="#cb16-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-7"><a href="#cb16-7" aria-hidden="true" tabindex="-1"></a><span class="fu">set.seed</span>(<span class="dv">123</span>)</span>
<span id="cb16-8"><a href="#cb16-8" aria-hidden="true" tabindex="-1"></a>t1_pph4 <span class="ot">&lt;-</span> <span class="fu">sample</span>(density[tissue<span class="sc">==</span>ref_tissue, pph4], <span class="fu">length</span>(h_mvmr<span class="sc">@</span>bx[,<span class="dv">1</span>]), <span class="at">prob =</span> density[tissue<span class="sc">==</span>ref_tissue, prob], <span class="at">replace =</span> <span class="cn">TRUE</span>)</span>
<span id="cb16-9"><a href="#cb16-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-10"><a href="#cb16-10" aria-hidden="true" tabindex="-1"></a>mr_weighted <span class="ot">&lt;-</span> tissue_pairs[, {</span>
<span id="cb16-11"><a href="#cb16-11" aria-hidden="true" tabindex="-1"></a>  h_tmp <span class="ot">&lt;-</span> h_mvmr</span>
<span id="cb16-12"><a href="#cb16-12" aria-hidden="true" tabindex="-1"></a>  t2_pph4 <span class="ot">&lt;-</span> <span class="fu">sample</span>(density[tissue<span class="sc">==</span>.BY<span class="sc">$</span>t2, pph4], <span class="fu">length</span>(h_tmp<span class="sc">@</span>bx[,<span class="dv">1</span>]), <span class="at">prob =</span> density[tissue<span class="sc">==</span>.BY<span class="sc">$</span>t2, prob], <span class="at">replace =</span> <span class="cn">TRUE</span>)</span>
<span id="cb16-13"><a href="#cb16-13" aria-hidden="true" tabindex="-1"></a>  h_tmp<span class="sc">@</span>bx[,<span class="dv">1</span>] <span class="ot">&lt;-</span> h_tmp<span class="sc">@</span>bx[,<span class="dv">1</span>] <span class="sc">*</span> t1_pph4</span>
<span id="cb16-14"><a href="#cb16-14" aria-hidden="true" tabindex="-1"></a>  h_tmp<span class="sc">@</span>bx[,<span class="dv">2</span>] <span class="ot">&lt;-</span> h_tmp<span class="sc">@</span>bx[,<span class="dv">2</span>] <span class="sc">*</span> t2_pph4</span>
<span id="cb16-15"><a href="#cb16-15" aria-hidden="true" tabindex="-1"></a>  h_tmp<span class="sc">@</span>index_snp <span class="ot">&lt;-</span> <span class="fu">ifelse</span>(t1_pph4<span class="sc">&gt;</span>coloc_thresh <span class="sc">|</span> t2_pph4<span class="sc">&gt;</span>coloc_thresh, <span class="cn">TRUE</span>, <span class="cn">FALSE</span>)</span>
<span id="cb16-16"><a href="#cb16-16" aria-hidden="true" tabindex="-1"></a>  h_tmp<span class="sc">@</span>exposure[<span class="dv">1</span>] <span class="ot">&lt;-</span> <span class="fu">paste0</span>(<span class="st">"Ref: "</span>, ref_tissue)</span>
<span id="cb16-17"><a href="#cb16-17" aria-hidden="true" tabindex="-1"></a>  h_tmp<span class="sc">@</span>exposure[<span class="dv">2</span>] <span class="ot">&lt;-</span> <span class="st">"Comparator"</span></span>
<span id="cb16-18"><a href="#cb16-18" aria-hidden="true" tabindex="-1"></a>  r <span class="ot">&lt;-</span> <span class="fu">run_mr</span>(h_tmp, <span class="at">methods=</span><span class="st">"mr_ivw"</span>)</span>
<span id="cb16-19"><a href="#cb16-19" aria-hidden="true" tabindex="-1"></a>  r[, <span class="st">`</span><span class="at">:=</span><span class="st">`</span>(<span class="at">n_sig_snps =</span> <span class="fu">c</span>(<span class="fu">sum</span>(t1_pph4<span class="sc">&gt;</span>coloc_thresh), <span class="fu">sum</span>(t2_pph4<span class="sc">&gt;</span>coloc_thresh)), </span>
<span id="cb16-20"><a href="#cb16-20" aria-hidden="true" tabindex="-1"></a>           <span class="at">n_overlap_snps =</span> <span class="fu">sum</span>(t1_pph4<span class="sc">&gt;</span>coloc_thresh <span class="sc">&amp;</span> t2_pph4<span class="sc">&gt;</span>coloc_thresh))]</span>
<span id="cb16-21"><a href="#cb16-21" aria-hidden="true" tabindex="-1"></a>}, by <span class="ot">=</span> .(t1,t2,t1_mean_pph4,t2_mean_pph4)]</span>
<span id="cb16-22"><a href="#cb16-22" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-23"><a href="#cb16-23" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(mr_weighted, <span class="fu">aes</span>(<span class="at">x=</span>t2, <span class="at">y=</span><span class="fu">exp</span>(b), <span class="at">ymin=</span><span class="fu">exp</span>(b<span class="sc">-</span>b_se<span class="sc">*</span><span class="fl">1.96</span>), <span class="at">ymax=</span><span class="fu">exp</span>(b<span class="sc">+</span>b_se<span class="sc">*</span><span class="fl">1.96</span>), <span class="at">color=</span>exposure)) <span class="sc">+</span></span>
<span id="cb16-24"><a href="#cb16-24" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_errorbar</span>(<span class="at">width=</span><span class="fl">0.2</span>, <span class="at">position =</span> <span class="fu">position_dodge</span>(<span class="fl">0.2</span>)) <span class="sc">+</span></span>
<span id="cb16-25"><a href="#cb16-25" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="fu">aes</span>(<span class="at">size=</span>n_sig_snps), <span class="at">position =</span> <span class="fu">position_dodge</span>(<span class="fl">0.2</span>)) <span class="sc">+</span></span>
<span id="cb16-26"><a href="#cb16-26" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">axis.text.x =</span> <span class="fu">element_text</span>(<span class="at">angle=</span><span class="dv">45</span>, <span class="at">hjust=</span><span class="dv">1</span>), </span>
<span id="cb16-27"><a href="#cb16-27" aria-hidden="true" tabindex="-1"></a>        <span class="at">axis.title.y =</span> <span class="fu">element_blank</span>()) <span class="sc">+</span></span>
<span id="cb16-28"><a href="#cb16-28" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">y =</span> <span class="st">"MR estimate (tpBMI vs HF)"</span>, <span class="at">color =</span> <span class="st">"Tissue"</span>, <span class="at">size =</span> <span class="st">"Num. SNPs contributed to model"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="index_files/figure-html/unnamed-chunk-13-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
</section>
</section>
<section id="number-of-colocalising-loci" class="level2">
<h2 class="anchored" data-anchor-id="number-of-colocalising-loci">Number of colocalising loci</h2>
<p>Is it just mostly to do with the difference in the number of colocalising variants in each tissue. Run MR for all pairs of tissues.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb17"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb17-1"><a href="#cb17-1" aria-hidden="true" tabindex="-1"></a>tissue_pairs <span class="ot">&lt;-</span> <span class="fu">expand.grid</span>(<span class="at">t1 =</span> <span class="fu">unique</span>(density<span class="sc">$</span>tissue), <span class="at">t2 =</span> <span class="fu">unique</span>(density<span class="sc">$</span>tissue), <span class="at">stringsAsFactors =</span> F) <span class="sc">|&gt;</span> <span class="fu">as.data.table</span>()</span>
<span id="cb17-2"><a href="#cb17-2" aria-hidden="true" tabindex="-1"></a>tissue_pairs[density[, .(<span class="at">mean_pph4 =</span> <span class="fu">mean</span>(pph4<span class="sc">*</span>prob)), by<span class="ot">=</span>tissue], t1_mean_pph4 <span class="sc">:</span><span class="er">=</span> i.mean_pph4, on<span class="ot">=</span><span class="fu">c</span>(<span class="st">"t1"</span><span class="ot">=</span><span class="st">"tissue"</span>)]</span>
<span id="cb17-3"><a href="#cb17-3" aria-hidden="true" tabindex="-1"></a>tissue_pairs[density[, .(<span class="at">mean_pph4 =</span> <span class="fu">mean</span>(pph4<span class="sc">*</span>prob)), by<span class="ot">=</span>tissue], t2_mean_pph4 <span class="sc">:</span><span class="er">=</span> i.mean_pph4, on<span class="ot">=</span><span class="fu">c</span>(<span class="st">"t2"</span><span class="ot">=</span><span class="st">"tissue"</span>)]</span>
<span id="cb17-4"><a href="#cb17-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-5"><a href="#cb17-5" aria-hidden="true" tabindex="-1"></a><span class="fu">set.seed</span>(<span class="dv">123</span>)</span>
<span id="cb17-6"><a href="#cb17-6" aria-hidden="true" tabindex="-1"></a>mr_weighted <span class="ot">&lt;-</span> tissue_pairs[, {</span>
<span id="cb17-7"><a href="#cb17-7" aria-hidden="true" tabindex="-1"></a>  h_tmp <span class="ot">&lt;-</span> h_mvmr</span>
<span id="cb17-8"><a href="#cb17-8" aria-hidden="true" tabindex="-1"></a>  t1_pph4 <span class="ot">&lt;-</span> <span class="fu">sample</span>(density[tissue<span class="sc">==</span>.BY<span class="sc">$</span>t1, pph4], <span class="fu">length</span>(h_tmp<span class="sc">@</span>bx[,<span class="dv">1</span>]), <span class="at">prob =</span> density[tissue<span class="sc">==</span>.BY<span class="sc">$</span>t1, prob], <span class="at">replace =</span> <span class="cn">TRUE</span>)</span>
<span id="cb17-9"><a href="#cb17-9" aria-hidden="true" tabindex="-1"></a>  t2_pph4 <span class="ot">&lt;-</span> <span class="fu">sample</span>(density[tissue<span class="sc">==</span>.BY<span class="sc">$</span>t2, pph4], <span class="fu">length</span>(h_tmp<span class="sc">@</span>bx[,<span class="dv">1</span>]), <span class="at">prob =</span> density[tissue<span class="sc">==</span>.BY<span class="sc">$</span>t2, prob], <span class="at">replace =</span> <span class="cn">TRUE</span>)</span>
<span id="cb17-10"><a href="#cb17-10" aria-hidden="true" tabindex="-1"></a>  h_tmp<span class="sc">@</span>bx[,<span class="dv">1</span>] <span class="ot">&lt;-</span> h_tmp<span class="sc">@</span>bx[,<span class="dv">1</span>] <span class="sc">*</span> t1_pph4</span>
<span id="cb17-11"><a href="#cb17-11" aria-hidden="true" tabindex="-1"></a>  h_tmp<span class="sc">@</span>bx[,<span class="dv">2</span>] <span class="ot">&lt;-</span> h_tmp<span class="sc">@</span>bx[,<span class="dv">2</span>] <span class="sc">*</span> t2_pph4</span>
<span id="cb17-12"><a href="#cb17-12" aria-hidden="true" tabindex="-1"></a>  h_tmp<span class="sc">@</span>index_snp <span class="ot">&lt;-</span> <span class="fu">ifelse</span>(t1_pph4<span class="sc">&gt;</span>coloc_thresh <span class="sc">|</span> t2_pph4<span class="sc">&gt;</span>coloc_thresh, <span class="cn">TRUE</span>, <span class="cn">FALSE</span>)</span>
<span id="cb17-13"><a href="#cb17-13" aria-hidden="true" tabindex="-1"></a>  h_tmp<span class="sc">@</span>exposure[<span class="dv">1</span>] <span class="ot">&lt;-</span> <span class="st">"t1"</span></span>
<span id="cb17-14"><a href="#cb17-14" aria-hidden="true" tabindex="-1"></a>  h_tmp<span class="sc">@</span>exposure[<span class="dv">2</span>] <span class="ot">&lt;-</span> <span class="st">"t2"</span></span>
<span id="cb17-15"><a href="#cb17-15" aria-hidden="true" tabindex="-1"></a>  r <span class="ot">&lt;-</span> <span class="fu">run_mr</span>(h_tmp, <span class="at">methods=</span><span class="st">"mr_ivw"</span>)</span>
<span id="cb17-16"><a href="#cb17-16" aria-hidden="true" tabindex="-1"></a>  r[, <span class="st">`</span><span class="at">:=</span><span class="st">`</span>(<span class="at">n_sig_snps =</span> <span class="fu">c</span>(<span class="fu">sum</span>(t1_pph4<span class="sc">&gt;</span>coloc_thresh), <span class="fu">sum</span>(t2_pph4<span class="sc">&gt;</span>coloc_thresh)),</span>
<span id="cb17-17"><a href="#cb17-17" aria-hidden="true" tabindex="-1"></a>           <span class="at">n_uni_snps =</span> <span class="fu">c</span>(<span class="fu">length</span>(<span class="fu">setdiff</span>(<span class="fu">which</span>(t1_pph4<span class="sc">&gt;</span>coloc_thresh), <span class="fu">which</span>(t2_pph4<span class="sc">&gt;</span>coloc_thresh))), <span class="fu">length</span>(<span class="fu">setdiff</span>(<span class="fu">which</span>(t2_pph4<span class="sc">&gt;</span>coloc_thresh), <span class="fu">which</span>(t1_pph4<span class="sc">&gt;</span>coloc_thresh)))),</span>
<span id="cb17-18"><a href="#cb17-18" aria-hidden="true" tabindex="-1"></a>           <span class="at">n_overlap_snps =</span> <span class="fu">sum</span>(t1_pph4<span class="sc">&gt;</span>coloc_thresh <span class="sc">&amp;</span> t2_pph4<span class="sc">&gt;</span>coloc_thresh), </span>
<span id="cb17-19"><a href="#cb17-19" aria-hidden="true" tabindex="-1"></a>           <span class="at">n_sig_diff =</span> <span class="fu">sum</span>(t1_pph4<span class="sc">&gt;</span>coloc_thresh) <span class="sc">-</span> <span class="fu">sum</span>(t2_pph4<span class="sc">&gt;</span>coloc_thresh), </span>
<span id="cb17-20"><a href="#cb17-20" aria-hidden="true" tabindex="-1"></a>           <span class="at">exp_b_diff =</span> <span class="fu">diff</span>(<span class="fu">range</span>(<span class="fu">exp</span>(b))))]</span>
<span id="cb17-21"><a href="#cb17-21" aria-hidden="true" tabindex="-1"></a>}, by <span class="ot">=</span> .(t1,t2,t1_mean_pph4,t2_mean_pph4)]</span>
<span id="cb17-22"><a href="#cb17-22" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-23"><a href="#cb17-23" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(mr_weighted, <span class="fu">aes</span>(<span class="at">x=</span>n_sig_snps, <span class="at">y=</span><span class="fu">exp</span>(b), <span class="at">ymin=</span><span class="fu">exp</span>(b<span class="sc">-</span>b_se<span class="sc">*</span><span class="fl">1.96</span>), <span class="at">ymax=</span><span class="fu">exp</span>(b<span class="sc">+</span>b_se<span class="sc">*</span><span class="fl">1.96</span>), <span class="at">color=</span>exposure)) <span class="sc">+</span></span>
<span id="cb17-24"><a href="#cb17-24" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_errorbar</span>(<span class="at">width=</span><span class="fl">0.2</span>, <span class="at">position =</span> <span class="fu">position_dodge</span>(<span class="fl">0.2</span>)) <span class="sc">+</span></span>
<span id="cb17-25"><a href="#cb17-25" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="at">position =</span> <span class="fu">position_dodge</span>(<span class="fl">0.2</span>)) <span class="sc">+</span></span>
<span id="cb17-26"><a href="#cb17-26" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">axis.text.x =</span> <span class="fu">element_text</span>(<span class="at">angle=</span><span class="dv">45</span>, <span class="at">hjust=</span><span class="dv">1</span>), </span>
<span id="cb17-27"><a href="#cb17-27" aria-hidden="true" tabindex="-1"></a>        <span class="at">axis.title.y =</span> <span class="fu">element_blank</span>()) <span class="sc">+</span></span>
<span id="cb17-28"><a href="#cb17-28" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">y =</span> <span class="st">"MR estimate (tpBMI vs HF)"</span>, <span class="at">color =</span> <span class="st">"Tissue"</span>, <span class="at">size =</span> <span class="st">"Num. SNPs contributed to model"</span>) <span class="sc">+</span></span>
<span id="cb17-29"><a href="#cb17-29" aria-hidden="true" tabindex="-1"></a>  <span class="fu">facet_wrap</span>(<span class="sc">~</span>exposure)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="index_files/figure-html/unnamed-chunk-14-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
</section>
<section id="interpretation" class="level2">
<h2 class="anchored" data-anchor-id="interpretation">Interpretation</h2>
<p>….</p>


</section>

</main> <!-- /main -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const isCodeAnnotation = (el) => {
    for (const clz of el.classList) {
      if (clz.startsWith('code-annotation-')) {                     
        return true;
      }
    }
    return false;
  }
  const onCopySuccess = function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  }
  const getTextToCopy = function(trigger) {
      const codeEl = trigger.previousElementSibling.cloneNode(true);
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
  }
  const clipboard = new window.ClipboardJS('.code-copy-button:not([data-in-quarto-modal])', {
    text: getTextToCopy
  });
  clipboard.on('success', onCopySuccess);
  if (window.document.getElementById('quarto-embedded-source-code-modal')) {
    const clipboardModal = new window.ClipboardJS('.code-copy-button[data-in-quarto-modal]', {
      text: getTextToCopy,
      container: window.document.getElementById('quarto-embedded-source-code-modal')
    });
    clipboardModal.on('success', onCopySuccess);
  }
    var localhostRegex = new RegExp(/^(?:http|https):\/\/localhost\:?[0-9]*\//);
    var mailtoRegex = new RegExp(/^mailto:/);
      var filterRegex = new RegExp('/' + window.location.host + '/');
    var isInternal = (href) => {
        return filterRegex.test(href) || localhostRegex.test(href) || mailtoRegex.test(href);
    }
    // Inspect non-navigation links and adorn them if external
 	var links = window.document.querySelectorAll('a[href]:not(.nav-link):not(.navbar-brand):not(.toc-action):not(.sidebar-link):not(.sidebar-item-toggle):not(.pagination-link):not(.no-external):not([aria-hidden]):not(.dropdown-item):not(.quarto-navigation-tool):not(.about-link)');
    for (var i=0; i<links.length; i++) {
      const link = links[i];
      if (!isInternal(link.href)) {
        // undo the damage that might have been done by quarto-nav.js in the case of
        // links that we want to consider external
        if (link.dataset.originalHref !== undefined) {
          link.href = link.dataset.originalHref;
        }
      }
    }
  function tippyHover(el, contentFn, onTriggerFn, onUntriggerFn) {
    const config = {
      allowHTML: true,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start',
    };
    if (contentFn) {
      config.content = contentFn;
    }
    if (onTriggerFn) {
      config.onTrigger = onTriggerFn;
    }
    if (onUntriggerFn) {
      config.onUntrigger = onUntriggerFn;
    }
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      if (note) {
        return note.innerHTML;
      } else {
        return "";
      }
    });
  }
  const xrefs = window.document.querySelectorAll('a.quarto-xref');
  const processXRef = (id, note) => {
    // Strip column container classes
    const stripColumnClz = (el) => {
      el.classList.remove("page-full", "page-columns");
      if (el.children) {
        for (const child of el.children) {
          stripColumnClz(child);
        }
      }
    }
    stripColumnClz(note)
    if (id === null || id.startsWith('sec-')) {
      // Special case sections, only their first couple elements
      const container = document.createElement("div");
      if (note.children && note.children.length > 2) {
        container.appendChild(note.children[0].cloneNode(true));
        for (let i = 1; i < note.children.length; i++) {
          const child = note.children[i];
          if (child.tagName === "P" && child.innerText === "") {
            continue;
          } else {
            container.appendChild(child.cloneNode(true));
            break;
          }
        }
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(container);
        }
        return container.innerHTML
      } else {
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(note);
        }
        return note.innerHTML;
      }
    } else {
      // Remove any anchor links if they are present
      const anchorLink = note.querySelector('a.anchorjs-link');
      if (anchorLink) {
        anchorLink.remove();
      }
      if (window.Quarto?.typesetMath) {
        window.Quarto.typesetMath(note);
      }
      if (note.classList.contains("callout")) {
        return note.outerHTML;
      } else {
        return note.innerHTML;
      }
    }
  }
  for (var i=0; i<xrefs.length; i++) {
    const xref = xrefs[i];
    tippyHover(xref, undefined, function(instance) {
      instance.disable();
      let url = xref.getAttribute('href');
      let hash = undefined; 
      if (url.startsWith('#')) {
        hash = url;
      } else {
        try { hash = new URL(url).hash; } catch {}
      }
      if (hash) {
        const id = hash.replace(/^#\/?/, "");
        const note = window.document.getElementById(id);
        if (note !== null) {
          try {
            const html = processXRef(id, note.cloneNode(true));
            instance.setContent(html);
          } finally {
            instance.enable();
            instance.show();
          }
        } else {
          // See if we can fetch this
          fetch(url.split('#')[0])
          .then(res => res.text())
          .then(html => {
            const parser = new DOMParser();
            const htmlDoc = parser.parseFromString(html, "text/html");
            const note = htmlDoc.getElementById(id);
            if (note !== null) {
              const html = processXRef(id, note);
              instance.setContent(html);
            } 
          }).finally(() => {
            instance.enable();
            instance.show();
          });
        }
      } else {
        // See if we can fetch a full url (with no hash to target)
        // This is a special case and we should probably do some content thinning / targeting
        fetch(url)
        .then(res => res.text())
        .then(html => {
          const parser = new DOMParser();
          const htmlDoc = parser.parseFromString(html, "text/html");
          const note = htmlDoc.querySelector('main.content');
          if (note !== null) {
            // This should only happen for chapter cross references
            // (since there is no id in the URL)
            // remove the first header
            if (note.children.length > 0 && note.children[0].tagName === "HEADER") {
              note.children[0].remove();
            }
            const html = processXRef(null, note);
            instance.setContent(html);
          } 
        }).finally(() => {
          instance.enable();
          instance.show();
        });
      }
    }, function(instance) {
    });
  }
      let selectedAnnoteEl;
      const selectorForAnnotation = ( cell, annotation) => {
        let cellAttr = 'data-code-cell="' + cell + '"';
        let lineAttr = 'data-code-annotation="' +  annotation + '"';
        const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
        return selector;
      }
      const selectCodeLines = (annoteEl) => {
        const doc = window.document;
        const targetCell = annoteEl.getAttribute("data-target-cell");
        const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
        const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
        const lines = annoteSpan.getAttribute("data-code-lines").split(",");
        const lineIds = lines.map((line) => {
          return targetCell + "-" + line;
        })
        let top = null;
        let height = null;
        let parent = null;
        if (lineIds.length > 0) {
            //compute the position of the single el (top and bottom and make a div)
            const el = window.document.getElementById(lineIds[0]);
            top = el.offsetTop;
            height = el.offsetHeight;
            parent = el.parentElement.parentElement;
          if (lineIds.length > 1) {
            const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
            const bottom = lastEl.offsetTop + lastEl.offsetHeight;
            height = bottom - top;
          }
          if (top !== null && height !== null && parent !== null) {
            // cook up a div (if necessary) and position it 
            let div = window.document.getElementById("code-annotation-line-highlight");
            if (div === null) {
              div = window.document.createElement("div");
              div.setAttribute("id", "code-annotation-line-highlight");
              div.style.position = 'absolute';
              parent.appendChild(div);
            }
            div.style.top = top - 2 + "px";
            div.style.height = height + 4 + "px";
            div.style.left = 0;
            let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
            if (gutterDiv === null) {
              gutterDiv = window.document.createElement("div");
              gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
              gutterDiv.style.position = 'absolute';
              const codeCell = window.document.getElementById(targetCell);
              const gutter = codeCell.querySelector('.code-annotation-gutter');
              gutter.appendChild(gutterDiv);
            }
            gutterDiv.style.top = top - 2 + "px";
            gutterDiv.style.height = height + 4 + "px";
          }
          selectedAnnoteEl = annoteEl;
        }
      };
      const unselectCodeLines = () => {
        const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
        elementsIds.forEach((elId) => {
          const div = window.document.getElementById(elId);
          if (div) {
            div.remove();
          }
        });
        selectedAnnoteEl = undefined;
      };
        // Handle positioning of the toggle
    window.addEventListener(
      "resize",
      throttle(() => {
        elRect = undefined;
        if (selectedAnnoteEl) {
          selectCodeLines(selectedAnnoteEl);
        }
      }, 10)
    );
    function throttle(fn, ms) {
    let throttle = false;
    let timer;
      return (...args) => {
        if(!throttle) { // first call gets through
            fn.apply(this, args);
            throttle = true;
        } else { // all the others get throttled
            if(timer) clearTimeout(timer); // cancel #2
            timer = setTimeout(() => {
              fn.apply(this, args);
              timer = throttle = false;
            }, ms);
        }
      };
    }
      // Attach click handler to the DT
      const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
      for (const annoteDlNode of annoteDls) {
        annoteDlNode.addEventListener('click', (event) => {
          const clickedEl = event.target;
          if (clickedEl !== selectedAnnoteEl) {
            unselectCodeLines();
            const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
            if (activeEl) {
              activeEl.classList.remove('code-annotation-active');
            }
            selectCodeLines(clickedEl);
            clickedEl.classList.add('code-annotation-active');
          } else {
            // Unselect the line
            unselectCodeLines();
            clickedEl.classList.remove('code-annotation-active');
          }
        });
      }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script>
</div> <!-- /content -->




</body></html>