<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.6.40">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">

<meta name="author" content="Nick Sunderland">
<meta name="dcterms.date" content="2025-01-10">

<title>Tissue-partitioned BMI instrument procedure – Nick Sunderland’s lab book</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
/* CSS for syntax highlighting */
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { display: inline-block; text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
  }
pre.numberSource { margin-left: 3em;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
</style>


<script src="../../site_libs/quarto-nav/quarto-nav.js"></script>
<script src="../../site_libs/quarto-nav/headroom.min.js"></script>
<script src="../../site_libs/clipboard/clipboard.min.js"></script>
<script src="../../site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="../../site_libs/quarto-search/fuse.min.js"></script>
<script src="../../site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="../../">
<script src="../../site_libs/quarto-html/quarto.js"></script>
<script src="../../site_libs/quarto-html/popper.min.js"></script>
<script src="../../site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="../../site_libs/quarto-html/anchor.min.js"></script>
<link href="../../site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="../../site_libs/quarto-html/quarto-syntax-highlighting-549806ee2085284f45b00abea8c6df48.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="../../site_libs/bootstrap/bootstrap.min.js"></script>
<link href="../../site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="../../site_libs/bootstrap/bootstrap-d045e70ebd9269c25e164ecdb4898f56.min.css" rel="stylesheet" append-hash="true" id="quarto-bootstrap" data-mode="light">
<script id="quarto-search-options" type="application/json">{
  "location": "navbar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "end",
  "type": "overlay",
  "limit": 50,
  "keyboard-shortcut": [
    "f",
    "/",
    "s"
  ],
  "show-item-context": false,
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-text-placeholder": "",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit",
    "search-label": "Search"
  }
}</script>


<link rel="stylesheet" href="../../styles.css">
</head>

<body class="nav-fixed fullcontent">

<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top">
    <nav class="navbar navbar-expand-lg " data-bs-theme="dark">
      <div class="navbar-container container-fluid">
      <div class="navbar-brand-container mx-auto">
    <a class="navbar-brand" href="../../index.html">
    <span class="navbar-title">Nick Sunderland’s lab book</span>
    </a>
  </div>
            <div id="quarto-search" class="" title="Search"></div>
          <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbarCollapse" aria-controls="navbarCollapse" role="menu" aria-expanded="false" aria-label="Toggle navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
  <span class="navbar-toggler-icon"></span>
</button>
          <div class="collapse navbar-collapse" id="navbarCollapse">
            <ul class="navbar-nav navbar-nav-scroll ms-auto">
  <li class="nav-item">
    <a class="nav-link" href="../../about.html"> 
<span class="menu-text">About</span></a>
  </li>  
  <li class="nav-item compact">
    <a class="nav-link" href="https://github.com/"> <i class="bi bi-github" role="img">
</i> 
<span class="menu-text"></span></a>
  </li>  
  <li class="nav-item compact">
    <a class="nav-link" href="https://twitter.com"> <i class="bi bi-twitter" role="img">
</i> 
<span class="menu-text"></span></a>
  </li>  
</ul>
          </div> <!-- /navcollapse -->
            <div class="quarto-navbar-tools">
</div>
      </div> <!-- /container-fluid -->
    </nav>
</header>
<!-- content -->
<div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-article page-navbar">
<!-- sidebar -->
<!-- margin-sidebar -->
    
<!-- main -->
<main class="content" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default">
<div class="quarto-title">
<h1 class="title">Tissue-partitioned BMI instrument procedure</h1>
</div>



<div class="quarto-title-meta">

    <div>
    <div class="quarto-title-meta-heading">Author</div>
    <div class="quarto-title-meta-contents">
             <p>Nick Sunderland </p>
          </div>
  </div>
    
    <div>
    <div class="quarto-title-meta-heading">Published</div>
    <div class="quarto-title-meta-contents">
      <p class="date">January 10, 2025</p>
    </div>
  </div>
  
    
  </div>
  


</header>


<section id="motivation" class="level2">
<h2 class="anchored" data-anchor-id="motivation">Motivation</h2>
<p>What does the weighting by PPH4 do to tissue partitioned BMI instruments? Let’s simulate some tissue partitioned BMI MR results.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb1"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(data.table)</span>
<span id="cb1-2"><a href="#cb1-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ggplot2)</span>
<span id="cb1-3"><a href="#cb1-3" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ggpubr)</span>
<span id="cb1-4"><a href="#cb1-4" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ggrepel)</span>
<span id="cb1-5"><a href="#cb1-5" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(genepi.utils)</span>
<span id="cb1-6"><a href="#cb1-6" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(tibble)</span>
<span id="cb1-7"><a href="#cb1-7" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(knitr)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="bmi-data" class="level2">
<h2 class="anchored" data-anchor-id="bmi-data">BMI data</h2>
<p>First load the BMI GWAS.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb2"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb2-1"><a href="#cb2-1" aria-hidden="true" tabindex="-1"></a>bmi    <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(<span class="fu">file.path</span>(TPMR_DIR, <span class="st">"output"</span>, <span class="st">"tmp_objects"</span>, <span class="st">"obj_bmi.RDS"</span>))</span>
<span id="cb2-2"><a href="#cb2-2" aria-hidden="true" tabindex="-1"></a>clumps <span class="ot">&lt;-</span> <span class="fu">clump</span>(<span class="fu">as.data.table</span>(bmi), <span class="at">p1=</span><span class="fl">5e-8</span>, <span class="at">r2=</span><span class="fl">0.001</span>, <span class="at">kb=</span><span class="dv">10000</span>, <span class="at">plink2=</span>PLINK2, <span class="at">plink_ref=</span>UKBB_REFERENCE)[index<span class="sc">==</span><span class="cn">TRUE</span>]</span>
<span id="cb2-3"><a href="#cb2-3" aria-hidden="true" tabindex="-1"></a>bmi    <span class="ot">&lt;-</span> <span class="fu">subset_gwas</span>(bmi, clumps<span class="sc">$</span>rsid)</span>
<span id="cb2-4"><a href="#cb2-4" aria-hidden="true" tabindex="-1"></a><span class="fu">kable</span>(<span class="fu">head</span>(clumps, <span class="dv">5</span>), <span class="at">caption =</span> <span class="st">"BMI lead variants"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<table class="caption-top table table-sm table-striped small">
<caption>BMI lead variants</caption>
<colgroup>
<col style="width: 4%">
<col style="width: 1%">
<col style="width: 4%">
<col style="width: 1%">
<col style="width: 1%">
<col style="width: 3%">
<col style="width: 4%">
<col style="width: 4%">
<col style="width: 1%">
<col style="width: 2%">
<col style="width: 2%">
<col style="width: 2%">
<col style="width: 13%">
<col style="width: 2%">
<col style="width: 3%">
<col style="width: 2%">
<col style="width: 1%">
<col style="width: 1%">
<col style="width: 1%">
<col style="width: 4%">
<col style="width: 4%">
<col style="width: 3%">
<col style="width: 3%">
<col style="width: 3%">
<col style="width: 4%">
<col style="width: 3%">
<col style="width: 4%">
<col style="width: 2%">
<col style="width: 2%">
</colgroup>
<thead>
<tr class="header">
<th style="text-align: left;">rsid</th>
<th style="text-align: left;">chr</th>
<th style="text-align: right;">bp</th>
<th style="text-align: left;">ea</th>
<th style="text-align: left;">oa</th>
<th style="text-align: right;">eaf</th>
<th style="text-align: right;">beta</th>
<th style="text-align: right;">se</th>
<th style="text-align: right;">p</th>
<th style="text-align: right;">n</th>
<th style="text-align: right;">ncase</th>
<th style="text-align: left;">trait</th>
<th style="text-align: left;">id</th>
<th style="text-align: left;">strand</th>
<th style="text-align: left;">imputed</th>
<th style="text-align: right;">info</th>
<th style="text-align: right;">q</th>
<th style="text-align: right;">q_p</th>
<th style="text-align: right;">i2</th>
<th style="text-align: left;">proxy_rsid</th>
<th style="text-align: left;">proxy_chr</th>
<th style="text-align: right;">proxy_bp</th>
<th style="text-align: left;">proxy_ea</th>
<th style="text-align: left;">proxy_oa</th>
<th style="text-align: right;">proxy_eaf</th>
<th style="text-align: right;">proxy_r2</th>
<th style="text-align: left;">snp</th>
<th style="text-align: left;">index</th>
<th style="text-align: left;">clump</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">rs10007906</td>
<td style="text-align: left;">4</td>
<td style="text-align: right;">31023610</td>
<td style="text-align: left;">A</td>
<td style="text-align: left;">C</td>
<td style="text-align: right;">0.360300</td>
<td style="text-align: right;">0.0133000</td>
<td style="text-align: right;">0.0018000</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">686479</td>
<td style="text-align: right;">NA</td>
<td style="text-align: left;">bmi</td>
<td style="text-align: left;">combined_bmi_giant_ukb_annot.txt</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">NA</td>
<td style="text-align: right;">NA</td>
<td style="text-align: right;">NA</td>
<td style="text-align: right;">NA</td>
<td style="text-align: right;">NA</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">NA</td>
<td style="text-align: right;">NA</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">NA</td>
<td style="text-align: right;">NA</td>
<td style="text-align: right;">NA</td>
<td style="text-align: left;">rs10007906</td>
<td style="text-align: left;">TRUE</td>
<td style="text-align: left;">291</td>
</tr>
<tr class="even">
<td style="text-align: left;">rs1000940</td>
<td style="text-align: left;">17</td>
<td style="text-align: right;">5283252</td>
<td style="text-align: left;">A</td>
<td style="text-align: left;">G</td>
<td style="text-align: right;">0.701300</td>
<td style="text-align: right;">-0.0154000</td>
<td style="text-align: right;">0.0018000</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">794558</td>
<td style="text-align: right;">NA</td>
<td style="text-align: left;">bmi</td>
<td style="text-align: left;">combined_bmi_giant_ukb_annot.txt</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">NA</td>
<td style="text-align: right;">NA</td>
<td style="text-align: right;">NA</td>
<td style="text-align: right;">NA</td>
<td style="text-align: right;">NA</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">NA</td>
<td style="text-align: right;">NA</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">NA</td>
<td style="text-align: right;">NA</td>
<td style="text-align: right;">NA</td>
<td style="text-align: left;">rs1000940</td>
<td style="text-align: left;">TRUE</td>
<td style="text-align: left;">142</td>
</tr>
<tr class="odd">
<td style="text-align: left;">rs1007934</td>
<td style="text-align: left;">14</td>
<td style="text-align: right;">73463479</td>
<td style="text-align: left;">A</td>
<td style="text-align: left;">G</td>
<td style="text-align: right;">0.415000</td>
<td style="text-align: right;">-0.0120000</td>
<td style="text-align: right;">0.0017000</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">692411</td>
<td style="text-align: right;">NA</td>
<td style="text-align: left;">bmi</td>
<td style="text-align: left;">combined_bmi_giant_ukb_annot.txt</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">NA</td>
<td style="text-align: right;">NA</td>
<td style="text-align: right;">NA</td>
<td style="text-align: right;">NA</td>
<td style="text-align: right;">NA</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">NA</td>
<td style="text-align: right;">NA</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">NA</td>
<td style="text-align: right;">NA</td>
<td style="text-align: right;">NA</td>
<td style="text-align: left;">rs1007934</td>
<td style="text-align: left;">TRUE</td>
<td style="text-align: left;">407</td>
</tr>
<tr class="even">
<td style="text-align: left;">rs10125137</td>
<td style="text-align: left;">9</td>
<td style="text-align: right;">118644190</td>
<td style="text-align: left;">T</td>
<td style="text-align: left;">C</td>
<td style="text-align: right;">0.641786</td>
<td style="text-align: right;">-0.0117273</td>
<td style="text-align: right;">0.0020523</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">463005</td>
<td style="text-align: right;">NA</td>
<td style="text-align: left;">bmi</td>
<td style="text-align: left;">combined_bmi_giant_ukb_annot.txt</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">NA</td>
<td style="text-align: right;">NA</td>
<td style="text-align: right;">NA</td>
<td style="text-align: right;">NA</td>
<td style="text-align: right;">NA</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">NA</td>
<td style="text-align: right;">NA</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">NA</td>
<td style="text-align: right;">NA</td>
<td style="text-align: right;">NA</td>
<td style="text-align: left;">rs10125137</td>
<td style="text-align: left;">TRUE</td>
<td style="text-align: left;">811</td>
</tr>
<tr class="odd">
<td style="text-align: left;">rs10131890</td>
<td style="text-align: left;">14</td>
<td style="text-align: right;">97258752</td>
<td style="text-align: left;">A</td>
<td style="text-align: left;">C</td>
<td style="text-align: right;">0.950300</td>
<td style="text-align: right;">-0.0222000</td>
<td style="text-align: right;">0.0039000</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">691777</td>
<td style="text-align: right;">NA</td>
<td style="text-align: left;">bmi</td>
<td style="text-align: left;">combined_bmi_giant_ukb_annot.txt</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">NA</td>
<td style="text-align: right;">NA</td>
<td style="text-align: right;">NA</td>
<td style="text-align: right;">NA</td>
<td style="text-align: right;">NA</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">NA</td>
<td style="text-align: right;">NA</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">NA</td>
<td style="text-align: right;">NA</td>
<td style="text-align: right;">NA</td>
<td style="text-align: left;">rs10131890</td>
<td style="text-align: left;">TRUE</td>
<td style="text-align: left;">846</td>
</tr>
</tbody>
</table>
</div>
</div>
</section>
<section id="heart-failure-data" class="level2">
<h2 class="anchored" data-anchor-id="heart-failure-data">Heart failure data</h2>
<p>Load the heart failure GWAS.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb3"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1" aria-hidden="true" tabindex="-1"></a>hf <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(<span class="fu">file.path</span>(TPMR_DIR, <span class="st">"output"</span>, <span class="st">"tmp_objects"</span>, <span class="st">"obj_hf_allcause.RDS"</span>))</span>
<span id="cb3-2"><a href="#cb3-2" aria-hidden="true" tabindex="-1"></a>hf <span class="ot">&lt;-</span> <span class="fu">subset_gwas</span>(hf, clumps<span class="sc">$</span>rsid)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="harmonise-the-datasets" class="level2">
<h2 class="anchored" data-anchor-id="harmonise-the-datasets">Harmonise the datasets</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb4"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb4-1"><a href="#cb4-1" aria-hidden="true" tabindex="-1"></a>h <span class="ot">&lt;-</span> <span class="fu">MR</span>(bmi, hf)</span>
<span id="cb4-2"><a href="#cb4-2" aria-hidden="true" tabindex="-1"></a><span class="fu">kable</span>(<span class="fu">head</span>(<span class="fu">as.data.table</span>(h),<span class="dv">5</span>), <span class="at">caption =</span> <span class="st">"Harmonised BMI:HF dataset"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<table class="caption-top table table-sm table-striped small">
<caption>Harmonised BMI:HF dataset</caption>
<colgroup>
<col style="width: 9%">
<col style="width: 2%">
<col style="width: 6%">
<col style="width: 1%">
<col style="width: 1%">
<col style="width: 7%">
<col style="width: 6%">
<col style="width: 1%">
<col style="width: 5%">
<col style="width: 4%">
<col style="width: 5%">
<col style="width: 7%">
<col style="width: 7%">
<col style="width: 6%">
<col style="width: 3%">
<col style="width: 5%">
<col style="width: 5%">
<col style="width: 7%">
</colgroup>
<thead>
<tr class="header">
<th style="text-align: left;">rsid</th>
<th style="text-align: left;">chr</th>
<th style="text-align: right;">bp</th>
<th style="text-align: left;">ea</th>
<th style="text-align: left;">oa</th>
<th style="text-align: right;">bx</th>
<th style="text-align: right;">bxse</th>
<th style="text-align: right;">px</th>
<th style="text-align: right;">by</th>
<th style="text-align: right;">byse</th>
<th style="text-align: right;">py</th>
<th style="text-align: left;">proxy_e_snp</th>
<th style="text-align: left;">proxy_o_snp</th>
<th style="text-align: left;">index_snp</th>
<th style="text-align: right;">group</th>
<th style="text-align: left;">ld_info</th>
<th style="text-align: left;">exposure</th>
<th style="text-align: left;">outcome</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">rs10007906_C_A</td>
<td style="text-align: left;">4</td>
<td style="text-align: right;">31023610</td>
<td style="text-align: left;">A</td>
<td style="text-align: left;">C</td>
<td style="text-align: right;">0.0133000</td>
<td style="text-align: right;">0.0018000</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">0.0039</td>
<td style="text-align: right;">0.0051</td>
<td style="text-align: right;">0.44980</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">TRUE</td>
<td style="text-align: right;">NA</td>
<td style="text-align: left;">FALSE</td>
<td style="text-align: left;">bmi</td>
<td style="text-align: left;">hf_allcause</td>
</tr>
<tr class="even">
<td style="text-align: left;">rs1000940_G_A</td>
<td style="text-align: left;">17</td>
<td style="text-align: right;">5283252</td>
<td style="text-align: left;">A</td>
<td style="text-align: left;">G</td>
<td style="text-align: right;">-0.0154000</td>
<td style="text-align: right;">0.0018000</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">0.0011</td>
<td style="text-align: right;">0.0050</td>
<td style="text-align: right;">0.82970</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">TRUE</td>
<td style="text-align: right;">NA</td>
<td style="text-align: left;">FALSE</td>
<td style="text-align: left;">bmi</td>
<td style="text-align: left;">hf_allcause</td>
</tr>
<tr class="odd">
<td style="text-align: left;">rs1007934_G_A</td>
<td style="text-align: left;">14</td>
<td style="text-align: right;">73463479</td>
<td style="text-align: left;">A</td>
<td style="text-align: left;">G</td>
<td style="text-align: right;">-0.0120000</td>
<td style="text-align: right;">0.0017000</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">-0.0092</td>
<td style="text-align: right;">0.0049</td>
<td style="text-align: right;">0.05931</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">TRUE</td>
<td style="text-align: right;">NA</td>
<td style="text-align: left;">FALSE</td>
<td style="text-align: left;">bmi</td>
<td style="text-align: left;">hf_allcause</td>
</tr>
<tr class="even">
<td style="text-align: left;">rs10125137_C_T</td>
<td style="text-align: left;">9</td>
<td style="text-align: right;">118644190</td>
<td style="text-align: left;">T</td>
<td style="text-align: left;">C</td>
<td style="text-align: right;">-0.0117273</td>
<td style="text-align: right;">0.0020523</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">0.0033</td>
<td style="text-align: right;">0.0049</td>
<td style="text-align: right;">0.50820</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">TRUE</td>
<td style="text-align: right;">NA</td>
<td style="text-align: left;">FALSE</td>
<td style="text-align: left;">bmi</td>
<td style="text-align: left;">hf_allcause</td>
</tr>
<tr class="odd">
<td style="text-align: left;">rs10131890_C_A</td>
<td style="text-align: left;">14</td>
<td style="text-align: right;">97258752</td>
<td style="text-align: left;">A</td>
<td style="text-align: left;">C</td>
<td style="text-align: right;">-0.0222000</td>
<td style="text-align: right;">0.0039000</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">-0.0211</td>
<td style="text-align: right;">0.0099</td>
<td style="text-align: right;">0.03310</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">TRUE</td>
<td style="text-align: right;">NA</td>
<td style="text-align: left;">FALSE</td>
<td style="text-align: left;">bmi</td>
<td style="text-align: left;">hf_allcause</td>
</tr>
</tbody>
</table>
</div>
</div>
</section>
<section id="run-the-mr" class="level2">
<h2 class="anchored" data-anchor-id="run-the-mr">Run the MR</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb5"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb5-1"><a href="#cb5-1" aria-hidden="true" tabindex="-1"></a>res <span class="ot">&lt;-</span> <span class="fu">run_mr</span>(h)</span>
<span id="cb5-2"><a href="#cb5-2" aria-hidden="true" tabindex="-1"></a><span class="fu">kable</span>(res, <span class="at">caption =</span> <span class="st">"BMI:HF Mendelian randomisation"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<table class="caption-top table table-sm table-striped small">
<caption>BMI:HF Mendelian randomisation</caption>
<colgroup>
<col style="width: 9%">
<col style="width: 5%">
<col style="width: 4%">
<col style="width: 5%">
<col style="width: 2%">
<col style="width: 4%">
<col style="width: 4%">
<col style="width: 1%">
<col style="width: 5%">
<col style="width: 4%">
<col style="width: 4%">
<col style="width: 4%">
<col style="width: 3%">
<col style="width: 4%">
<col style="width: 4%">
<col style="width: 7%">
<col style="width: 2%">
<col style="width: 4%">
<col style="width: 7%">
<col style="width: 7%">
</colgroup>
<thead>
<tr class="header">
<th style="text-align: left;">method</th>
<th style="text-align: left;">correlation</th>
<th style="text-align: left;">exposure</th>
<th style="text-align: left;">outcome</th>
<th style="text-align: right;">n_snp</th>
<th style="text-align: right;">b</th>
<th style="text-align: right;">b_se</th>
<th style="text-align: right;">p</th>
<th style="text-align: right;">intercept</th>
<th style="text-align: right;">int_se</th>
<th style="text-align: right;">int_p</th>
<th style="text-align: right;">qstat</th>
<th style="text-align: right;">qstat_p</th>
<th style="text-align: right;">fstat</th>
<th style="text-align: right;">condfstat</th>
<th style="text-align: right;">overdispersion</th>
<th style="text-align: right;">n_pc</th>
<th style="text-align: right;">n_hunted</th>
<th style="text-align: right;">slopehunter_pi</th>
<th style="text-align: right;">slopehunter_ent</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">mr_ivw</td>
<td style="text-align: left;">FALSE</td>
<td style="text-align: left;">bmi</td>
<td style="text-align: left;">hf_allcause</td>
<td style="text-align: right;">874</td>
<td style="text-align: right;">0.4872820</td>
<td style="text-align: right;">0.0165599</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">0.0000000</td>
<td style="text-align: right;">NA</td>
<td style="text-align: right;">NA</td>
<td style="text-align: right;">1592.460</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">60.70091</td>
<td style="text-align: right;">NA</td>
<td style="text-align: right;">NA</td>
<td style="text-align: right;">NA</td>
<td style="text-align: right;">NA</td>
<td style="text-align: right;">NA</td>
<td style="text-align: right;">NA</td>
</tr>
<tr class="even">
<td style="text-align: left;">mr_egger</td>
<td style="text-align: left;">FALSE</td>
<td style="text-align: left;">bmi</td>
<td style="text-align: left;">hf_allcause</td>
<td style="text-align: right;">874</td>
<td style="text-align: right;">0.5367727</td>
<td style="text-align: right;">0.0464790</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">-0.0008232</td>
<td style="text-align: right;">0.0007224</td>
<td style="text-align: right;">0.2544728</td>
<td style="text-align: right;">1590.092</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">NA</td>
<td style="text-align: right;">NA</td>
<td style="text-align: right;">NA</td>
<td style="text-align: right;">NA</td>
<td style="text-align: right;">NA</td>
<td style="text-align: right;">NA</td>
<td style="text-align: right;">NA</td>
</tr>
<tr class="odd">
<td style="text-align: left;">mr_weighted_median</td>
<td style="text-align: left;">FALSE</td>
<td style="text-align: left;">bmi</td>
<td style="text-align: left;">hf_allcause</td>
<td style="text-align: right;">874</td>
<td style="text-align: right;">0.5078217</td>
<td style="text-align: right;">0.0236696</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">0.0000000</td>
<td style="text-align: right;">NA</td>
<td style="text-align: right;">NA</td>
<td style="text-align: right;">NA</td>
<td style="text-align: right;">NA</td>
<td style="text-align: right;">NA</td>
<td style="text-align: right;">NA</td>
<td style="text-align: right;">NA</td>
<td style="text-align: right;">NA</td>
<td style="text-align: right;">NA</td>
<td style="text-align: right;">NA</td>
<td style="text-align: right;">NA</td>
</tr>
<tr class="even">
<td style="text-align: left;">mr_weighted_mode</td>
<td style="text-align: left;">FALSE</td>
<td style="text-align: left;">bmi</td>
<td style="text-align: left;">hf_allcause</td>
<td style="text-align: right;">874</td>
<td style="text-align: right;">0.5456300</td>
<td style="text-align: right;">0.0523114</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">0.0000000</td>
<td style="text-align: right;">NA</td>
<td style="text-align: right;">NA</td>
<td style="text-align: right;">NA</td>
<td style="text-align: right;">NA</td>
<td style="text-align: right;">NA</td>
<td style="text-align: right;">NA</td>
<td style="text-align: right;">NA</td>
<td style="text-align: right;">NA</td>
<td style="text-align: right;">NA</td>
<td style="text-align: right;">NA</td>
<td style="text-align: right;">NA</td>
</tr>
</tbody>
</table>
</div>
</div>
</section>
<section id="plot-mr" class="level2">
<h2 class="anchored" data-anchor-id="plot-mr">Plot MR</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb6"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb6-1"><a href="#cb6-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plot_mr</span>(h, res)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="index_files/figure-html/unnamed-chunk-5-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
</section>
<section id="select-variants---mr-with-a-subset-of-bmi-variants" class="level2">
<h2 class="anchored" data-anchor-id="select-variants---mr-with-a-subset-of-bmi-variants">Select variants - MR with a subset of BMI variants</h2>
<p>In the Leyden et al.&nbsp;2022 paper 86 variant colocalise with adipose tissue eQTLs and 140 colocalise with brain tissue eQTLs. There is overlap between the two tissues of 43 variants.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb7"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb7-1"><a href="#cb7-1" aria-hidden="true" tabindex="-1"></a><span class="fu">set.seed</span>(<span class="dv">123</span>)</span>
<span id="cb7-2"><a href="#cb7-2" aria-hidden="true" tabindex="-1"></a>tissue1_varid <span class="ot">&lt;-</span> <span class="fu">sample</span>(h<span class="sc">@</span>snps, <span class="dv">140</span>)</span>
<span id="cb7-3"><a href="#cb7-3" aria-hidden="true" tabindex="-1"></a>overlap_varid <span class="ot">&lt;-</span> <span class="fu">sample</span>(tissue1_varid, <span class="dv">43</span>)</span>
<span id="cb7-4"><a href="#cb7-4" aria-hidden="true" tabindex="-1"></a>tissue2_varid <span class="ot">&lt;-</span> <span class="fu">c</span>(overlap_varid, <span class="fu">sample</span>(h<span class="sc">@</span>snps[<span class="sc">!</span>h<span class="sc">@</span>snps <span class="sc">%in%</span> overlap_varid], <span class="dv">86-43</span>))</span>
<span id="cb7-5"><a href="#cb7-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-6"><a href="#cb7-6" aria-hidden="true" tabindex="-1"></a><span class="co"># turn of variants not in either tissue</span></span>
<span id="cb7-7"><a href="#cb7-7" aria-hidden="true" tabindex="-1"></a>h<span class="sc">@</span>index_snp[<span class="sc">!</span>h<span class="sc">@</span>snps <span class="sc">%in%</span> <span class="fu">c</span>(tissue1_varid, tissue2_varid)] <span class="ot">&lt;-</span> <span class="cn">FALSE</span></span>
<span id="cb7-8"><a href="#cb7-8" aria-hidden="true" tabindex="-1"></a>h<span class="sc">@</span>exposure <span class="ot">&lt;-</span> <span class="st">"BMI tissue 1/2 SNPs"</span></span>
<span id="cb7-9"><a href="#cb7-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-10"><a href="#cb7-10" aria-hidden="true" tabindex="-1"></a>res1 <span class="ot">&lt;-</span> <span class="fu">run_mr</span>(h)</span>
<span id="cb7-11"><a href="#cb7-11" aria-hidden="true" tabindex="-1"></a><span class="fu">kable</span>(res1, <span class="at">caption =</span> <span class="st">"BMI:HF Mendelian randomisation (Tissue 1 &amp; 2 SNPs only)"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<table class="caption-top table table-sm table-striped small">
<caption>BMI:HF Mendelian randomisation (Tissue 1 &amp; 2 SNPs only)</caption>
<colgroup>
<col style="width: 8%">
<col style="width: 5%">
<col style="width: 9%">
<col style="width: 5%">
<col style="width: 2%">
<col style="width: 4%">
<col style="width: 4%">
<col style="width: 2%">
<col style="width: 4%">
<col style="width: 4%">
<col style="width: 4%">
<col style="width: 4%">
<col style="width: 3%">
<col style="width: 4%">
<col style="width: 4%">
<col style="width: 6%">
<col style="width: 2%">
<col style="width: 4%">
<col style="width: 6%">
<col style="width: 7%">
</colgroup>
<thead>
<tr class="header">
<th style="text-align: left;">method</th>
<th style="text-align: left;">correlation</th>
<th style="text-align: left;">exposure</th>
<th style="text-align: left;">outcome</th>
<th style="text-align: right;">n_snp</th>
<th style="text-align: right;">b</th>
<th style="text-align: right;">b_se</th>
<th style="text-align: right;">p</th>
<th style="text-align: right;">intercept</th>
<th style="text-align: right;">int_se</th>
<th style="text-align: right;">int_p</th>
<th style="text-align: right;">qstat</th>
<th style="text-align: right;">qstat_p</th>
<th style="text-align: right;">fstat</th>
<th style="text-align: right;">condfstat</th>
<th style="text-align: right;">overdispersion</th>
<th style="text-align: right;">n_pc</th>
<th style="text-align: right;">n_hunted</th>
<th style="text-align: right;">slopehunter_pi</th>
<th style="text-align: right;">slopehunter_ent</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">mr_ivw</td>
<td style="text-align: left;">FALSE</td>
<td style="text-align: left;">BMI tissue 1/2 SNPs</td>
<td style="text-align: left;">hf_allcause</td>
<td style="text-align: right;">177</td>
<td style="text-align: right;">0.5108471</td>
<td style="text-align: right;">0.0363407</td>
<td style="text-align: right;">0e+00</td>
<td style="text-align: right;">0.0000000</td>
<td style="text-align: right;">NA</td>
<td style="text-align: right;">NA</td>
<td style="text-align: right;">304.2402</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">61.28134</td>
<td style="text-align: right;">NA</td>
<td style="text-align: right;">NA</td>
<td style="text-align: right;">NA</td>
<td style="text-align: right;">NA</td>
<td style="text-align: right;">NA</td>
<td style="text-align: right;">NA</td>
</tr>
<tr class="even">
<td style="text-align: left;">mr_egger</td>
<td style="text-align: left;">FALSE</td>
<td style="text-align: left;">BMI tissue 1/2 SNPs</td>
<td style="text-align: left;">hf_allcause</td>
<td style="text-align: right;">177</td>
<td style="text-align: right;">0.5475313</td>
<td style="text-align: right;">0.1037260</td>
<td style="text-align: right;">1e-07</td>
<td style="text-align: right;">-0.0006034</td>
<td style="text-align: right;">0.0015974</td>
<td style="text-align: right;">0.7056336</td>
<td style="text-align: right;">303.9924</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">NA</td>
<td style="text-align: right;">NA</td>
<td style="text-align: right;">NA</td>
<td style="text-align: right;">NA</td>
<td style="text-align: right;">NA</td>
<td style="text-align: right;">NA</td>
<td style="text-align: right;">NA</td>
</tr>
<tr class="odd">
<td style="text-align: left;">mr_weighted_median</td>
<td style="text-align: left;">FALSE</td>
<td style="text-align: left;">BMI tissue 1/2 SNPs</td>
<td style="text-align: left;">hf_allcause</td>
<td style="text-align: right;">177</td>
<td style="text-align: right;">0.5730810</td>
<td style="text-align: right;">0.0494983</td>
<td style="text-align: right;">0e+00</td>
<td style="text-align: right;">0.0000000</td>
<td style="text-align: right;">NA</td>
<td style="text-align: right;">NA</td>
<td style="text-align: right;">NA</td>
<td style="text-align: right;">NA</td>
<td style="text-align: right;">NA</td>
<td style="text-align: right;">NA</td>
<td style="text-align: right;">NA</td>
<td style="text-align: right;">NA</td>
<td style="text-align: right;">NA</td>
<td style="text-align: right;">NA</td>
<td style="text-align: right;">NA</td>
</tr>
<tr class="even">
<td style="text-align: left;">mr_weighted_mode</td>
<td style="text-align: left;">FALSE</td>
<td style="text-align: left;">BMI tissue 1/2 SNPs</td>
<td style="text-align: left;">hf_allcause</td>
<td style="text-align: right;">177</td>
<td style="text-align: right;">0.5887868</td>
<td style="text-align: right;">0.1022882</td>
<td style="text-align: right;">0e+00</td>
<td style="text-align: right;">0.0000000</td>
<td style="text-align: right;">NA</td>
<td style="text-align: right;">NA</td>
<td style="text-align: right;">NA</td>
<td style="text-align: right;">NA</td>
<td style="text-align: right;">NA</td>
<td style="text-align: right;">NA</td>
<td style="text-align: right;">NA</td>
<td style="text-align: right;">NA</td>
<td style="text-align: right;">NA</td>
<td style="text-align: right;">NA</td>
<td style="text-align: right;">NA</td>
</tr>
</tbody>
</table>
</div>
<div class="sourceCode cell-code" id="cb8"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb8-1"><a href="#cb8-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plot_mr</span>(h, res1)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="index_files/figure-html/unnamed-chunk-6-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
</section>
<section id="run-multivariable-mr-unweighted" class="level2">
<h2 class="anchored" data-anchor-id="run-multivariable-mr-unweighted">Run multivariable MR (unweighted)</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb9"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb9-1"><a href="#cb9-1" aria-hidden="true" tabindex="-1"></a><span class="co"># copy of BMI for tissue 1</span></span>
<span id="cb9-2"><a href="#cb9-2" aria-hidden="true" tabindex="-1"></a>bmi_t1 <span class="ot">&lt;-</span> bmi</span>
<span id="cb9-3"><a href="#cb9-3" aria-hidden="true" tabindex="-1"></a>bmi_t1<span class="sc">@</span>id <span class="ot">&lt;-</span> <span class="st">"bmi tissue 1"</span></span>
<span id="cb9-4"><a href="#cb9-4" aria-hidden="true" tabindex="-1"></a>bmi_t1<span class="sc">@</span>trait <span class="ot">&lt;-</span> <span class="st">"bmi tissue 1"</span></span>
<span id="cb9-5"><a href="#cb9-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb9-6"><a href="#cb9-6" aria-hidden="true" tabindex="-1"></a><span class="co"># copy of BMI for tissue 2</span></span>
<span id="cb9-7"><a href="#cb9-7" aria-hidden="true" tabindex="-1"></a>bmi_t2 <span class="ot">&lt;-</span> bmi</span>
<span id="cb9-8"><a href="#cb9-8" aria-hidden="true" tabindex="-1"></a>bmi_t2<span class="sc">@</span>id <span class="ot">&lt;-</span> <span class="st">"bmi tissue 2"</span></span>
<span id="cb9-9"><a href="#cb9-9" aria-hidden="true" tabindex="-1"></a>bmi_t2<span class="sc">@</span>trait <span class="ot">&lt;-</span> <span class="st">"bmi tissue 2"</span></span>
<span id="cb9-10"><a href="#cb9-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb9-11"><a href="#cb9-11" aria-hidden="true" tabindex="-1"></a><span class="co"># harmonise</span></span>
<span id="cb9-12"><a href="#cb9-12" aria-hidden="true" tabindex="-1"></a>h_mvmr <span class="ot">&lt;-</span> <span class="fu">MR</span>(<span class="fu">list</span>(bmi_t1, bmi_t2), hf)</span>
<span id="cb9-13"><a href="#cb9-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb9-14"><a href="#cb9-14" aria-hidden="true" tabindex="-1"></a><span class="co"># turn of variants not in either tissue</span></span>
<span id="cb9-15"><a href="#cb9-15" aria-hidden="true" tabindex="-1"></a>h_mvmr<span class="sc">@</span>index_snp[<span class="sc">!</span>h_mvmr<span class="sc">@</span>snps <span class="sc">%in%</span> <span class="fu">c</span>(tissue1_varid, tissue2_varid)] <span class="ot">&lt;-</span> <span class="cn">FALSE</span></span>
<span id="cb9-16"><a href="#cb9-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb9-17"><a href="#cb9-17" aria-hidden="true" tabindex="-1"></a>res2 <span class="ot">&lt;-</span> <span class="fu">run_mr</span>(h_mvmr, <span class="at">methods=</span><span class="st">"mr_ivw"</span>)</span>
<span id="cb9-18"><a href="#cb9-18" aria-hidden="true" tabindex="-1"></a><span class="fu">kable</span>(res2, <span class="at">caption =</span> <span class="st">"BMI:HF MV Mendelian randomisation"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<table class="caption-top table table-sm table-striped small">
<caption>BMI:HF MV Mendelian randomisation</caption>
<colgroup>
<col style="width: 4%">
<col style="width: 6%">
<col style="width: 6%">
<col style="width: 6%">
<col style="width: 3%">
<col style="width: 5%">
<col style="width: 5%">
<col style="width: 1%">
<col style="width: 5%">
<col style="width: 3%">
<col style="width: 3%">
<col style="width: 4%">
<col style="width: 4%">
<col style="width: 3%">
<col style="width: 5%">
<col style="width: 7%">
<col style="width: 2%">
<col style="width: 4%">
<col style="width: 7%">
<col style="width: 8%">
</colgroup>
<thead>
<tr class="header">
<th style="text-align: left;">method</th>
<th style="text-align: left;">correlation</th>
<th style="text-align: left;">exposure</th>
<th style="text-align: left;">outcome</th>
<th style="text-align: right;">n_snp</th>
<th style="text-align: right;">b</th>
<th style="text-align: right;">b_se</th>
<th style="text-align: right;">p</th>
<th style="text-align: right;">intercept</th>
<th style="text-align: right;">int_se</th>
<th style="text-align: right;">int_p</th>
<th style="text-align: right;">qstat</th>
<th style="text-align: right;">qstat_p</th>
<th style="text-align: right;">fstat</th>
<th style="text-align: right;">condfstat</th>
<th style="text-align: right;">overdispersion</th>
<th style="text-align: right;">n_pc</th>
<th style="text-align: right;">n_hunted</th>
<th style="text-align: right;">slopehunter_pi</th>
<th style="text-align: right;">slopehunter_ent</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">mr_mvivw</td>
<td style="text-align: left;">FALSE</td>
<td style="text-align: left;">bmi tissue 1</td>
<td style="text-align: left;">hf_allcause</td>
<td style="text-align: right;">177</td>
<td style="text-align: right;">0.5108471</td>
<td style="text-align: right;">0.0363407</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">NA</td>
<td style="text-align: right;">NA</td>
<td style="text-align: right;">302.5116</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">NA</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">NA</td>
<td style="text-align: right;">NA</td>
<td style="text-align: right;">NA</td>
<td style="text-align: right;">NA</td>
<td style="text-align: right;">NA</td>
</tr>
<tr class="even">
<td style="text-align: left;">mr_mvivw</td>
<td style="text-align: left;">FALSE</td>
<td style="text-align: left;">bmi tissue 2</td>
<td style="text-align: left;">hf_allcause</td>
<td style="text-align: right;">177</td>
<td style="text-align: right;">0.5108471</td>
<td style="text-align: right;">0.0363407</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">NA</td>
<td style="text-align: right;">NA</td>
<td style="text-align: right;">302.5116</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">NA</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">NA</td>
<td style="text-align: right;">NA</td>
<td style="text-align: right;">NA</td>
<td style="text-align: right;">NA</td>
<td style="text-align: right;">NA</td>
</tr>
</tbody>
</table>
</div>
<div class="sourceCode cell-code" id="cb10"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb10-1"><a href="#cb10-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plot_mr</span>(h_mvmr, res2)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="index_files/figure-html/unnamed-chunk-7-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
</section>
<section id="eqtl-colocalising-bmi-variants" class="level2">
<h2 class="anchored" data-anchor-id="eqtl-colocalising-bmi-variants">eQTL colocalising BMI variants</h2>
<section id="number-of-colocalising-variants-per-tissue" class="level3">
<h3 class="anchored" data-anchor-id="number-of-colocalising-variants-per-tissue">Number of colocalising variants per tissue</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb11"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb11-1"><a href="#cb11-1" aria-hidden="true" tabindex="-1"></a>coloc_thresh <span class="ot">&lt;-</span> <span class="fl">0.8</span></span>
<span id="cb11-2"><a href="#cb11-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-3"><a href="#cb11-3" aria-hidden="true" tabindex="-1"></a>coloc_results <span class="ot">&lt;-</span> <span class="fu">fread</span>(<span class="fu">file.path</span>(TPMR_DIR, <span class="st">"output"</span>, <span class="st">"tables"</span>, <span class="st">"coloc_results.tsv.gz"</span>))</span>
<span id="cb11-4"><a href="#cb11-4" aria-hidden="true" tabindex="-1"></a>coloc_counts <span class="ot">&lt;-</span> (coloc_results</span>
<span id="cb11-5"><a href="#cb11-5" aria-hidden="true" tabindex="-1"></a>  [locus_trait<span class="sc">==</span><span class="st">"bmi"</span> <span class="sc">&amp;</span> <span class="fu">grepl</span>(<span class="st">"^bmi"</span>, trait.x) <span class="sc">&amp;</span> <span class="sc">!</span><span class="fu">grepl</span>(<span class="st">"^bmi|^hf_"</span>, trait.y) <span class="sc">&amp;</span> <span class="fu">grepl</span>(<span class="st">"protein_coding"</span>, gene_type.y)]</span>
<span id="cb11-6"><a href="#cb11-6" aria-hidden="true" tabindex="-1"></a>  [, <span class="at">num_coloc_genes :=</span> <span class="fu">sum</span>(pph4<span class="sc">&gt;</span>coloc_thresh), <span class="at">by=</span>trait.y]</span>
<span id="cb11-7"><a href="#cb11-7" aria-hidden="true" tabindex="-1"></a>  [, .SD[<span class="fu">which.max</span>(pph4)], <span class="at">by=</span>.(locus_index_rsid, trait.y)]</span>
<span id="cb11-8"><a href="#cb11-8" aria-hidden="true" tabindex="-1"></a>  [, .(<span class="at">num_coloc_genes=</span> num_coloc_genes[<span class="dv">1</span>],</span>
<span id="cb11-9"><a href="#cb11-9" aria-hidden="true" tabindex="-1"></a>       <span class="at">num_coloc_pph4 =</span> <span class="fu">sum</span>(pph4<span class="sc">&gt;</span>coloc_thresh),</span>
<span id="cb11-10"><a href="#cb11-10" aria-hidden="true" tabindex="-1"></a>       <span class="at">mean_pph4      =</span> <span class="fu">mean</span>(pph4),</span>
<span id="cb11-11"><a href="#cb11-11" aria-hidden="true" tabindex="-1"></a>       <span class="at">mean_sample_n  =</span> <span class="fu">mean</span>(sample_n.y)), <span class="at">by=</span><span class="fu">c</span>(<span class="at">tissue=</span><span class="st">"trait.y"</span>)]</span>
<span id="cb11-12"><a href="#cb11-12" aria-hidden="true" tabindex="-1"></a>  [, <span class="at">tissue :=</span> <span class="fu">factor</span>(tissue, <span class="at">levels =</span> tissue[<span class="fu">order</span>(<span class="sc">-</span><span class="st">`</span><span class="at">num_coloc_pph4</span><span class="st">`</span>)])]</span>
<span id="cb11-13"><a href="#cb11-13" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb11-14"><a href="#cb11-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-15"><a href="#cb11-15" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(coloc_counts, <span class="fu">aes</span>(<span class="at">y =</span> tissue, <span class="at">x =</span> <span class="st">`</span><span class="at">num_coloc_pph4</span><span class="st">`</span>)) <span class="sc">+</span></span>
<span id="cb11-16"><a href="#cb11-16" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_classic</span>() <span class="sc">+</span></span>
<span id="cb11-17"><a href="#cb11-17" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_col</span>(<span class="at">fill=</span> <span class="st">"blue"</span>, <span class="at">alpha =</span> <span class="fl">0.3</span>) <span class="sc">+</span></span>
<span id="cb11-18"><a href="#cb11-18" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">subtitle =</span> <span class="fu">paste</span>(<span class="st">"coloc thresh:"</span>, coloc_thresh), </span>
<span id="cb11-19"><a href="#cb11-19" aria-hidden="true" tabindex="-1"></a>       <span class="at">x =</span> <span class="st">"Number of colocalising BMI variants"</span>) <span class="sc">+</span></span>
<span id="cb11-20"><a href="#cb11-20" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">axis.title.y =</span> <span class="fu">element_blank</span>())</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="index_files/figure-html/unnamed-chunk-8-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
</section>
<section id="determinants-of-colocalisation-number" class="level3">
<h3 class="anchored" data-anchor-id="determinants-of-colocalisation-number">Determinants of colocalisation number</h3>
<p>Does this depend on eqtl sample size????</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb12"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb12-1"><a href="#cb12-1" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(coloc_counts, <span class="fu">aes</span>(<span class="at">y =</span> mean_sample_n, <span class="at">x =</span> num_coloc_pph4, <span class="at">color =</span> mean_pph4, <span class="at">size =</span> num_coloc_genes)) <span class="sc">+</span></span>
<span id="cb12-2"><a href="#cb12-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>() <span class="sc">+</span></span>
<span id="cb12-3"><a href="#cb12-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_text_repel</span>(<span class="fu">aes</span>(<span class="at">label =</span> tissue), <span class="at">size =</span> <span class="dv">2</span>, <span class="at">color=</span><span class="st">"black"</span>, <span class="at">max.overlaps =</span> <span class="cn">Inf</span>) <span class="sc">+</span></span>
<span id="cb12-4"><a href="#cb12-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_color_viridis_c</span>(<span class="at">option=</span><span class="st">"mako"</span>) <span class="sc">+</span></span>
<span id="cb12-5"><a href="#cb12-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_classic</span>() <span class="sc">+</span></span>
<span id="cb12-6"><a href="#cb12-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">subtitle =</span> <span class="fu">paste</span>(<span class="st">"coloc thresh:"</span>, coloc_thresh),</span>
<span id="cb12-7"><a href="#cb12-7" aria-hidden="true" tabindex="-1"></a>       <span class="at">color =</span> <span class="st">"Mean PPH4"</span>,</span>
<span id="cb12-8"><a href="#cb12-8" aria-hidden="true" tabindex="-1"></a>       <span class="at">size =</span> <span class="st">"No. coloc genes"</span>,</span>
<span id="cb12-9"><a href="#cb12-9" aria-hidden="true" tabindex="-1"></a>       <span class="at">y =</span> <span class="st">"No. colocalising BMI variants"</span>,</span>
<span id="cb12-10"><a href="#cb12-10" aria-hidden="true" tabindex="-1"></a>       <span class="at">x =</span> <span class="st">"Sample size"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="index_files/figure-html/unnamed-chunk-9-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
</section>
<section id="effect-of-number-of-colocalising-loci-on-mvmr-result---1" class="level3">
<h3 class="anchored" data-anchor-id="effect-of-number-of-colocalising-loci-on-mvmr-result---1">Effect of number of colocalising loci on MVMR result - 1</h3>
<p>Simple analysis: hold constant the number of colocalising variants in tissue 1 (set to median: 73) and hold the PPH4 weighting for colocalising variants at <code>0.9</code> and the for the non-colocalising variants at <code>0.2</code>.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb13"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb13-1"><a href="#cb13-1" aria-hidden="true" tabindex="-1"></a>num_loci <span class="ot">&lt;-</span> <span class="fu">expand.grid</span>(<span class="at">num_t1  =</span> <span class="fu">round</span>(<span class="fu">median</span>(coloc_counts<span class="sc">$</span>num_coloc_pph4)), </span>
<span id="cb13-2"><a href="#cb13-2" aria-hidden="true" tabindex="-1"></a>                        <span class="at">num_t2  =</span> <span class="fu">seq</span>(<span class="fu">min</span>(coloc_counts<span class="sc">$</span>num_coloc_pph4), <span class="fu">max</span>(coloc_counts<span class="sc">$</span>num_coloc_pph4)),</span>
<span id="cb13-3"><a href="#cb13-3" aria-hidden="true" tabindex="-1"></a>                        <span class="at">pph4_hi =</span> <span class="fl">0.9</span>, </span>
<span id="cb13-4"><a href="#cb13-4" aria-hidden="true" tabindex="-1"></a>                        <span class="at">pph4_lo =</span> <span class="fl">0.2</span>) <span class="sc">|&gt;</span> <span class="fu">as.data.table</span>()</span>
<span id="cb13-5"><a href="#cb13-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-6"><a href="#cb13-6" aria-hidden="true" tabindex="-1"></a>unique_num_loci <span class="ot">&lt;-</span> <span class="fu">sort</span>(<span class="fu">unique</span>(<span class="fu">c</span>(num_loci<span class="sc">$</span>num_t1, num_loci<span class="sc">$</span>num_t2)))</span>
<span id="cb13-7"><a href="#cb13-7" aria-hidden="true" tabindex="-1"></a>unique_num_loci <span class="ot">&lt;-</span> <span class="fu">setNames</span>(unique_num_loci, <span class="fu">as.character</span>(unique_num_loci))</span>
<span id="cb13-8"><a href="#cb13-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-9"><a href="#cb13-9" aria-hidden="true" tabindex="-1"></a>loci_indicies <span class="ot">&lt;-</span> <span class="fu">lapply</span>(unique_num_loci, <span class="cf">function</span>(num_l) {</span>
<span id="cb13-10"><a href="#cb13-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">set.seed</span>(<span class="dv">123</span>)</span>
<span id="cb13-11"><a href="#cb13-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">sample</span>(<span class="dv">1</span><span class="sc">:</span><span class="fu">length</span>(h_mvmr<span class="sc">@</span>snps), num_l, <span class="at">replace =</span> <span class="cn">FALSE</span>)</span>
<span id="cb13-12"><a href="#cb13-12" aria-hidden="true" tabindex="-1"></a>})</span>
<span id="cb13-13"><a href="#cb13-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-14"><a href="#cb13-14" aria-hidden="true" tabindex="-1"></a>mr_num_loci <span class="ot">&lt;-</span> num_loci[, {</span>
<span id="cb13-15"><a href="#cb13-15" aria-hidden="true" tabindex="-1"></a>  h_tmp   <span class="ot">&lt;-</span> <span class="fu">copy</span>(h_mvmr)</span>
<span id="cb13-16"><a href="#cb13-16" aria-hidden="true" tabindex="-1"></a>  t1_incides <span class="ot">&lt;-</span> loci_indicies[[<span class="fu">as.character</span>(num_t1)]]</span>
<span id="cb13-17"><a href="#cb13-17" aria-hidden="true" tabindex="-1"></a>  t2_incides <span class="ot">&lt;-</span> loci_indicies[[<span class="fu">as.character</span>(num_t2)]]</span>
<span id="cb13-18"><a href="#cb13-18" aria-hidden="true" tabindex="-1"></a>  h_tmp<span class="sc">@</span>index_snp <span class="ot">&lt;-</span> <span class="fu">ifelse</span>(<span class="dv">1</span><span class="sc">:</span><span class="fu">length</span>(h_tmp<span class="sc">@</span>snps) <span class="sc">%in%</span> <span class="fu">union</span>(t1_incides, t2_incides), <span class="cn">TRUE</span>, <span class="cn">FALSE</span>) </span>
<span id="cb13-19"><a href="#cb13-19" aria-hidden="true" tabindex="-1"></a>  h_tmp<span class="sc">@</span>bx[t1_incides,<span class="dv">1</span>] <span class="ot">&lt;-</span> h_tmp<span class="sc">@</span>bx[t1_incides,<span class="dv">1</span>] <span class="sc">*</span> pph4_hi</span>
<span id="cb13-20"><a href="#cb13-20" aria-hidden="true" tabindex="-1"></a>  h_tmp<span class="sc">@</span>bx[t2_incides,<span class="dv">2</span>] <span class="ot">&lt;-</span> h_tmp<span class="sc">@</span>bx[t2_incides,<span class="dv">2</span>] <span class="sc">*</span> pph4_hi</span>
<span id="cb13-21"><a href="#cb13-21" aria-hidden="true" tabindex="-1"></a>  h_tmp<span class="sc">@</span>bx[<span class="fu">setdiff</span>(t2_incides, t1_incides),<span class="dv">1</span>] <span class="ot">&lt;-</span> h_tmp<span class="sc">@</span>bx[<span class="fu">setdiff</span>(t2_incides, t1_incides),<span class="dv">1</span>] <span class="sc">*</span> pph4_lo</span>
<span id="cb13-22"><a href="#cb13-22" aria-hidden="true" tabindex="-1"></a>  h_tmp<span class="sc">@</span>bx[<span class="fu">setdiff</span>(t1_incides, t2_incides),<span class="dv">2</span>] <span class="ot">&lt;-</span> h_tmp<span class="sc">@</span>bx[<span class="fu">setdiff</span>(t1_incides, t2_incides),<span class="dv">2</span>] <span class="sc">*</span> pph4_lo</span>
<span id="cb13-23"><a href="#cb13-23" aria-hidden="true" tabindex="-1"></a>  h_tmp<span class="sc">@</span>exposure <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"Tissue 1"</span>, <span class="st">"Tissue 2"</span>)</span>
<span id="cb13-24"><a href="#cb13-24" aria-hidden="true" tabindex="-1"></a>  r <span class="ot">&lt;-</span> <span class="fu">run_mr</span>(h_tmp, <span class="at">methods=</span><span class="st">"mr_ivw"</span>)</span>
<span id="cb13-25"><a href="#cb13-25" aria-hidden="true" tabindex="-1"></a>}, by <span class="ot">=</span> .(num_t1, num_t2)]</span>
<span id="cb13-26"><a href="#cb13-26" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-27"><a href="#cb13-27" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(mr_num_loci[, <span class="st">`</span><span class="at">:=</span><span class="st">`</span>(<span class="at">b_lb =</span> b<span class="sc">-</span>b_se<span class="sc">*</span><span class="fl">1.96</span>, </span>
<span id="cb13-28"><a href="#cb13-28" aria-hidden="true" tabindex="-1"></a>                          <span class="at">b_ub =</span> b<span class="sc">+</span>b_se<span class="sc">*</span><span class="fl">1.96</span>)],</span>
<span id="cb13-29"><a href="#cb13-29" aria-hidden="true" tabindex="-1"></a>       <span class="fu">aes</span>(<span class="at">x=</span>num_t2, <span class="at">y=</span>b, <span class="at">ymin=</span>b_lb, <span class="at">ymax=</span>b_ub, <span class="at">color=</span>exposure)) <span class="sc">+</span></span>
<span id="cb13-30"><a href="#cb13-30" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_errorbar</span>(<span class="at">width=</span><span class="fl">0.2</span>, <span class="at">position =</span> <span class="fu">position_dodge</span>(<span class="fl">0.2</span>)) <span class="sc">+</span></span>
<span id="cb13-31"><a href="#cb13-31" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="at">position =</span> <span class="fu">position_dodge</span>(<span class="fl">0.2</span>)) <span class="sc">+</span></span>
<span id="cb13-32"><a href="#cb13-32" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_bw</span>() <span class="sc">+</span></span>
<span id="cb13-33"><a href="#cb13-33" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">y =</span> <span class="st">"MVMR estimate (tpBMI vs HF)"</span>, </span>
<span id="cb13-34"><a href="#cb13-34" aria-hidden="true" tabindex="-1"></a>       <span class="at">x =</span> <span class="st">"Number of tissue 2 loci"</span>, </span>
<span id="cb13-35"><a href="#cb13-35" aria-hidden="true" tabindex="-1"></a>       <span class="at">color =</span> <span class="st">"Tissue"</span>, </span>
<span id="cb13-36"><a href="#cb13-36" aria-hidden="true" tabindex="-1"></a>       <span class="at">subtitle =</span> <span class="fu">paste0</span>(<span class="st">"Number of loci for tissue 1 fixed at "</span>, num_loci<span class="sc">$</span>num_t1[<span class="dv">1</span>]))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="index_files/figure-html/unnamed-chunk-10-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
</section>
<section id="effect-of-number-of-colocalising-loci-on-mvmr-result---2" class="level3">
<h3 class="anchored" data-anchor-id="effect-of-number-of-colocalising-loci-on-mvmr-result---2">Effect of number of colocalising loci on MVMR result - 2</h3>
<p>Now do the same but vary both the number of colocalising variants in tissue 1 and 2 (grid search).</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb14"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb14-1"><a href="#cb14-1" aria-hidden="true" tabindex="-1"></a>num_loci <span class="ot">&lt;-</span> <span class="fu">expand.grid</span>(<span class="at">num_t1  =</span> <span class="fu">seq</span>(<span class="fu">min</span>(coloc_counts<span class="sc">$</span>num_coloc_pph4), <span class="fu">max</span>(coloc_counts<span class="sc">$</span>num_coloc_pph4), <span class="dv">50</span>), </span>
<span id="cb14-2"><a href="#cb14-2" aria-hidden="true" tabindex="-1"></a>                        <span class="at">num_t2  =</span> <span class="fu">seq</span>(<span class="fu">min</span>(coloc_counts<span class="sc">$</span>num_coloc_pph4), <span class="fu">max</span>(coloc_counts<span class="sc">$</span>num_coloc_pph4), <span class="dv">50</span>),</span>
<span id="cb14-3"><a href="#cb14-3" aria-hidden="true" tabindex="-1"></a>                        <span class="at">overlap =</span> <span class="fu">seq</span>(<span class="dv">0</span>, <span class="dv">1</span>, <span class="fl">0.2</span>),</span>
<span id="cb14-4"><a href="#cb14-4" aria-hidden="true" tabindex="-1"></a>                        <span class="at">seed    =</span> <span class="dv">2000</span><span class="sc">:</span><span class="dv">2100</span>,</span>
<span id="cb14-5"><a href="#cb14-5" aria-hidden="true" tabindex="-1"></a>                        <span class="at">pph4_hi =</span> <span class="fl">0.9</span>, </span>
<span id="cb14-6"><a href="#cb14-6" aria-hidden="true" tabindex="-1"></a>                        <span class="at">pph4_lo =</span> <span class="fl">0.1</span>) <span class="sc">|&gt;</span> <span class="fu">as.data.table</span>()</span>
<span id="cb14-7"><a href="#cb14-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb14-8"><a href="#cb14-8" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(future)</span>
<span id="cb14-9"><a href="#cb14-9" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(furrr)</span>
<span id="cb14-10"><a href="#cb14-10" aria-hidden="true" tabindex="-1"></a><span class="fu">plan</span>(multisession)</span>
<span id="cb14-11"><a href="#cb14-11" aria-hidden="true" tabindex="-1"></a>mr_num_loci <span class="ot">&lt;-</span> <span class="fu">future_pmap_dfr</span>(num_loci, <span class="cf">function</span>(num_t1, num_t2, overlap, seed, pph4_hi, pph4_lo) {</span>
<span id="cb14-12"><a href="#cb14-12" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb14-13"><a href="#cb14-13" aria-hidden="true" tabindex="-1"></a>  h_tmp   <span class="ot">&lt;-</span> <span class="fu">copy</span>(h_mvmr)</span>
<span id="cb14-14"><a href="#cb14-14" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb14-15"><a href="#cb14-15" aria-hidden="true" tabindex="-1"></a>  <span class="co"># randomly sort the BMI SNP indices</span></span>
<span id="cb14-16"><a href="#cb14-16" aria-hidden="true" tabindex="-1"></a>  <span class="fu">set.seed</span>(seed)</span>
<span id="cb14-17"><a href="#cb14-17" aria-hidden="true" tabindex="-1"></a>  sram_idx <span class="ot">&lt;-</span> <span class="fu">sample.int</span>(<span class="fu">length</span>(h_mvmr<span class="sc">@</span>snps))</span>
<span id="cb14-18"><a href="#cb14-18" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb14-19"><a href="#cb14-19" aria-hidden="true" tabindex="-1"></a>  <span class="co"># indicies of the variants to use</span></span>
<span id="cb14-20"><a href="#cb14-20" aria-hidden="true" tabindex="-1"></a>  overlap_n <span class="ot">&lt;-</span> <span class="fu">round</span>(<span class="fu">min</span>(num_t1, num_t2) <span class="sc">*</span> overlap)</span>
<span id="cb14-21"><a href="#cb14-21" aria-hidden="true" tabindex="-1"></a>  t1i       <span class="ot">&lt;-</span> sram_idx[<span class="dv">1</span><span class="sc">:</span>num_t1]</span>
<span id="cb14-22"><a href="#cb14-22" aria-hidden="true" tabindex="-1"></a>  t2i       <span class="ot">&lt;-</span> sram_idx[(num_t1<span class="sc">-</span>overlap_n<span class="sc">+</span><span class="dv">1</span>)<span class="sc">:</span>(num_t2 <span class="sc">+</span> num_t1<span class="sc">-</span>overlap_n)]</span>
<span id="cb14-23"><a href="#cb14-23" aria-hidden="true" tabindex="-1"></a>  t12i      <span class="ot">&lt;-</span> <span class="fu">intersect</span>(t1i, t2i)  <span class="co"># both tissues</span></span>
<span id="cb14-24"><a href="#cb14-24" aria-hidden="true" tabindex="-1"></a>  ut1i      <span class="ot">&lt;-</span> <span class="fu">setdiff</span>(t1i, t12i)   <span class="co"># only tissue 1</span></span>
<span id="cb14-25"><a href="#cb14-25" aria-hidden="true" tabindex="-1"></a>  ut2i      <span class="ot">&lt;-</span> <span class="fu">setdiff</span>(t2i, t12i)   <span class="co"># only tissue 2</span></span>
<span id="cb14-26"><a href="#cb14-26" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb14-27"><a href="#cb14-27" aria-hidden="true" tabindex="-1"></a>  <span class="co"># which snps go into the instrument</span></span>
<span id="cb14-28"><a href="#cb14-28" aria-hidden="true" tabindex="-1"></a>  h_tmp<span class="sc">@</span>index_snp <span class="ot">&lt;-</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">length</span>(h_mvmr<span class="sc">@</span>snps) <span class="sc">%in%</span> <span class="fu">c</span>(t12i, ut1i, ut2i)</span>
<span id="cb14-29"><a href="#cb14-29" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb14-30"><a href="#cb14-30" aria-hidden="true" tabindex="-1"></a>  <span class="co"># weight tissue 1 betas</span></span>
<span id="cb14-31"><a href="#cb14-31" aria-hidden="true" tabindex="-1"></a>  h_tmp<span class="sc">@</span>bx[ut2i,<span class="dv">1</span>] <span class="ot">&lt;-</span> h_tmp<span class="sc">@</span>bx[ut2i,<span class="dv">1</span>] <span class="sc">*</span> pph4_lo </span>
<span id="cb14-32"><a href="#cb14-32" aria-hidden="true" tabindex="-1"></a>  h_tmp<span class="sc">@</span>bx[t1i,<span class="dv">1</span>]  <span class="ot">&lt;-</span> h_tmp<span class="sc">@</span>bx[t1i,<span class="dv">1</span>] <span class="sc">*</span> pph4_hi </span>
<span id="cb14-33"><a href="#cb14-33" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb14-34"><a href="#cb14-34" aria-hidden="true" tabindex="-1"></a>  <span class="co"># weight tissue 2 betas</span></span>
<span id="cb14-35"><a href="#cb14-35" aria-hidden="true" tabindex="-1"></a>  h_tmp<span class="sc">@</span>bx[ut1i,<span class="dv">2</span>] <span class="ot">&lt;-</span> h_tmp<span class="sc">@</span>bx[ut1i,<span class="dv">2</span>] <span class="sc">*</span> pph4_lo </span>
<span id="cb14-36"><a href="#cb14-36" aria-hidden="true" tabindex="-1"></a>  h_tmp<span class="sc">@</span>bx[t2i,<span class="dv">2</span>]  <span class="ot">&lt;-</span> h_tmp<span class="sc">@</span>bx[t2i,<span class="dv">2</span>] <span class="sc">*</span> pph4_hi </span>
<span id="cb14-37"><a href="#cb14-37" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb14-38"><a href="#cb14-38" aria-hidden="true" tabindex="-1"></a>  <span class="co"># naming</span></span>
<span id="cb14-39"><a href="#cb14-39" aria-hidden="true" tabindex="-1"></a>  h_tmp<span class="sc">@</span>exposure <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"MV-beta tissue 1"</span>, <span class="st">"MV-beta tissue 2"</span>)</span>
<span id="cb14-40"><a href="#cb14-40" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb14-41"><a href="#cb14-41" aria-hidden="true" tabindex="-1"></a>  <span class="co"># run mr</span></span>
<span id="cb14-42"><a href="#cb14-42" aria-hidden="true" tabindex="-1"></a>  r <span class="ot">&lt;-</span> <span class="fu">run_mr</span>(h_tmp, <span class="at">methods=</span><span class="st">"mr_ivw"</span>)</span>
<span id="cb14-43"><a href="#cb14-43" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb14-44"><a href="#cb14-44" aria-hidden="true" tabindex="-1"></a>  r[, <span class="st">`</span><span class="at">:=</span><span class="st">`</span>(<span class="at">num_t1   =</span> num_t1, </span>
<span id="cb14-45"><a href="#cb14-45" aria-hidden="true" tabindex="-1"></a>           <span class="at">num_t2   =</span> num_t2, </span>
<span id="cb14-46"><a href="#cb14-46" aria-hidden="true" tabindex="-1"></a>           <span class="at">num_diff =</span> num_t1<span class="sc">-</span>num_t2, </span>
<span id="cb14-47"><a href="#cb14-47" aria-hidden="true" tabindex="-1"></a>           <span class="at">overlap  =</span> overlap, </span>
<span id="cb14-48"><a href="#cb14-48" aria-hidden="true" tabindex="-1"></a>           <span class="at">seed     =</span> seed, </span>
<span id="cb14-49"><a href="#cb14-49" aria-hidden="true" tabindex="-1"></a>           <span class="at">pph4_lo  =</span> pph4_lo, </span>
<span id="cb14-50"><a href="#cb14-50" aria-hidden="true" tabindex="-1"></a>           <span class="at">pph4_hi  =</span> pph4_hi)]</span>
<span id="cb14-51"><a href="#cb14-51" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb14-52"><a href="#cb14-52" aria-hidden="true" tabindex="-1"></a>  <span class="fu">return</span>(r)</span>
<span id="cb14-53"><a href="#cb14-53" aria-hidden="true" tabindex="-1"></a>})</span>
<span id="cb14-54"><a href="#cb14-54" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb14-55"><a href="#cb14-55" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb14-56"><a href="#cb14-56" aria-hidden="true" tabindex="-1"></a><span class="co"># ggplot(mr_num_loci, aes(x=as.factor(num_diff), y=b, fill=exposure, color=num_t1)) +</span></span>
<span id="cb14-57"><a href="#cb14-57" aria-hidden="true" tabindex="-1"></a><span class="co">#   geom_hline(yintercept = 0.0, linetype="dashed", color="darkgray") +</span></span>
<span id="cb14-58"><a href="#cb14-58" aria-hidden="true" tabindex="-1"></a><span class="co">#   geom_violin(width=1, position= position_dodge(width = 0.5), color=NA) +</span></span>
<span id="cb14-59"><a href="#cb14-59" aria-hidden="true" tabindex="-1"></a><span class="co">#   theme_classic() +</span></span>
<span id="cb14-60"><a href="#cb14-60" aria-hidden="true" tabindex="-1"></a><span class="co">#   labs(x = "Difference in instrument size (num SNPs, T1-T2)", </span></span>
<span id="cb14-61"><a href="#cb14-61" aria-hidden="true" tabindex="-1"></a><span class="co">#        y = "MVMR estimate", </span></span>
<span id="cb14-62"><a href="#cb14-62" aria-hidden="true" tabindex="-1"></a><span class="co">#        fill = "Tissue") +</span></span>
<span id="cb14-63"><a href="#cb14-63" aria-hidden="true" tabindex="-1"></a><span class="co">#   facet_wrap(~overlap, </span></span>
<span id="cb14-64"><a href="#cb14-64" aria-hidden="true" tabindex="-1"></a><span class="co">#              labeller = labeller(overlap = function(x) paste0("Loci overlap: ", scales::percent(as.numeric(x)))))</span></span>
<span id="cb14-65"><a href="#cb14-65" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb14-66"><a href="#cb14-66" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb14-67"><a href="#cb14-67" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(mr_num_loci[, <span class="at">num_diff_lab :=</span> <span class="fu">paste0</span>(<span class="st">"Delta~nSNP[T1-T2]:~ "</span>, num_diff)], </span>
<span id="cb14-68"><a href="#cb14-68" aria-hidden="true" tabindex="-1"></a>       <span class="fu">aes</span>(<span class="at">x=</span><span class="fu">as.factor</span>(overlap), <span class="at">y=</span>b, <span class="at">fill=</span>exposure, <span class="at">color=</span>num_t1)) <span class="sc">+</span></span>
<span id="cb14-69"><a href="#cb14-69" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_hline</span>(<span class="at">yintercept =</span> <span class="fl">0.0</span>, <span class="at">linetype=</span><span class="st">"dashed"</span>, <span class="at">color=</span><span class="st">"darkgray"</span>) <span class="sc">+</span></span>
<span id="cb14-70"><a href="#cb14-70" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_violin</span>(<span class="at">width=</span><span class="dv">1</span>, <span class="at">position=</span> <span class="fu">position_dodge</span>(<span class="at">width =</span> <span class="fl">0.5</span>), <span class="at">color=</span><span class="cn">NA</span>) <span class="sc">+</span></span>
<span id="cb14-71"><a href="#cb14-71" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_x_discrete</span>(<span class="at">labels =</span> <span class="cf">function</span>(x) scales<span class="sc">::</span><span class="fu">percent</span>(<span class="fu">as.numeric</span>(x))) <span class="sc">+</span></span>
<span id="cb14-72"><a href="#cb14-72" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_classic</span>() <span class="sc">+</span></span>
<span id="cb14-73"><a href="#cb14-73" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">legend.position =</span> <span class="st">"top"</span>) <span class="sc">+</span></span>
<span id="cb14-74"><a href="#cb14-74" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">x =</span> <span class="st">"Instrument SNP overlap (% of smaller instrument)"</span>, </span>
<span id="cb14-75"><a href="#cb14-75" aria-hidden="true" tabindex="-1"></a>       <span class="at">y =</span> <span class="st">"MVMR estimate"</span>, </span>
<span id="cb14-76"><a href="#cb14-76" aria-hidden="true" tabindex="-1"></a>       <span class="at">fill =</span> <span class="st">"Tissue"</span>) <span class="sc">+</span></span>
<span id="cb14-77"><a href="#cb14-77" aria-hidden="true" tabindex="-1"></a>  <span class="fu">facet_wrap</span>(<span class="sc">~</span>num_diff_lab, </span>
<span id="cb14-78"><a href="#cb14-78" aria-hidden="true" tabindex="-1"></a>             <span class="at">nrow =</span> <span class="dv">1</span>,</span>
<span id="cb14-79"><a href="#cb14-79" aria-hidden="true" tabindex="-1"></a>             <span class="at">labeller =</span> label_parsed)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="index_files/figure-html/unnamed-chunk-11-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
</section>
</section>
<section id="bmieqtl-pph4-distributions-by-tissue" class="level2">
<h2 class="anchored" data-anchor-id="bmieqtl-pph4-distributions-by-tissue">BMI:eQTL PPH4 distributions by tissue</h2>
<p>Now lets look at the actual distributions of the PPH4 values in each tissue, since they are not fixed like in the above analyses. Take the maximum PPH4 from genes in close proximity to each variant, in each tissue and plot the distributions - these are the weights that would be used in the tissue-partitioned MR procedure.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb15"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb15-1"><a href="#cb15-1" aria-hidden="true" tabindex="-1"></a>num_tissues <span class="ot">&lt;-</span> <span class="dv">6</span></span>
<span id="cb15-2"><a href="#cb15-2" aria-hidden="true" tabindex="-1"></a>tissues_subset <span class="ot">&lt;-</span> <span class="fu">levels</span>(coloc_counts<span class="sc">$</span>tissue)[<span class="fu">as.integer</span>(<span class="fu">seq</span>(<span class="dv">1</span>, <span class="fu">length</span>(<span class="fu">levels</span>(coloc_counts<span class="sc">$</span>tissue)), <span class="at">length.out=</span>num_tissues))] </span>
<span id="cb15-3"><a href="#cb15-3" aria-hidden="true" tabindex="-1"></a>coloc_subset <span class="ot">&lt;-</span> coloc_results[locus_trait<span class="sc">==</span><span class="st">"bmi"</span> <span class="sc">&amp;</span> trait.x<span class="sc">==</span><span class="st">"bmi"</span> <span class="sc">&amp;</span> (trait.y <span class="sc">%in%</span> tissues_subset), .SD[<span class="fu">which.max</span>(pph4)], by<span class="ot">=</span>.(locus_index_rsid,trait.y)]</span>
<span id="cb15-4"><a href="#cb15-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-5"><a href="#cb15-5" aria-hidden="true" tabindex="-1"></a>density <span class="ot">&lt;-</span> coloc_subset[, {</span>
<span id="cb15-6"><a href="#cb15-6" aria-hidden="true" tabindex="-1"></a>  den <span class="ot">&lt;-</span> <span class="fu">density</span>(pph4, <span class="at">n=</span><span class="dv">1000</span>)</span>
<span id="cb15-7"><a href="#cb15-7" aria-hidden="true" tabindex="-1"></a>  .(<span class="at">pph4 =</span> scales<span class="sc">::</span><span class="fu">rescale</span>(den<span class="sc">$</span>x, <span class="dv">0</span><span class="sc">:</span><span class="dv">1</span>), <span class="at">prob =</span> den<span class="sc">$</span>y)</span>
<span id="cb15-8"><a href="#cb15-8" aria-hidden="true" tabindex="-1"></a>}, by <span class="ot">=</span> <span class="fu">c</span>(<span class="at">tissue=</span><span class="st">"trait.y"</span>)][, tissue <span class="sc">:</span><span class="er">=</span> <span class="fu">factor</span>(tissue, <span class="at">levels =</span> <span class="fu">levels</span>(coloc_counts<span class="sc">$</span>tissue))]</span>
<span id="cb15-9"><a href="#cb15-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-10"><a href="#cb15-10" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(density, <span class="fu">aes</span>(<span class="at">x=</span>pph4, <span class="at">y=</span>prob, <span class="at">color=</span>tissue)) <span class="sc">+</span></span>
<span id="cb15-11"><a href="#cb15-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_classic</span>() <span class="sc">+</span></span>
<span id="cb15-12"><a href="#cb15-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_line</span>() <span class="sc">+</span></span>
<span id="cb15-13"><a href="#cb15-13" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">y =</span> <span class="st">"Density"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="index_files/figure-html/unnamed-chunk-12-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
</section>
<section id="use-of-these-distributions-in-simulations" class="level2">
<h2 class="anchored" data-anchor-id="use-of-these-distributions-in-simulations">Use of these distributions in simulations</h2>
<p>If we want ‘real-world’ PPH4 weights for use in simulations we can just sample from these density distributions. Here is a quick check to show that sampling from them replicates the shape.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb16"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb16-1"><a href="#cb16-1" aria-hidden="true" tabindex="-1"></a>tissues <span class="ot">&lt;-</span> <span class="fu">expand.grid</span>(<span class="at">tissue =</span> <span class="fu">unique</span>(density<span class="sc">$</span>tissue), <span class="at">stringsAsFactors =</span> F) <span class="sc">|&gt;</span> <span class="fu">as.data.table</span>()</span>
<span id="cb16-2"><a href="#cb16-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-3"><a href="#cb16-3" aria-hidden="true" tabindex="-1"></a><span class="fu">set.seed</span>(<span class="dv">123</span>)</span>
<span id="cb16-4"><a href="#cb16-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-5"><a href="#cb16-5" aria-hidden="true" tabindex="-1"></a>res_sim_density <span class="ot">&lt;-</span> tissues[, {</span>
<span id="cb16-6"><a href="#cb16-6" aria-hidden="true" tabindex="-1"></a>  .(<span class="at">sim_pph4 =</span> <span class="fu">sample</span>(density[tissue<span class="sc">==</span>.BY<span class="sc">$</span>tissue, pph4], <span class="fu">round</span>(<span class="fu">length</span>(h_mvmr<span class="sc">@</span>bx[,<span class="dv">1</span>])<span class="sc">/</span><span class="dv">2</span>), <span class="at">prob =</span> density[tissue<span class="sc">==</span>.BY<span class="sc">$</span>tissue, prob], <span class="at">replace =</span> <span class="cn">TRUE</span>))</span>
<span id="cb16-7"><a href="#cb16-7" aria-hidden="true" tabindex="-1"></a>}, by <span class="ot">=</span> .(tissue)]</span>
<span id="cb16-8"><a href="#cb16-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-9"><a href="#cb16-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-10"><a href="#cb16-10" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(res_sim_density, <span class="fu">aes</span>(<span class="at">x=</span>sim_pph4, <span class="at">color=</span>tissue)) <span class="sc">+</span></span>
<span id="cb16-11"><a href="#cb16-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_classic</span>() <span class="sc">+</span></span>
<span id="cb16-12"><a href="#cb16-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_density</span>() <span class="sc">+</span></span>
<span id="cb16-13"><a href="#cb16-13" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">y =</span> <span class="st">"Simulated density"</span>, </span>
<span id="cb16-14"><a href="#cb16-14" aria-hidden="true" tabindex="-1"></a>       <span class="at">x =</span> <span class="st">"Simulated PPH4"</span>, </span>
<span id="cb16-15"><a href="#cb16-15" aria-hidden="true" tabindex="-1"></a>       <span class="at">color =</span> <span class="st">"Tissue"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="index_files/figure-html/unnamed-chunk-13-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
</section>
<section id="effect-different-pph4-weights" class="level2">
<h2 class="anchored" data-anchor-id="effect-different-pph4-weights">Effect different PPH4 weights</h2>
<p>Weight the BMI betas by their respective maximum PPH4 value, see how this changes the relationship between the beta for BMI and beta for HF, dependent on which tissue is providing the betas. Tissues with greater or fewer colocalising variants affect push the betas up or down more, respectively. I hold one tissue constant (reference) to more easily appreciate how the other tissue’s colocalisation characteristics affect things.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb17"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb17-1"><a href="#cb17-1" aria-hidden="true" tabindex="-1"></a>ref_tissue   <span class="ot">&lt;-</span> <span class="fu">levels</span>(density<span class="sc">$</span>tissue)[<span class="dv">1</span>]</span>
<span id="cb17-2"><a href="#cb17-2" aria-hidden="true" tabindex="-1"></a>tissue_pairs <span class="ot">&lt;-</span> <span class="fu">expand.grid</span>(<span class="at">t1 =</span> ref_tissue, <span class="at">t2 =</span> <span class="fu">unique</span>(density<span class="sc">$</span>tissue), <span class="at">stringsAsFactors =</span> F) <span class="sc">|&gt;</span> <span class="fu">as.data.table</span>()</span>
<span id="cb17-3"><a href="#cb17-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-4"><a href="#cb17-4" aria-hidden="true" tabindex="-1"></a><span class="fu">set.seed</span>(<span class="dv">123</span>)</span>
<span id="cb17-5"><a href="#cb17-5" aria-hidden="true" tabindex="-1"></a>t1_pph4 <span class="ot">&lt;-</span> <span class="fu">sample</span>(density[tissue<span class="sc">==</span>ref_tissue, pph4], <span class="fu">length</span>(h_mvmr<span class="sc">@</span>bx[,<span class="dv">1</span>]), <span class="at">prob =</span> density[tissue<span class="sc">==</span>ref_tissue, prob], <span class="at">replace =</span> <span class="cn">TRUE</span>)</span>
<span id="cb17-6"><a href="#cb17-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-7"><a href="#cb17-7" aria-hidden="true" tabindex="-1"></a>res_weighted <span class="ot">&lt;-</span> tissue_pairs[, {</span>
<span id="cb17-8"><a href="#cb17-8" aria-hidden="true" tabindex="-1"></a>  h_tmp <span class="ot">&lt;-</span> h_mvmr</span>
<span id="cb17-9"><a href="#cb17-9" aria-hidden="true" tabindex="-1"></a>  t2_pph4 <span class="ot">&lt;-</span> <span class="fu">sample</span>(density[tissue<span class="sc">==</span>.BY<span class="sc">$</span>t2, pph4], <span class="fu">length</span>(h_tmp<span class="sc">@</span>bx[,<span class="dv">1</span>]), <span class="at">prob =</span> density[tissue<span class="sc">==</span>.BY<span class="sc">$</span>t2, prob], <span class="at">replace =</span> <span class="cn">TRUE</span>)</span>
<span id="cb17-10"><a href="#cb17-10" aria-hidden="true" tabindex="-1"></a>  h_tmp<span class="sc">@</span>bx[,<span class="dv">1</span>] <span class="ot">&lt;-</span> h_tmp<span class="sc">@</span>bx[,<span class="dv">1</span>] <span class="sc">*</span> t1_pph4</span>
<span id="cb17-11"><a href="#cb17-11" aria-hidden="true" tabindex="-1"></a>  h_tmp<span class="sc">@</span>bx[,<span class="dv">2</span>] <span class="ot">&lt;-</span> h_tmp<span class="sc">@</span>bx[,<span class="dv">2</span>] <span class="sc">*</span> t2_pph4</span>
<span id="cb17-12"><a href="#cb17-12" aria-hidden="true" tabindex="-1"></a>  h_tmp<span class="sc">@</span>index_snp <span class="ot">&lt;-</span> <span class="fu">ifelse</span>(t1_pph4<span class="sc">&gt;</span>coloc_thresh <span class="sc">|</span> t2_pph4<span class="sc">&gt;</span>coloc_thresh, <span class="cn">TRUE</span>, <span class="cn">FALSE</span>)</span>
<span id="cb17-13"><a href="#cb17-13" aria-hidden="true" tabindex="-1"></a>  .(<span class="at">tissue   =</span> <span class="fu">rep</span>(<span class="fu">c</span>(<span class="fu">paste0</span>(<span class="st">"Ref: "</span>, ref_tissue),<span class="st">"Comparator"</span>), <span class="at">each=</span><span class="fu">sum</span>(h_tmp<span class="sc">@</span>index_snp)), </span>
<span id="cb17-14"><a href="#cb17-14" aria-hidden="true" tabindex="-1"></a>    <span class="at">beta_bmi =</span> <span class="fu">c</span>(h_tmp<span class="sc">@</span>bx[h_tmp<span class="sc">@</span>index_snp,<span class="dv">1</span>], h_tmp<span class="sc">@</span>bx[h_tmp<span class="sc">@</span>index_snp,<span class="dv">2</span>]), </span>
<span id="cb17-15"><a href="#cb17-15" aria-hidden="true" tabindex="-1"></a>    <span class="at">beta_hf  =</span> <span class="fu">rep</span>(h_tmp<span class="sc">@</span>by[h_tmp<span class="sc">@</span>index_snp],<span class="dv">2</span>),</span>
<span id="cb17-16"><a href="#cb17-16" aria-hidden="true" tabindex="-1"></a>    <span class="at">mean_pph4_t1 =</span> <span class="fu">mean</span>(t1_pph4[t1_pph4<span class="sc">&gt;</span>coloc_thresh <span class="sc">|</span> t2_pph4<span class="sc">&gt;</span>coloc_thresh]),</span>
<span id="cb17-17"><a href="#cb17-17" aria-hidden="true" tabindex="-1"></a>    <span class="at">mean_pph4_t2 =</span> <span class="fu">mean</span>(t2_pph4[t1_pph4<span class="sc">&gt;</span>coloc_thresh <span class="sc">|</span> t2_pph4<span class="sc">&gt;</span>coloc_thresh]), </span>
<span id="cb17-18"><a href="#cb17-18" aria-hidden="true" tabindex="-1"></a>    <span class="at">rmse_pph4    =</span> <span class="fu">sqrt</span>(<span class="fu">mean</span>((t1_pph4[t1_pph4<span class="sc">&gt;</span>coloc_thresh <span class="sc">|</span> t2_pph4<span class="sc">&gt;</span>coloc_thresh] <span class="sc">-</span> t2_pph4[t1_pph4<span class="sc">&gt;</span>coloc_thresh <span class="sc">|</span> t2_pph4<span class="sc">&gt;</span>coloc_thresh])<span class="sc">^</span><span class="dv">2</span>))</span>
<span id="cb17-19"><a href="#cb17-19" aria-hidden="true" tabindex="-1"></a>   )</span>
<span id="cb17-20"><a href="#cb17-20" aria-hidden="true" tabindex="-1"></a>}, by <span class="ot">=</span> .(t1,t2)]</span>
<span id="cb17-21"><a href="#cb17-21" aria-hidden="true" tabindex="-1"></a>res_weighted[, t2 <span class="sc">:</span><span class="er">=</span> <span class="fu">factor</span>(t2, <span class="at">levels =</span> <span class="fu">unique</span>(t2[<span class="fu">order</span>(<span class="sc">-</span>mean_pph4_t1)]))]</span>
<span id="cb17-22"><a href="#cb17-22" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-23"><a href="#cb17-23" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(res_weighted, <span class="fu">aes</span>(<span class="at">x=</span>beta_bmi, <span class="at">y=</span>beta_hf, <span class="at">color=</span>tissue)) <span class="sc">+</span></span>
<span id="cb17-24"><a href="#cb17-24" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>() <span class="sc">+</span></span>
<span id="cb17-25"><a href="#cb17-25" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_text</span>(<span class="at">data =</span> <span class="fu">unique</span>(res_weighted[<span class="fu">grepl</span>(<span class="st">"Ref"</span>, tissue)], <span class="at">by=</span><span class="st">"t2"</span>), </span>
<span id="cb17-26"><a href="#cb17-26" aria-hidden="true" tabindex="-1"></a>            <span class="fu">aes</span>(<span class="at">x =</span> <span class="fl">0.0</span>, </span>
<span id="cb17-27"><a href="#cb17-27" aria-hidden="true" tabindex="-1"></a>                <span class="at">y =</span> <span class="sc">-</span><span class="fl">0.04</span>,</span>
<span id="cb17-28"><a href="#cb17-28" aria-hidden="true" tabindex="-1"></a>                <span class="at">label =</span> <span class="fu">sprintf</span>(<span class="st">"Tis mPPH4=%.3f"</span>, mean_pph4_t2)),</span>
<span id="cb17-29"><a href="#cb17-29" aria-hidden="true" tabindex="-1"></a>            <span class="at">color=</span><span class="st">"black"</span>, <span class="at">size=</span><span class="dv">2</span>) <span class="sc">+</span></span>
<span id="cb17-30"><a href="#cb17-30" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_text</span>(<span class="at">data =</span> <span class="fu">unique</span>(res_weighted[<span class="sc">!</span><span class="fu">grepl</span>(<span class="st">"Ref"</span>, tissue)], <span class="at">by=</span><span class="st">"t2"</span>), </span>
<span id="cb17-31"><a href="#cb17-31" aria-hidden="true" tabindex="-1"></a>            <span class="fu">aes</span>(<span class="at">x =</span> <span class="fl">0.0</span>, </span>
<span id="cb17-32"><a href="#cb17-32" aria-hidden="true" tabindex="-1"></a>                <span class="at">y =</span> <span class="sc">-</span><span class="fl">0.07</span>,</span>
<span id="cb17-33"><a href="#cb17-33" aria-hidden="true" tabindex="-1"></a>                <span class="at">label =</span> <span class="fu">sprintf</span>(<span class="st">"Ref mPPH4=%.3f"</span>, mean_pph4_t1)),</span>
<span id="cb17-34"><a href="#cb17-34" aria-hidden="true" tabindex="-1"></a>            <span class="at">color=</span><span class="st">"black"</span>, <span class="at">size=</span><span class="dv">2</span>) <span class="sc">+</span></span>
<span id="cb17-35"><a href="#cb17-35" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_smooth</span>(<span class="at">method=</span><span class="st">"lm"</span>, <span class="at">se=</span>F,<span class="at">fullrange =</span> <span class="cn">TRUE</span>) <span class="sc">+</span></span>
<span id="cb17-36"><a href="#cb17-36" aria-hidden="true" tabindex="-1"></a>  <span class="fu">facet_wrap</span>(<span class="sc">~</span>t2, <span class="at">labeller =</span> <span class="fu">labeller</span>(<span class="at">t2 =</span> <span class="cf">function</span>(x) <span class="fu">gsub</span>(<span class="st">"_"</span>, <span class="st">" "</span>, x))) <span class="sc">+</span></span>
<span id="cb17-37"><a href="#cb17-37" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">subtitle =</span> <span class="fu">paste</span>(<span class="st">"coloc thresh:"</span>, coloc_thresh), </span>
<span id="cb17-38"><a href="#cb17-38" aria-hidden="true" tabindex="-1"></a>       <span class="at">y =</span> <span class="st">"Heart failure beta"</span>, </span>
<span id="cb17-39"><a href="#cb17-39" aria-hidden="true" tabindex="-1"></a>       <span class="at">x =</span> <span class="st">"BMI beta (PPH4-weighted)"</span>, </span>
<span id="cb17-40"><a href="#cb17-40" aria-hidden="true" tabindex="-1"></a>       <span class="at">color =</span> <span class="st">"Tissue"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="index_files/figure-html/unnamed-chunk-14-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
</section>
<section id="ivw-regression-by-different-pph4-weighting-distributions" class="level2">
<h2 class="anchored" data-anchor-id="ivw-regression-by-different-pph4-weighting-distributions">IVW regression by different PPH4 weighting distributions</h2>
<p>Perform MVMR with the weighted BMI instruments. Lets look at holding a poorly colocalising tissue constant compared to the best colocalising tissues.</p>
<section id="function-to-weight-betas-and-run-mr" class="level3">
<h3 class="anchored" data-anchor-id="function-to-weight-betas-and-run-mr">Function to weight betas and run MR</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb18"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb18-1"><a href="#cb18-1" aria-hidden="true" tabindex="-1"></a><span class="co">#' @title TPMR helper function</span></span>
<span id="cb18-2"><a href="#cb18-2" aria-hidden="true" tabindex="-1"></a><span class="co">#' @param mrobj genepi.utils MR object (exposures BMI:BMI, outcome HF)</span></span>
<span id="cb18-3"><a href="#cb18-3" aria-hidden="true" tabindex="-1"></a><span class="co">#' @param density data.table of PPH4 density distributions cols: tissue | pph4 | prob</span></span>
<span id="cb18-4"><a href="#cb18-4" aria-hidden="true" tabindex="-1"></a><span class="co">#' @param ref character vector, tissues in density$tissue</span></span>
<span id="cb18-5"><a href="#cb18-5" aria-hidden="true" tabindex="-1"></a><span class="co">#' @param comp character vector, tissues in density$tissue</span></span>
<span id="cb18-6"><a href="#cb18-6" aria-hidden="true" tabindex="-1"></a><span class="co">#' @param coloc_thresh numeric 0-1</span></span>
<span id="cb18-7"><a href="#cb18-7" aria-hidden="true" tabindex="-1"></a><span class="co">#' @param rep integer, number of replications (random sampling of BMI variants into instrument)</span></span>
<span id="cb18-8"><a href="#cb18-8" aria-hidden="true" tabindex="-1"></a><span class="co">#' @returns data.table of MR results</span></span>
<span id="cb18-9"><a href="#cb18-9" aria-hidden="true" tabindex="-1"></a><span class="co">#'</span></span>
<span id="cb18-10"><a href="#cb18-10" aria-hidden="true" tabindex="-1"></a>tpmr <span class="ot">&lt;-</span> <span class="cf">function</span>(obj, ref, comp, coloc_thresh, <span class="at">rep =</span> <span class="dv">1</span>, <span class="at">density =</span> <span class="cn">NULL</span>, <span class="at">pph4 =</span> <span class="cn">NULL</span>) {</span>
<span id="cb18-11"><a href="#cb18-11" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb18-12"><a href="#cb18-12" aria-hidden="true" tabindex="-1"></a>  <span class="co"># either density or actual PPH4 values</span></span>
<span id="cb18-13"><a href="#cb18-13" aria-hidden="true" tabindex="-1"></a>  <span class="fu">stopifnot</span>(<span class="st">"Provide either denisty distributions or actual PPH4 values"</span> <span class="ot">=</span> </span>
<span id="cb18-14"><a href="#cb18-14" aria-hidden="true" tabindex="-1"></a>              <span class="fu">sum</span>(<span class="fu">sapply</span>(<span class="fu">list</span>(density, pph4), is.null)) <span class="sc">==</span> <span class="dv">1</span>)</span>
<span id="cb18-15"><a href="#cb18-15" aria-hidden="true" tabindex="-1"></a>  <span class="fu">stopifnot</span>(<span class="st">"Replications must be 1 if using actual PPH4 values"</span> <span class="ot">=</span> <span class="fu">ifelse</span>(rep <span class="sc">&gt;</span> <span class="dv">1</span> <span class="sc">&amp;</span> <span class="sc">!</span><span class="fu">is.null</span>(pph4), F, T))</span>
<span id="cb18-16"><a href="#cb18-16" aria-hidden="true" tabindex="-1"></a>              </span>
<span id="cb18-17"><a href="#cb18-17" aria-hidden="true" tabindex="-1"></a>  <span class="co"># run the MR for each of the exposure combinations</span></span>
<span id="cb18-18"><a href="#cb18-18" aria-hidden="true" tabindex="-1"></a>  tissue_pairs <span class="ot">&lt;-</span> <span class="fu">expand.grid</span>(<span class="at">t1 =</span> <span class="fu">as.character</span>(ref), </span>
<span id="cb18-19"><a href="#cb18-19" aria-hidden="true" tabindex="-1"></a>                              <span class="at">t2 =</span> <span class="fu">as.character</span>(comp),</span>
<span id="cb18-20"><a href="#cb18-20" aria-hidden="true" tabindex="-1"></a>                              <span class="at">seed =</span> <span class="dv">2000</span><span class="sc">:</span>(<span class="dv">2000</span><span class="sc">+</span>rep<span class="dv">-1</span>),</span>
<span id="cb18-21"><a href="#cb18-21" aria-hidden="true" tabindex="-1"></a>                              <span class="at">stringsAsFactors =</span> F) <span class="sc">|&gt;</span> <span class="fu">as.data.table</span>()</span>
<span id="cb18-22"><a href="#cb18-22" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb18-23"><a href="#cb18-23" aria-hidden="true" tabindex="-1"></a>  <span class="fu">library</span>(future)</span>
<span id="cb18-24"><a href="#cb18-24" aria-hidden="true" tabindex="-1"></a>  <span class="fu">library</span>(furrr)</span>
<span id="cb18-25"><a href="#cb18-25" aria-hidden="true" tabindex="-1"></a>  <span class="fu">plan</span>(multisession)</span>
<span id="cb18-26"><a href="#cb18-26" aria-hidden="true" tabindex="-1"></a>  mr_weighted <span class="ot">&lt;-</span> <span class="fu">future_pmap_dfr</span>(tissue_pairs, <span class="cf">function</span>(t1, t2, seed) {</span>
<span id="cb18-27"><a href="#cb18-27" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb18-28"><a href="#cb18-28" aria-hidden="true" tabindex="-1"></a>    mrobj <span class="ot">&lt;-</span> <span class="fu">copy</span>(obj)</span>
<span id="cb18-29"><a href="#cb18-29" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb18-30"><a href="#cb18-30" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span> (<span class="sc">!</span><span class="fu">is.null</span>(density)) {</span>
<span id="cb18-31"><a href="#cb18-31" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb18-32"><a href="#cb18-32" aria-hidden="true" tabindex="-1"></a>      <span class="co"># generate the PPH4 values randomly for the variants</span></span>
<span id="cb18-33"><a href="#cb18-33" aria-hidden="true" tabindex="-1"></a>      <span class="fu">set.seed</span>(seed)</span>
<span id="cb18-34"><a href="#cb18-34" aria-hidden="true" tabindex="-1"></a>      t1_pph4 <span class="ot">&lt;-</span> <span class="fu">sample</span>(density[tissue<span class="sc">==</span>t1, pph4], <span class="fu">length</span>(mrobj<span class="sc">@</span>snps), <span class="at">prob =</span> density[tissue<span class="sc">==</span>t1, prob], <span class="at">replace =</span> <span class="cn">TRUE</span>)</span>
<span id="cb18-35"><a href="#cb18-35" aria-hidden="true" tabindex="-1"></a>      <span class="fu">set.seed</span>(seed<span class="sc">+</span><span class="dv">1</span>)</span>
<span id="cb18-36"><a href="#cb18-36" aria-hidden="true" tabindex="-1"></a>      t2_pph4 <span class="ot">&lt;-</span> <span class="fu">sample</span>(density[tissue<span class="sc">==</span>t2, pph4], <span class="fu">length</span>(mrobj<span class="sc">@</span>snps), <span class="at">prob =</span> density[tissue<span class="sc">==</span>t2, prob], <span class="at">replace =</span> <span class="cn">TRUE</span>)</span>
<span id="cb18-37"><a href="#cb18-37" aria-hidden="true" tabindex="-1"></a>     </span>
<span id="cb18-38"><a href="#cb18-38" aria-hidden="true" tabindex="-1"></a>    } <span class="cf">else</span> <span class="cf">if</span>(<span class="sc">!</span><span class="fu">is.null</span>(pph4)) {</span>
<span id="cb18-39"><a href="#cb18-39" aria-hidden="true" tabindex="-1"></a>      </span>
<span id="cb18-40"><a href="#cb18-40" aria-hidden="true" tabindex="-1"></a>      chrpos <span class="ot">&lt;-</span> <span class="fu">data.table</span>(<span class="at">chr =</span> <span class="fu">as.integer</span>(mrobj<span class="sc">@</span>chr), <span class="at">bp =</span> mrobj<span class="sc">@</span>bp, <span class="at">t1 =</span> t1, <span class="at">t2 =</span> t2)</span>
<span id="cb18-41"><a href="#cb18-41" aria-hidden="true" tabindex="-1"></a>      chrpos[pph4, t1.pph4 <span class="sc">:</span><span class="er">=</span> i.pph4, on<span class="ot">=</span><span class="fu">c</span>(<span class="st">"chr"</span><span class="ot">=</span><span class="st">"chr"</span>,<span class="st">"bp"</span><span class="ot">=</span><span class="st">"bp"</span>,<span class="st">"t1"</span><span class="ot">=</span><span class="st">"tissue"</span>)]</span>
<span id="cb18-42"><a href="#cb18-42" aria-hidden="true" tabindex="-1"></a>      chrpos[pph4, t2.pph4 <span class="sc">:</span><span class="er">=</span> i.pph4, on<span class="ot">=</span><span class="fu">c</span>(<span class="st">"chr"</span><span class="ot">=</span><span class="st">"chr"</span>,<span class="st">"bp"</span><span class="ot">=</span><span class="st">"bp"</span>,<span class="st">"t2"</span><span class="ot">=</span><span class="st">"tissue"</span>)]</span>
<span id="cb18-43"><a href="#cb18-43" aria-hidden="true" tabindex="-1"></a>      t1_pph4 <span class="ot">&lt;-</span> chrpos<span class="sc">$</span>t1.pph4</span>
<span id="cb18-44"><a href="#cb18-44" aria-hidden="true" tabindex="-1"></a>      t2_pph4 <span class="ot">&lt;-</span> chrpos<span class="sc">$</span>t2.pph4</span>
<span id="cb18-45"><a href="#cb18-45" aria-hidden="true" tabindex="-1"></a>      t1_pph4[<span class="fu">is.na</span>(t1_pph4)] <span class="ot">&lt;-</span> <span class="dv">0</span></span>
<span id="cb18-46"><a href="#cb18-46" aria-hidden="true" tabindex="-1"></a>      t2_pph4[<span class="fu">is.na</span>(t2_pph4)] <span class="ot">&lt;-</span> <span class="dv">0</span></span>
<span id="cb18-47"><a href="#cb18-47" aria-hidden="true" tabindex="-1"></a>      </span>
<span id="cb18-48"><a href="#cb18-48" aria-hidden="true" tabindex="-1"></a>    }</span>
<span id="cb18-49"><a href="#cb18-49" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb18-50"><a href="#cb18-50" aria-hidden="true" tabindex="-1"></a>    <span class="co"># based on the distribution of PPH4, which SNPs are included</span></span>
<span id="cb18-51"><a href="#cb18-51" aria-hidden="true" tabindex="-1"></a>    include <span class="ot">&lt;-</span> <span class="fu">ifelse</span>(t1_pph4 <span class="sc">&gt;</span> coloc_thresh <span class="sc">|</span> t2_pph4 <span class="sc">&gt;</span> coloc_thresh, <span class="cn">TRUE</span>, <span class="cn">FALSE</span>)</span>
<span id="cb18-52"><a href="#cb18-52" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb18-53"><a href="#cb18-53" aria-hidden="true" tabindex="-1"></a>    <span class="co"># update the MR object, turn off the excluded snps (@index_snp slot)</span></span>
<span id="cb18-54"><a href="#cb18-54" aria-hidden="true" tabindex="-1"></a>    mrobj<span class="sc">@</span>bx[,<span class="dv">1</span>]      <span class="ot">&lt;-</span> mrobj<span class="sc">@</span>bx[,<span class="dv">1</span>] <span class="sc">*</span> t1_pph4</span>
<span id="cb18-55"><a href="#cb18-55" aria-hidden="true" tabindex="-1"></a>    mrobj<span class="sc">@</span>bx[,<span class="dv">2</span>]      <span class="ot">&lt;-</span> mrobj<span class="sc">@</span>bx[,<span class="dv">2</span>] <span class="sc">*</span> t2_pph4</span>
<span id="cb18-56"><a href="#cb18-56" aria-hidden="true" tabindex="-1"></a>    mrobj<span class="sc">@</span>index_snp   <span class="ot">&lt;-</span> include</span>
<span id="cb18-57"><a href="#cb18-57" aria-hidden="true" tabindex="-1"></a>    mrobj<span class="sc">@</span>exposure[<span class="dv">1</span>] <span class="ot">&lt;-</span> <span class="fu">ifelse</span>(<span class="fu">length</span>(ref)<span class="sc">==</span><span class="dv">1</span>, <span class="fu">paste0</span>(<span class="st">"Ref: "</span>, ref_tissue), <span class="st">"Tissue 1"</span>)</span>
<span id="cb18-58"><a href="#cb18-58" aria-hidden="true" tabindex="-1"></a>    mrobj<span class="sc">@</span>exposure[<span class="dv">2</span>] <span class="ot">&lt;-</span> <span class="fu">ifelse</span>(<span class="fu">length</span>(ref)<span class="sc">==</span><span class="dv">1</span>, <span class="st">"Comparator"</span>, <span class="st">"Tissue 2"</span>)</span>
<span id="cb18-59"><a href="#cb18-59" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb18-60"><a href="#cb18-60" aria-hidden="true" tabindex="-1"></a>    <span class="co"># run the MR</span></span>
<span id="cb18-61"><a href="#cb18-61" aria-hidden="true" tabindex="-1"></a>    r <span class="ot">&lt;-</span> <span class="fu">run_mr</span>(mrobj, <span class="at">methods=</span><span class="st">"mr_ivw"</span>)</span>
<span id="cb18-62"><a href="#cb18-62" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb18-63"><a href="#cb18-63" aria-hidden="true" tabindex="-1"></a>    <span class="co"># add info about the instrument used</span></span>
<span id="cb18-64"><a href="#cb18-64" aria-hidden="true" tabindex="-1"></a>    r[, <span class="st">`</span><span class="at">:=</span><span class="st">`</span>(<span class="at">t1             =</span> t1, </span>
<span id="cb18-65"><a href="#cb18-65" aria-hidden="true" tabindex="-1"></a>             <span class="at">t2             =</span> t2,</span>
<span id="cb18-66"><a href="#cb18-66" aria-hidden="true" tabindex="-1"></a>             <span class="at">seed           =</span> seed,</span>
<span id="cb18-67"><a href="#cb18-67" aria-hidden="true" tabindex="-1"></a>             <span class="at">n_sig_snps     =</span> <span class="fu">c</span>(<span class="fu">sum</span>(t1_pph4 <span class="sc">&gt;</span> coloc_thresh), <span class="fu">sum</span>(t2_pph4 <span class="sc">&gt;</span> coloc_thresh)), </span>
<span id="cb18-68"><a href="#cb18-68" aria-hidden="true" tabindex="-1"></a>             <span class="at">n_overlap_snps =</span> <span class="fu">sum</span>(t1_pph4 <span class="sc">&gt;</span> coloc_thresh <span class="sc">&amp;</span> t2_pph4 <span class="sc">&gt;</span> coloc_thresh),</span>
<span id="cb18-69"><a href="#cb18-69" aria-hidden="true" tabindex="-1"></a>             <span class="at">pct_t1_overlap =</span> <span class="fu">sum</span>(t1_pph4 <span class="sc">&gt;</span> coloc_thresh <span class="sc">&amp;</span> t2_pph4 <span class="sc">&gt;</span> coloc_thresh) <span class="sc">/</span> <span class="fu">length</span>(t1_pph4),</span>
<span id="cb18-70"><a href="#cb18-70" aria-hidden="true" tabindex="-1"></a>             <span class="at">mean_pph4_t1   =</span> <span class="fu">mean</span>(t1_pph4[include]),</span>
<span id="cb18-71"><a href="#cb18-71" aria-hidden="true" tabindex="-1"></a>             <span class="at">mean_pph4_t2   =</span> <span class="fu">mean</span>(t2_pph4[include]),</span>
<span id="cb18-72"><a href="#cb18-72" aria-hidden="true" tabindex="-1"></a>             <span class="at">rmse_pph4      =</span> <span class="fu">sqrt</span>(<span class="fu">mean</span>((t1_pph4[include] <span class="sc">-</span> t2_pph4[include])<span class="sc">^</span><span class="dv">2</span>)), </span>
<span id="cb18-73"><a href="#cb18-73" aria-hidden="true" tabindex="-1"></a>             <span class="at">rmse_wbeta     =</span> <span class="fu">sqrt</span>(<span class="fu">mean</span>((mrobj<span class="sc">@</span>bx[,<span class="dv">1</span>] <span class="sc">-</span> mrobj<span class="sc">@</span>bx[,<span class="dv">2</span>])<span class="sc">^</span><span class="dv">2</span>)))]</span>
<span id="cb18-74"><a href="#cb18-74" aria-hidden="true" tabindex="-1"></a>    r[, <span class="fu">c</span>(<span class="st">"overdispersion"</span>, <span class="st">"n_pc"</span>, <span class="st">"n_hunted"</span>, <span class="st">"slopehunter_pi"</span>, <span class="st">"slopehunter_ent"</span>) <span class="sc">:</span><span class="er">=</span> <span class="cn">NULL</span>]</span>
<span id="cb18-75"><a href="#cb18-75" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb18-76"><a href="#cb18-76" aria-hidden="true" tabindex="-1"></a>  })</span>
<span id="cb18-77"><a href="#cb18-77" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb18-78"><a href="#cb18-78" aria-hidden="true" tabindex="-1"></a>  <span class="fu">return</span>(mr_weighted)</span>
<span id="cb18-79"><a href="#cb18-79" aria-hidden="true" tabindex="-1"></a>}</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="different-pph4-distributions-against-each-other" class="level3">
<h3 class="anchored" data-anchor-id="different-pph4-distributions-against-each-other">Different PPH4 distributions against each other</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb19"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb19-1"><a href="#cb19-1" aria-hidden="true" tabindex="-1"></a><span class="co"># run TPMR</span></span>
<span id="cb19-2"><a href="#cb19-2" aria-hidden="true" tabindex="-1"></a>mr_weighted <span class="ot">&lt;-</span> <span class="fu">tpmr</span>(<span class="at">obj=</span>h_mvmr, <span class="at">density=</span>density, </span>
<span id="cb19-3"><a href="#cb19-3" aria-hidden="true" tabindex="-1"></a>                    <span class="at">ref=</span> <span class="fu">unique</span>(density<span class="sc">$</span>tissue), </span>
<span id="cb19-4"><a href="#cb19-4" aria-hidden="true" tabindex="-1"></a>                    <span class="at">comp=</span><span class="fu">unique</span>(density<span class="sc">$</span>tissue), </span>
<span id="cb19-5"><a href="#cb19-5" aria-hidden="true" tabindex="-1"></a>                    <span class="at">coloc_thresh=</span><span class="fl">0.8</span>, </span>
<span id="cb19-6"><a href="#cb19-6" aria-hidden="true" tabindex="-1"></a>                    <span class="at">rep =</span> <span class="dv">100</span>)</span>
<span id="cb19-7"><a href="#cb19-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-8"><a href="#cb19-8" aria-hidden="true" tabindex="-1"></a><span class="co"># reorder for plotting </span></span>
<span id="cb19-9"><a href="#cb19-9" aria-hidden="true" tabindex="-1"></a>mr_weighted[, mean_pph4_t1_lab <span class="sc">:</span><span class="er">=</span> <span class="fu">round</span>(<span class="fu">mean</span>(mean_pph4_t1),<span class="dv">2</span>), by<span class="ot">=</span>.(t1)]</span>
<span id="cb19-10"><a href="#cb19-10" aria-hidden="true" tabindex="-1"></a>mr_weighted[, mean_pph4_t2_lab <span class="sc">:</span><span class="er">=</span> <span class="fu">round</span>(<span class="fu">mean</span>(mean_pph4_t2),<span class="dv">2</span>), by<span class="ot">=</span>.(t2)]</span>
<span id="cb19-11"><a href="#cb19-11" aria-hidden="true" tabindex="-1"></a>mr_weighted[, mean_n_sig_snps <span class="sc">:</span><span class="er">=</span> <span class="fu">round</span>(<span class="fu">mean</span>(mean_pph4_t2),<span class="dv">0</span>), by<span class="ot">=</span>.(t1,t2)]</span>
<span id="cb19-12"><a href="#cb19-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-13"><a href="#cb19-13" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(mr_weighted, </span>
<span id="cb19-14"><a href="#cb19-14" aria-hidden="true" tabindex="-1"></a>       <span class="fu">aes</span>(<span class="at">x=</span><span class="fu">as.factor</span>(mean_pph4_t2_lab), <span class="at">y=</span>b, <span class="at">color=</span>exposure)) <span class="sc">+</span></span>
<span id="cb19-15"><a href="#cb19-15" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_boxplot</span>() <span class="sc">+</span></span>
<span id="cb19-16"><a href="#cb19-16" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_bw</span>() <span class="sc">+</span></span>
<span id="cb19-17"><a href="#cb19-17" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">y =</span> <span class="st">"MVMR estimate"</span>, <span class="at">x =</span> <span class="st">"T2 mean PPH4"</span>, <span class="at">color =</span> <span class="st">"Tissue"</span>) <span class="sc">+</span></span>
<span id="cb19-18"><a href="#cb19-18" aria-hidden="true" tabindex="-1"></a>  <span class="fu">facet_wrap</span>(<span class="sc">~</span> <span class="fu">as.factor</span>(mean_pph4_t1_lab), </span>
<span id="cb19-19"><a href="#cb19-19" aria-hidden="true" tabindex="-1"></a>             <span class="at">labeller =</span> <span class="fu">labeller</span>(<span class="st">`</span><span class="at">as.factor(mean_pph4_t1_lab)</span><span class="st">`</span><span class="ot">=</span><span class="cf">function</span>(x) <span class="fu">paste0</span>(<span class="st">"T1 mean PPH4: "</span>,x)))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="index_files/figure-html/unnamed-chunk-16-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="sourceCode cell-code" id="cb20"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb20-1"><a href="#cb20-1" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(mr_weighted[, .(<span class="at">b=</span><span class="fu">mean</span>(b)), <span class="at">by=</span>.(t1,t2,exposure)][, .(<span class="at">b_diff=</span>b[<span class="dv">1</span>]<span class="sc">-</span>b[<span class="dv">2</span>]), <span class="at">by=</span>.(t1,t2)], </span>
<span id="cb20-2"><a href="#cb20-2" aria-hidden="true" tabindex="-1"></a>       <span class="fu">aes</span>(<span class="at">y=</span>t1, <span class="at">x=</span>t2, <span class="at">fill=</span>b_diff)) <span class="sc">+</span></span>
<span id="cb20-3"><a href="#cb20-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_tile</span>() <span class="sc">+</span></span>
<span id="cb20-4"><a href="#cb20-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_fill_gradient2</span>() <span class="sc">+</span></span>
<span id="cb20-5"><a href="#cb20-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">axis.text.x  =</span> <span class="fu">element_text</span>(<span class="at">angle=</span><span class="dv">45</span>, <span class="at">hjust=</span><span class="dv">1</span>)) <span class="sc">+</span></span>
<span id="cb20-6"><a href="#cb20-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">y =</span> <span class="st">"Tissue 1"</span>, <span class="at">x =</span> <span class="st">"Tissue 2"</span>, <span class="at">fill =</span> <span class="st">"T1-T2 beta diff"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="index_files/figure-html/unnamed-chunk-16-2.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
</section>
</section>
<section id="all-tissue-pairs-mr-actual-pph4-values" class="level2">
<h2 class="anchored" data-anchor-id="all-tissue-pairs-mr-actual-pph4-values">All tissue pairs MR (actual PPH4 values)</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb21"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb21-1"><a href="#cb21-1" aria-hidden="true" tabindex="-1"></a>tissue_pph4_vals <span class="ot">&lt;-</span> coloc_results[locus_trait<span class="sc">==</span><span class="st">"bmi"</span> <span class="sc">&amp;</span> </span>
<span id="cb21-2"><a href="#cb21-2" aria-hidden="true" tabindex="-1"></a>                                    trait.x<span class="sc">==</span><span class="st">"bmi"</span> <span class="sc">&amp;</span> </span>
<span id="cb21-3"><a href="#cb21-3" aria-hidden="true" tabindex="-1"></a>                                    gene_type.y<span class="sc">==</span><span class="st">"protein_coding"</span>, </span>
<span id="cb21-4"><a href="#cb21-4" aria-hidden="true" tabindex="-1"></a>                                  .SD[<span class="fu">which.max</span>(pph4)], </span>
<span id="cb21-5"><a href="#cb21-5" aria-hidden="true" tabindex="-1"></a>                                  by<span class="ot">=</span>.(locus_index_rsid, <span class="at">tissue=</span>trait.y), </span>
<span id="cb21-6"><a href="#cb21-6" aria-hidden="true" tabindex="-1"></a>                                  .SDcols<span class="ot">=</span><span class="fu">c</span>(<span class="st">"chr"</span>, <span class="st">"bp"</span>, <span class="st">"pph4"</span>)]</span>
<span id="cb21-7"><a href="#cb21-7" aria-hidden="true" tabindex="-1"></a>tissue_pph4_vals <span class="ot">&lt;-</span> tidyr<span class="sc">::</span><span class="fu">complete</span>(tissue_pph4_vals, locus_index_rsid, tissue) <span class="sc">|&gt;</span> <span class="fu">as.data.table</span>()</span>
<span id="cb21-8"><a href="#cb21-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb21-9"><a href="#cb21-9" aria-hidden="true" tabindex="-1"></a>mr_actual <span class="ot">&lt;-</span> <span class="fu">tpmr</span>(<span class="at">obj=</span>h_mvmr, <span class="at">pph4=</span>tissue_pph4_vals, </span>
<span id="cb21-10"><a href="#cb21-10" aria-hidden="true" tabindex="-1"></a>                  <span class="at">ref=</span> <span class="fu">unique</span>(tissue_pph4_vals<span class="sc">$</span>tissue), </span>
<span id="cb21-11"><a href="#cb21-11" aria-hidden="true" tabindex="-1"></a>                  <span class="at">comp=</span><span class="fu">unique</span>(tissue_pph4_vals<span class="sc">$</span>tissue), </span>
<span id="cb21-12"><a href="#cb21-12" aria-hidden="true" tabindex="-1"></a>                  <span class="at">coloc_thresh=</span><span class="fl">0.8</span>)</span>
<span id="cb21-13"><a href="#cb21-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb21-14"><a href="#cb21-14" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>((mr_actual</span>
<span id="cb21-15"><a href="#cb21-15" aria-hidden="true" tabindex="-1"></a>        [coloc_counts, <span class="at">t1_n :=</span> i.mean_sample_n, <span class="at">on=</span><span class="fu">c</span>(<span class="st">"t1"</span><span class="ot">=</span><span class="st">"tissue"</span>)]</span>
<span id="cb21-16"><a href="#cb21-16" aria-hidden="true" tabindex="-1"></a>        [coloc_counts, <span class="at">t2_n :=</span> i.mean_sample_n, <span class="at">on=</span><span class="fu">c</span>(<span class="st">"t2"</span><span class="ot">=</span><span class="st">"tissue"</span>)]</span>
<span id="cb21-17"><a href="#cb21-17" aria-hidden="true" tabindex="-1"></a>        [, <span class="st">`</span><span class="at">:=</span><span class="st">`</span>(<span class="at">b_diff     =</span> b[exposure<span class="sc">==</span><span class="st">"Tissue 1"</span>]<span class="sc">-</span>b[exposure<span class="sc">==</span><span class="st">"Tissue 2"</span>],</span>
<span id="cb21-18"><a href="#cb21-18" aria-hidden="true" tabindex="-1"></a>                <span class="at">snp_diff   =</span> n_sig_snps[exposure<span class="sc">==</span><span class="st">"Tissue 1"</span>]<span class="sc">-</span>n_sig_snps[exposure<span class="sc">==</span><span class="st">"Tissue 2"</span>],</span>
<span id="cb21-19"><a href="#cb21-19" aria-hidden="true" tabindex="-1"></a>                <span class="at">pph4_diff  =</span> mean_pph4_t1<span class="sc">-</span>mean_pph4_t2, </span>
<span id="cb21-20"><a href="#cb21-20" aria-hidden="true" tabindex="-1"></a>                <span class="at">nsamp_diff =</span> t1_n <span class="sc">-</span> t2_n), <span class="at">by=</span>.(t1,t2)]</span>
<span id="cb21-21"><a href="#cb21-21" aria-hidden="true" tabindex="-1"></a>        [, <span class="st">`</span><span class="at">:=</span><span class="st">`</span>(<span class="at">t1 =</span> <span class="fu">factor</span>(t1, <span class="at">levels =</span> <span class="fu">unique</span>(t1[<span class="fu">order</span>(<span class="sc">-</span>b_diff)])),</span>
<span id="cb21-22"><a href="#cb21-22" aria-hidden="true" tabindex="-1"></a>                <span class="at">t2 =</span> <span class="fu">factor</span>(t2, <span class="at">levels =</span> <span class="fu">unique</span>(t1[<span class="fu">order</span>(<span class="sc">-</span>b_diff)])))]),</span>
<span id="cb21-23"><a href="#cb21-23" aria-hidden="true" tabindex="-1"></a>       <span class="fu">aes</span>(<span class="at">y=</span>t1, <span class="at">x=</span>t2, <span class="at">fill=</span>b_diff)) <span class="sc">+</span></span>
<span id="cb21-24"><a href="#cb21-24" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_tile</span>() <span class="sc">+</span></span>
<span id="cb21-25"><a href="#cb21-25" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_fill_gradient2</span>() <span class="sc">+</span></span>
<span id="cb21-26"><a href="#cb21-26" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">axis.text.x  =</span> <span class="fu">element_text</span>(<span class="at">angle=</span><span class="dv">45</span>, <span class="at">hjust=</span><span class="dv">1</span>)) <span class="sc">+</span></span>
<span id="cb21-27"><a href="#cb21-27" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">y =</span> <span class="st">"Tissue 1"</span>, <span class="at">x =</span> <span class="st">"Tissue 2"</span>, <span class="at">fill =</span> <span class="st">"T1-T2 beta diff"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="index_files/figure-html/unnamed-chunk-17-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
<section id="mvmr-estimates-depend-on-difference-in-instrument-size-which-also-depends-on-difference-in-sample-size" class="level3">
<h3 class="anchored" data-anchor-id="mvmr-estimates-depend-on-difference-in-instrument-size-which-also-depends-on-difference-in-sample-size">MVMR estimates depend on difference in instrument size (which also depends on difference in sample size)</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb22"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb22-1"><a href="#cb22-1" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>((mr_actual</span>
<span id="cb22-2"><a href="#cb22-2" aria-hidden="true" tabindex="-1"></a>        [coloc_counts, <span class="at">t1_n :=</span> i.mean_sample_n, <span class="at">on=</span><span class="fu">c</span>(<span class="st">"t1"</span><span class="ot">=</span><span class="st">"tissue"</span>)]</span>
<span id="cb22-3"><a href="#cb22-3" aria-hidden="true" tabindex="-1"></a>        [coloc_counts, <span class="at">t2_n :=</span> i.mean_sample_n, <span class="at">on=</span><span class="fu">c</span>(<span class="st">"t2"</span><span class="ot">=</span><span class="st">"tissue"</span>)]</span>
<span id="cb22-4"><a href="#cb22-4" aria-hidden="true" tabindex="-1"></a>        [, <span class="st">`</span><span class="at">:=</span><span class="st">`</span>(<span class="at">b_diff     =</span> b[exposure<span class="sc">==</span><span class="st">"Tissue 1"</span>]<span class="sc">-</span>b[exposure<span class="sc">==</span><span class="st">"Tissue 2"</span>],</span>
<span id="cb22-5"><a href="#cb22-5" aria-hidden="true" tabindex="-1"></a>                <span class="at">snp_diff   =</span> n_sig_snps[exposure<span class="sc">==</span><span class="st">"Tissue 1"</span>]<span class="sc">-</span>n_sig_snps[exposure<span class="sc">==</span><span class="st">"Tissue 2"</span>],</span>
<span id="cb22-6"><a href="#cb22-6" aria-hidden="true" tabindex="-1"></a>                <span class="at">pph4_diff  =</span> mean_pph4_t1<span class="sc">-</span>mean_pph4_t2, </span>
<span id="cb22-7"><a href="#cb22-7" aria-hidden="true" tabindex="-1"></a>                <span class="at">nsamp_diff =</span> t1_n <span class="sc">-</span> t2_n), <span class="at">by=</span>.(t1,t2)]</span>
<span id="cb22-8"><a href="#cb22-8" aria-hidden="true" tabindex="-1"></a>       ),</span>
<span id="cb22-9"><a href="#cb22-9" aria-hidden="true" tabindex="-1"></a>       <span class="fu">aes</span>(<span class="at">y=</span>b_diff, <span class="at">x=</span>snp_diff, <span class="at">color=</span>nsamp_diff)) <span class="sc">+</span></span>
<span id="cb22-10"><a href="#cb22-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_hline</span>(<span class="at">yintercept =</span> <span class="fl">0.0</span>, <span class="at">linetype=</span><span class="st">"dotted"</span>, <span class="at">color=</span><span class="st">"grey"</span>) <span class="sc">+</span></span>
<span id="cb22-11"><a href="#cb22-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_vline</span>(<span class="at">xintercept =</span> <span class="fl">0.0</span>, <span class="at">linetype=</span><span class="st">"dotted"</span>, <span class="at">color=</span><span class="st">"grey"</span>) <span class="sc">+</span></span>
<span id="cb22-12"><a href="#cb22-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="at">alpha=</span><span class="fl">0.3</span>) <span class="sc">+</span></span>
<span id="cb22-13"><a href="#cb22-13" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_color_viridis_c</span>(<span class="at">option=</span><span class="st">"magma"</span>) <span class="sc">+</span></span>
<span id="cb22-14"><a href="#cb22-14" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_smooth</span>(<span class="at">method=</span><span class="st">"lm"</span>, <span class="at">se=</span><span class="cn">FALSE</span>, <span class="at">color=</span><span class="st">"red"</span>) <span class="sc">+</span></span>
<span id="cb22-15"><a href="#cb22-15" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_classic2</span>() <span class="sc">+</span></span>
<span id="cb22-16"><a href="#cb22-16" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">legend.position =</span> <span class="st">"top"</span>) <span class="sc">+</span></span>
<span id="cb22-17"><a href="#cb22-17" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">y =</span> <span class="fu">expression</span>(Delta[T1<span class="sc">-</span>T2]<span class="sc">~</span>MVMR<span class="sc">~</span>estimate), </span>
<span id="cb22-18"><a href="#cb22-18" aria-hidden="true" tabindex="-1"></a>       <span class="at">x =</span> <span class="fu">expression</span>(Delta[T1<span class="sc">-</span>T2]<span class="sc">~</span>No.<span class="sc">~</span>coloc<span class="sc">~</span>SNPs), </span>
<span id="cb22-19"><a href="#cb22-19" aria-hidden="true" tabindex="-1"></a>       <span class="at">color =</span> <span class="fu">expression</span>(Delta[T1<span class="sc">-</span>T2]<span class="sc">~</span>Tissue<span class="sc">~</span>sample<span class="sc">~</span>size))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="index_files/figure-html/unnamed-chunk-18-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
</section>
</section>
<section id="interpretation" class="level2">
<h2 class="anchored" data-anchor-id="interpretation">Interpretation</h2>
<p>Is the MVMR estimate mostly related to the difference in the relative number of colocalising SNPs in tissue A vs.&nbsp;tissue B? The tissue with fewer colocalising SNPs will liekly be overpowered in the MVMR by the tissue with a greater number of colocalising SNPs. Yet, the number of colocalising SNPs is closely related to the sample size of the eQTL dataset.</p>


</section>

</main> <!-- /main -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const isCodeAnnotation = (el) => {
    for (const clz of el.classList) {
      if (clz.startsWith('code-annotation-')) {                     
        return true;
      }
    }
    return false;
  }
  const onCopySuccess = function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  }
  const getTextToCopy = function(trigger) {
      const codeEl = trigger.previousElementSibling.cloneNode(true);
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
  }
  const clipboard = new window.ClipboardJS('.code-copy-button:not([data-in-quarto-modal])', {
    text: getTextToCopy
  });
  clipboard.on('success', onCopySuccess);
  if (window.document.getElementById('quarto-embedded-source-code-modal')) {
    const clipboardModal = new window.ClipboardJS('.code-copy-button[data-in-quarto-modal]', {
      text: getTextToCopy,
      container: window.document.getElementById('quarto-embedded-source-code-modal')
    });
    clipboardModal.on('success', onCopySuccess);
  }
    var localhostRegex = new RegExp(/^(?:http|https):\/\/localhost\:?[0-9]*\//);
    var mailtoRegex = new RegExp(/^mailto:/);
      var filterRegex = new RegExp('/' + window.location.host + '/');
    var isInternal = (href) => {
        return filterRegex.test(href) || localhostRegex.test(href) || mailtoRegex.test(href);
    }
    // Inspect non-navigation links and adorn them if external
 	var links = window.document.querySelectorAll('a[href]:not(.nav-link):not(.navbar-brand):not(.toc-action):not(.sidebar-link):not(.sidebar-item-toggle):not(.pagination-link):not(.no-external):not([aria-hidden]):not(.dropdown-item):not(.quarto-navigation-tool):not(.about-link)');
    for (var i=0; i<links.length; i++) {
      const link = links[i];
      if (!isInternal(link.href)) {
        // undo the damage that might have been done by quarto-nav.js in the case of
        // links that we want to consider external
        if (link.dataset.originalHref !== undefined) {
          link.href = link.dataset.originalHref;
        }
      }
    }
  function tippyHover(el, contentFn, onTriggerFn, onUntriggerFn) {
    const config = {
      allowHTML: true,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start',
    };
    if (contentFn) {
      config.content = contentFn;
    }
    if (onTriggerFn) {
      config.onTrigger = onTriggerFn;
    }
    if (onUntriggerFn) {
      config.onUntrigger = onUntriggerFn;
    }
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      if (note) {
        return note.innerHTML;
      } else {
        return "";
      }
    });
  }
  const xrefs = window.document.querySelectorAll('a.quarto-xref');
  const processXRef = (id, note) => {
    // Strip column container classes
    const stripColumnClz = (el) => {
      el.classList.remove("page-full", "page-columns");
      if (el.children) {
        for (const child of el.children) {
          stripColumnClz(child);
        }
      }
    }
    stripColumnClz(note)
    if (id === null || id.startsWith('sec-')) {
      // Special case sections, only their first couple elements
      const container = document.createElement("div");
      if (note.children && note.children.length > 2) {
        container.appendChild(note.children[0].cloneNode(true));
        for (let i = 1; i < note.children.length; i++) {
          const child = note.children[i];
          if (child.tagName === "P" && child.innerText === "") {
            continue;
          } else {
            container.appendChild(child.cloneNode(true));
            break;
          }
        }
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(container);
        }
        return container.innerHTML
      } else {
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(note);
        }
        return note.innerHTML;
      }
    } else {
      // Remove any anchor links if they are present
      const anchorLink = note.querySelector('a.anchorjs-link');
      if (anchorLink) {
        anchorLink.remove();
      }
      if (window.Quarto?.typesetMath) {
        window.Quarto.typesetMath(note);
      }
      if (note.classList.contains("callout")) {
        return note.outerHTML;
      } else {
        return note.innerHTML;
      }
    }
  }
  for (var i=0; i<xrefs.length; i++) {
    const xref = xrefs[i];
    tippyHover(xref, undefined, function(instance) {
      instance.disable();
      let url = xref.getAttribute('href');
      let hash = undefined; 
      if (url.startsWith('#')) {
        hash = url;
      } else {
        try { hash = new URL(url).hash; } catch {}
      }
      if (hash) {
        const id = hash.replace(/^#\/?/, "");
        const note = window.document.getElementById(id);
        if (note !== null) {
          try {
            const html = processXRef(id, note.cloneNode(true));
            instance.setContent(html);
          } finally {
            instance.enable();
            instance.show();
          }
        } else {
          // See if we can fetch this
          fetch(url.split('#')[0])
          .then(res => res.text())
          .then(html => {
            const parser = new DOMParser();
            const htmlDoc = parser.parseFromString(html, "text/html");
            const note = htmlDoc.getElementById(id);
            if (note !== null) {
              const html = processXRef(id, note);
              instance.setContent(html);
            } 
          }).finally(() => {
            instance.enable();
            instance.show();
          });
        }
      } else {
        // See if we can fetch a full url (with no hash to target)
        // This is a special case and we should probably do some content thinning / targeting
        fetch(url)
        .then(res => res.text())
        .then(html => {
          const parser = new DOMParser();
          const htmlDoc = parser.parseFromString(html, "text/html");
          const note = htmlDoc.querySelector('main.content');
          if (note !== null) {
            // This should only happen for chapter cross references
            // (since there is no id in the URL)
            // remove the first header
            if (note.children.length > 0 && note.children[0].tagName === "HEADER") {
              note.children[0].remove();
            }
            const html = processXRef(null, note);
            instance.setContent(html);
          } 
        }).finally(() => {
          instance.enable();
          instance.show();
        });
      }
    }, function(instance) {
    });
  }
      let selectedAnnoteEl;
      const selectorForAnnotation = ( cell, annotation) => {
        let cellAttr = 'data-code-cell="' + cell + '"';
        let lineAttr = 'data-code-annotation="' +  annotation + '"';
        const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
        return selector;
      }
      const selectCodeLines = (annoteEl) => {
        const doc = window.document;
        const targetCell = annoteEl.getAttribute("data-target-cell");
        const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
        const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
        const lines = annoteSpan.getAttribute("data-code-lines").split(",");
        const lineIds = lines.map((line) => {
          return targetCell + "-" + line;
        })
        let top = null;
        let height = null;
        let parent = null;
        if (lineIds.length > 0) {
            //compute the position of the single el (top and bottom and make a div)
            const el = window.document.getElementById(lineIds[0]);
            top = el.offsetTop;
            height = el.offsetHeight;
            parent = el.parentElement.parentElement;
          if (lineIds.length > 1) {
            const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
            const bottom = lastEl.offsetTop + lastEl.offsetHeight;
            height = bottom - top;
          }
          if (top !== null && height !== null && parent !== null) {
            // cook up a div (if necessary) and position it 
            let div = window.document.getElementById("code-annotation-line-highlight");
            if (div === null) {
              div = window.document.createElement("div");
              div.setAttribute("id", "code-annotation-line-highlight");
              div.style.position = 'absolute';
              parent.appendChild(div);
            }
            div.style.top = top - 2 + "px";
            div.style.height = height + 4 + "px";
            div.style.left = 0;
            let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
            if (gutterDiv === null) {
              gutterDiv = window.document.createElement("div");
              gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
              gutterDiv.style.position = 'absolute';
              const codeCell = window.document.getElementById(targetCell);
              const gutter = codeCell.querySelector('.code-annotation-gutter');
              gutter.appendChild(gutterDiv);
            }
            gutterDiv.style.top = top - 2 + "px";
            gutterDiv.style.height = height + 4 + "px";
          }
          selectedAnnoteEl = annoteEl;
        }
      };
      const unselectCodeLines = () => {
        const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
        elementsIds.forEach((elId) => {
          const div = window.document.getElementById(elId);
          if (div) {
            div.remove();
          }
        });
        selectedAnnoteEl = undefined;
      };
        // Handle positioning of the toggle
    window.addEventListener(
      "resize",
      throttle(() => {
        elRect = undefined;
        if (selectedAnnoteEl) {
          selectCodeLines(selectedAnnoteEl);
        }
      }, 10)
    );
    function throttle(fn, ms) {
    let throttle = false;
    let timer;
      return (...args) => {
        if(!throttle) { // first call gets through
            fn.apply(this, args);
            throttle = true;
        } else { // all the others get throttled
            if(timer) clearTimeout(timer); // cancel #2
            timer = setTimeout(() => {
              fn.apply(this, args);
              timer = throttle = false;
            }, ms);
        }
      };
    }
      // Attach click handler to the DT
      const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
      for (const annoteDlNode of annoteDls) {
        annoteDlNode.addEventListener('click', (event) => {
          const clickedEl = event.target;
          if (clickedEl !== selectedAnnoteEl) {
            unselectCodeLines();
            const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
            if (activeEl) {
              activeEl.classList.remove('code-annotation-active');
            }
            selectCodeLines(clickedEl);
            clickedEl.classList.add('code-annotation-active');
          } else {
            // Unselect the line
            unselectCodeLines();
            clickedEl.classList.remove('code-annotation-active');
          }
        });
      }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script>
</div> <!-- /content -->




</body></html>