---
title: "Tissue-partitioned BMI instrument procedure"
author: "Nick Sunderland"
date: "2025-01-10"
execute:
  fig-path: "posts/2025-01-10-tissue-partitioned-mvmr/" 
---

## Motivation
What does the weighting by PPH4 do to tissue partitioned BMI instruments? Let's simulate some tissue partitioned BMI MR results.

```{r load-packages}
#| warning: false
#| message: false

library(data.table)
library(ggplot2)
library(ggpubr)
library(genepi.utils)
library(tibble)
library(knitr)
```


## BMI data
First load the BMI GWAS.

```{r}
#| message: false

bmi    <- readRDS(file.path(TPMR_DIR, "output", "tmp_objects", "obj_bmi.RDS"))
clumps <- clump(as.data.table(bmi), p1=5e-8, r2=0.001, kb=10000, plink2=PLINK2, plink_ref=UKBB_REFERENCE)[index==TRUE]
bmi    <- subset_gwas(bmi, clumps$rsid)
kable(head(clumps, 5), caption = "BMI lead variants")
```

## Heart failure data
Load the heart failure GWAS.

```{r}
#| message: false

hf <- readRDS(file.path(TPMR_DIR, "output", "tmp_objects", "obj_hf_allcause.RDS"))
hf <- subset_gwas(hf, clumps$rsid)
```

## Harmonise the datasets
```{r}
#| message: false

h <- MR(bmi, hf)
kable(head(as.data.table(h),5), caption = "Harmonised BMI:HF dataset")
```

## Run the MR
```{r}
res <- run_mr(h)
kable(res, caption = "BMI:HF Mendelian randomisation")
```

## Plot MR
```{r, warning=FALSE, message=FALSE}
plot_mr(h, res)
```

## Select variants - MR with a subset of BMI variants
In the Leyden et al. 2022 paper 86 variant colocalise with adipose tissue eQTLs and 140 colocalise with brain tissue eQTLs. There is overlap between the two tissues of 43 variants.
```{r, warning=FALSE, message=FALSE}
#| warning: false
#| message: false

set.seed(123)
tissue1_varid <- sample(h@snps, 140)
overlap_varid <- sample(tissue1_varid, 43)
tissue2_varid <- c(overlap_varid, sample(h@snps[!h@snps %in% overlap_varid], 86-43))

# turn of variants not in either tissue
h@index_snp[!h@snps %in% c(tissue1_varid, tissue2_varid)] <- FALSE
h@exposure <- "BMI tissue 1/2 SNPs"

res1 <- run_mr(h)
kable(res1, caption = "BMI:HF Mendelian randomisation (Tissue 1 & 2 SNPs only)")

plot_mr(h, res1)
```

## Run multivariable MR (unweighted)
```{r}
#| warning: false
#| message: false

# copy of BMI for tissue 1
bmi_t1 <- bmi
bmi_t1@id <- "bmi tissue 1"
bmi_t1@trait <- "bmi tissue 1"

# copy of BMI for tissue 2
bmi_t2 <- bmi
bmi_t2@id <- "bmi tissue 2"
bmi_t2@trait <- "bmi tissue 2"

# harmonise
h_mvmr <- MR(list(bmi_t1, bmi_t2), hf)

# turn of variants not in either tissue
h_mvmr@index_snp[!h_mvmr@snps %in% c(tissue1_varid, tissue2_varid)] <- FALSE

res2 <- run_mr(h_mvmr, methods="mr_ivw")
kable(res2, caption = "BMI:HF MV Mendelian randomisation")

plot_mr(h_mvmr, res2)
```

## BMI:eQTL PPH4 distributions by tissue

### Number of colocalising variants per tissue
```{r}
coloc_thresh <- 0.8

coloc_results <- fread(file.path(TPMR_DIR, "output", "tables", "coloc_results.tsv.gz"))
coloc_counts <- coloc_results[locus_trait=="bmi" & grepl("^bmi", trait.x) & !grepl("^bmi|^hf_", trait.y), .SD[which.max(pph4)], by=.(locus_index_rsid,trait.y)][, .(`num_coloc_pph4` = sum(pph4>coloc_thresh)), by=c(tissue="trait.y")][, tissue := factor(tissue, levels = tissue[order(-`num_coloc_pph4`)])]

ggplot(coloc_counts, aes(y = tissue, x = `num_coloc_pph4`)) +
  theme_classic() +
  geom_col(fill= "blue", alpha = 0.3) +
  labs(subtitle = paste("coloc thresh:", coloc_thresh), 
       x = "Number of colocalising BMI variants") +
  theme(axis.title.y = element_blank())
```

### Max. PPH4 distributions by tissue
Take the maximum PPH4 from genes in close proximity to each variant, in each tissue. 
```{r}
num_tissues <- 20
tissues_subset <- levels(coloc_counts$tissue)[as.integer(seq(1, length(levels(coloc_counts$tissue)), length.out=num_tissues))] 
coloc_subset <- coloc_results[locus_trait=="bmi" & trait.x=="bmi" & (trait.y %in% tissues_subset), .SD[which.max(pph4)], by=.(locus_index_rsid,trait.y)]

density <- coloc_subset[, {
  den <- density(pph4, n=1000)
  .(pph4 = scales::rescale(den$x, 0:1), prob = den$y)
}, by = c(tissue="trait.y")][, tissue := factor(tissue, levels = levels(coloc_counts$tissue))]

ggplot(density, aes(x=pph4, y=prob, color=tissue)) +
  theme_classic() +
  geom_line() +
  labs(y = "Density")
```

## Can we achieve these densities randomly 
Yes, just sample the PPH4 from the actual density distributions above.
```{r}
#| warning: false
#| message: false

tissues <- expand.grid(tissue = unique(density$tissue), stringsAsFactors = F) |> as.data.table()

set.seed(123)

res_sim_density <- tissues[, {
  .(sim_pph4 = sample(density[tissue==.BY$tissue, pph4], length(h_mvmr@bx[,1]), prob = density[tissue==.BY$tissue, prob], replace = TRUE))
}, by = .(tissue)]


ggplot(res_sim_density, aes(x=sim_pph4, color=tissue)) +
  theme_classic() +
  geom_density() +
  labs(y = "Simulated density", 
       x = "Simulated PPH4")
```


## Effect different PPH4 weights
Weight the BMI betas by their respective maximum PPH4 value, see how this changes the relationship between the beta for BMI and beta for HF, dependent on which tissue is providing the betas. Tissues with greater or fewer colocalising variants affect push the betas up or down more, respectively. I hold one tissue constant (reference) to more easily appreciate how the other tissue's colocalisation characteristics affect things. 
```{r}
#| warning: false
#| message: false

ref_tissue <- levels(density$tissue)[1]
tissue_pairs <- expand.grid(t1 = ref_tissue, t2 = unique(density$tissue), stringsAsFactors = F) |> as.data.table()
tissue_pairs[density[, .(mean_pph4 = mean(pph4*prob)), by=tissue], t1_mean_pph4 := i.mean_pph4, on=c("t1"="tissue")]
tissue_pairs[density[, .(mean_pph4 = mean(pph4*prob)), by=tissue], t2_mean_pph4 := i.mean_pph4, on=c("t2"="tissue")]

set.seed(123)
t1_pph4 <- sample(density[tissue==ref_tissue, pph4], length(h_mvmr@bx[,1]), prob = density[tissue==ref_tissue, prob], replace = TRUE)

res_weighted <- tissue_pairs[, {
  h_tmp <- h_mvmr
  t2_pph4 <- sample(density[tissue==.BY$t2, pph4], length(h_tmp@bx[,1]), prob = density[tissue==.BY$t2, prob], replace = TRUE)
  h_tmp@bx[,1] <- h_tmp@bx[,1] * t1_pph4
  h_tmp@bx[,2] <- h_tmp@bx[,2] * t2_pph4
  h_tmp@index_snp <- ifelse(t1_pph4>coloc_thresh | t2_pph4>coloc_thresh, TRUE, FALSE)
  .(tissue   = rep(c(paste0("Ref: ", ref_tissue),"Comparator"), each=sum(h_tmp@index_snp)), 
    beta_bmi = c(h_tmp@bx[h_tmp@index_snp,1], h_tmp@bx[h_tmp@index_snp,2]), 
    beta_hf  = rep(h_tmp@by[h_tmp@index_snp],2),
    mean_pph4_t1 = mean(t1_pph4[t1_pph4>coloc_thresh | t2_pph4>coloc_thresh]),
    mean_pph4_t2 = mean(t2_pph4[t1_pph4>coloc_thresh | t2_pph4>coloc_thresh]))
}, by = .(t1,t2)]


ggplot(res_weighted, aes(x=beta_bmi, y=beta_hf, color=tissue)) +
  geom_point() +
  geom_text(data = unique(res_weighted[grepl("Ref", tissue)], by="t2"), 
            aes(x = 0.0, 
                y = -0.04,
                label = sprintf("Tis mPPH4=%.3f", mean_pph4_t2)),
            color="black", size=2) +
  geom_text(data = unique(res_weighted[!grepl("Ref", tissue)], by="t2"), 
            aes(x = 0.0, 
                y = -0.07,
                label = sprintf("Ref mPPH4=%.3f", mean_pph4_t1)),
            color="black", size=2) +
  geom_smooth(method="lm", se=F,fullrange = TRUE) +
  facet_wrap(~t2, labeller = labeller(t2 = function(x) gsub("_", " ", x))) +
  labs(y = "Heart failure beta", 
       x = "BMI beta (PPH4-weighted)", 
       color = "Tissue")
```

## IVW regression by different PPH4 weighting distributions
Perform MVMR with the weighted BMI instruments. Lets look at holding a poorly colocalising tissue constant compared to the best colocalising tissues.

### Reference: top colocalising tissue
Different distributions against the top colocalising tissue. 
```{r}
#| warning: false
#| message: false

ref_tissue <- levels(density$tissue)[1] # top colocalising tissue
tissue_pairs <- expand.grid(t1 = ref_tissue, t2 = unique(density$tissue), stringsAsFactors = F) |> as.data.table()
tissue_pairs[density[, .(mean_pph4 = mean(pph4*prob)), by=tissue], t1_mean_pph4 := i.mean_pph4, on=c("t1"="tissue")]
tissue_pairs[density[, .(mean_pph4 = mean(pph4*prob)), by=tissue], t2_mean_pph4 := i.mean_pph4, on=c("t2"="tissue")]

set.seed(123)
t1_pph4 <- sample(density[tissue==ref_tissue, pph4], length(h_mvmr@bx[,1]), prob = density[tissue==ref_tissue, prob], replace = TRUE)

mr_weighted <- tissue_pairs[, {
  h_tmp <- h_mvmr
  t2_pph4 <- sample(density[tissue==.BY$t2, pph4], length(h_tmp@bx[,1]), prob = density[tissue==.BY$t2, prob], replace = TRUE)
  h_tmp@bx[,1] <- h_tmp@bx[,1] * t1_pph4
  h_tmp@bx[,2] <- h_tmp@bx[,2] * t2_pph4
  h_tmp@index_snp <- ifelse(t1_pph4>coloc_thresh | t2_pph4>coloc_thresh, TRUE, FALSE)
  h_tmp@exposure[1] <- paste0("Ref: ", ref_tissue)
  h_tmp@exposure[2] <- "Comparator"
  r <- run_mr(h_tmp, methods="mr_ivw")
  r[, `:=`(n_sig_snps = c(sum(t1_pph4>coloc_thresh), sum(t2_pph4>coloc_thresh)), 
           n_overlap_snps = sum(t1_pph4>coloc_thresh & t2_pph4>coloc_thresh))]
}, by = .(t1,t2,t1_mean_pph4,t2_mean_pph4)]

ggplot(mr_weighted, aes(x=t2, y=exp(b), ymin=exp(b-b_se*1.96), ymax=exp(b+b_se*1.96), color=exposure)) +
  geom_errorbar(width=0.2, position = position_dodge(0.2)) +
  geom_point(aes(size=n_sig_snps), position = position_dodge(0.2)) +
  theme(axis.text.x = element_text(angle=45, hjust=1), 
        axis.title.y = element_blank()) +
  labs(y = "MR estimate (tpBMI vs HF)", color = "Tissue", size = "Num. SNPs contributed to model")
```
### Reference: worst colocalising tissue
Then, different distributions against the worst colocalising tissue. 
```{r}
#| warning: false
#| message: false

ref_tissue <- levels(density$tissue)[length(levels(density$tissue))] # bottom/worst colocalising tissue

tissue_pairs <- expand.grid(t1 = ref_tissue, t2 = unique(density$tissue), stringsAsFactors = F) |> as.data.table()
tissue_pairs[density[, .(mean_pph4 = mean(pph4*prob)), by=tissue], t1_mean_pph4 := i.mean_pph4, on=c("t1"="tissue")]
tissue_pairs[density[, .(mean_pph4 = mean(pph4*prob)), by=tissue], t2_mean_pph4 := i.mean_pph4, on=c("t2"="tissue")]

set.seed(123)
t1_pph4 <- sample(density[tissue==ref_tissue, pph4], length(h_mvmr@bx[,1]), prob = density[tissue==ref_tissue, prob], replace = TRUE)

mr_weighted <- tissue_pairs[, {
  h_tmp <- h_mvmr
  t2_pph4 <- sample(density[tissue==.BY$t2, pph4], length(h_tmp@bx[,1]), prob = density[tissue==.BY$t2, prob], replace = TRUE)
  h_tmp@bx[,1] <- h_tmp@bx[,1] * t1_pph4
  h_tmp@bx[,2] <- h_tmp@bx[,2] * t2_pph4
  h_tmp@index_snp <- ifelse(t1_pph4>coloc_thresh | t2_pph4>coloc_thresh, TRUE, FALSE)
  h_tmp@exposure[1] <- paste0("Ref: ", ref_tissue)
  h_tmp@exposure[2] <- "Comparator"
  r <- run_mr(h_tmp, methods="mr_ivw")
  r[, `:=`(n_sig_snps = c(sum(t1_pph4>coloc_thresh), sum(t2_pph4>coloc_thresh)), 
           n_overlap_snps = sum(t1_pph4>coloc_thresh & t2_pph4>coloc_thresh))]
}, by = .(t1,t2,t1_mean_pph4,t2_mean_pph4)]

ggplot(mr_weighted, aes(x=t2, y=exp(b), ymin=exp(b-b_se*1.96), ymax=exp(b+b_se*1.96), color=exposure)) +
  geom_errorbar(width=0.2, position = position_dodge(0.2)) +
  geom_point(aes(size=n_sig_snps), position = position_dodge(0.2)) +
  theme(axis.text.x = element_text(angle=45, hjust=1), 
        axis.title.y = element_blank()) +
  labs(y = "MR estimate (tpBMI vs HF)", color = "Tissue", size = "Num. SNPs contributed to model")
```


## Number of colocalising loci
Is it just mostly to do with the difference in the number of colocalising variants in each tissue. Run MR for all pairs of tissues.
```{r}
#| warning: false
#| message: false

tissue_pairs <- expand.grid(t1 = unique(density$tissue), t2 = unique(density$tissue), stringsAsFactors = F) |> as.data.table()
tissue_pairs[density[, .(mean_pph4 = mean(pph4*prob)), by=tissue], t1_mean_pph4 := i.mean_pph4, on=c("t1"="tissue")]
tissue_pairs[density[, .(mean_pph4 = mean(pph4*prob)), by=tissue], t2_mean_pph4 := i.mean_pph4, on=c("t2"="tissue")]

set.seed(123)
mr_weighted <- tissue_pairs[, {
  h_tmp <- h_mvmr
  t1_pph4 <- sample(density[tissue==.BY$t1, pph4], length(h_tmp@bx[,1]), prob = density[tissue==.BY$t1, prob], replace = TRUE)
  t2_pph4 <- sample(density[tissue==.BY$t2, pph4], length(h_tmp@bx[,1]), prob = density[tissue==.BY$t2, prob], replace = TRUE)
  h_tmp@bx[,1] <- h_tmp@bx[,1] * t1_pph4
  h_tmp@bx[,2] <- h_tmp@bx[,2] * t2_pph4
  h_tmp@index_snp <- ifelse(t1_pph4>coloc_thresh | t2_pph4>coloc_thresh, TRUE, FALSE)
  h_tmp@exposure[1] <- "t1"
  h_tmp@exposure[2] <- "t2"
  r <- run_mr(h_tmp, methods="mr_ivw")
  r[, `:=`(n_sig_snps = c(sum(t1_pph4>coloc_thresh), sum(t2_pph4>coloc_thresh)),
           n_uni_snps = c(length(setdiff(which(t1_pph4>coloc_thresh), which(t2_pph4>coloc_thresh))), length(setdiff(which(t2_pph4>coloc_thresh), which(t1_pph4>coloc_thresh)))),
           n_overlap_snps = sum(t1_pph4>coloc_thresh & t2_pph4>coloc_thresh), 
           n_sig_diff = sum(t1_pph4>coloc_thresh) - sum(t2_pph4>coloc_thresh), 
           exp_b_diff = diff(range(exp(b))))]
}, by = .(t1,t2,t1_mean_pph4,t2_mean_pph4)]

ggplot(mr_weighted, aes(x=n_sig_snps, y=exp(b), ymin=exp(b-b_se*1.96), ymax=exp(b+b_se*1.96), color=exposure)) +
  geom_errorbar(width=0.2, position = position_dodge(0.2)) +
  geom_point(position = position_dodge(0.2)) +
  theme(axis.text.x = element_text(angle=45, hjust=1), 
        axis.title.y = element_blank()) +
  labs(y = "MR estimate (tpBMI vs HF)", color = "Tissue", size = "Num. SNPs contributed to model") +
  facet_wrap(~exposure)
```

## Interpretation
....